<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Definition of Clock, MG and ML scores | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</title>
  <meta name="description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Definition of Clock, MG and ML scores | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="tanaylab.github.io/metabric_rrbs" />
  
  <meta property="og:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="github-repo" content="tanaylab/metabric_rrbs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Definition of Clock, MG and ML scores | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  
  <meta name="twitter:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  

<meta name="author" content="Aviezer Lifshitz" />


<meta name="date" content="2021-08-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tumor-microenvironment-tme-deconvolution.html"/>
<link rel="next" href="loss-clock.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#run-the-notebooks"><i class="fa fa-check"></i><b>1.1</b> Run the notebooks</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notebook-order"><i class="fa fa-check"></i><b>1.3</b> Notebook order</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#find-a-specific-figure"><i class="fa fa-check"></i><b>1.4</b> Find a specific figure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html"><i class="fa fa-check"></i><b>2</b> METABRIC methylation data statistics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#breakdown-of-samples"><i class="fa fa-check"></i><b>2.1</b> Breakdown of samples</a></li>
<li class="chapter" data-level="2.2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#number-of-samples"><i class="fa fa-check"></i><b>2.2</b> Number of samples</a></li>
<li class="chapter" data-level="2.3" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#total-number-of-reads"><i class="fa fa-check"></i><b>2.3</b> Total number of reads</a></li>
<li class="chapter" data-level="2.4" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-cpg-coverage-per-sample"><i class="fa fa-check"></i><b>2.4</b> Distribution of CpG coverage per sample</a></li>
<li class="chapter" data-level="2.5" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-covered-cpgs"><i class="fa fa-check"></i><b>2.5</b> Distribution of covered CpGs</a></li>
<li class="chapter" data-level="2.6" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#promoter-coverage"><i class="fa fa-check"></i><b>2.6</b> Promoter coverage</a></li>
<li class="chapter" data-level="2.7" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#coverage-per-clinical-annotations"><i class="fa fa-check"></i><b>2.7</b> Coverage per clinical annotations</a></li>
<li class="chapter" data-level="2.8" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage"><i class="fa fa-check"></i><b>2.8</b> Marginal CpG coverage</a><ul>
<li class="chapter" data-level="2.8.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage-of-k4me1-hotspots"><i class="fa fa-check"></i><b>2.8.1</b> Marginal CpG coverage of k4me1 hotspots</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#sample-cellularity"><i class="fa fa-check"></i><b>2.9</b> Sample cellularity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html"><i class="fa fa-check"></i><b>3</b> Tumor microenvironment (TME) deconvolution</a><ul>
<li class="chapter" data-level="3.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#initialize"><i class="fa fa-check"></i><b>3.2</b> Initialize</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#load-data"><i class="fa fa-check"></i><b>3.3</b> Load data</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#deconvolve-tme"><i class="fa fa-check"></i><b>3.4</b> Deconvolve TME</a><ul>
<li class="chapter" data-level="3.4.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er-samples"><i class="fa fa-check"></i><b>3.4.1</b> ER+ samples</a></li>
<li class="chapter" data-level="3.4.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er--samples"><i class="fa fa-check"></i><b>3.4.2</b> ER- samples</a></li>
<li class="chapter" data-level="3.4.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#normal-samples"><i class="fa fa-check"></i><b>3.4.3</b> Normal samples</a></li>
<li class="chapter" data-level="3.4.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-normalized-methylation"><i class="fa fa-check"></i><b>3.4.4</b> Merge normalized methylation</a></li>
<li class="chapter" data-level="3.4.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-tme-scores"><i class="fa fa-check"></i><b>3.4.5</b> Merge TME scores</a></li>
<li class="chapter" data-level="3.4.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#caf-and-immune-genes"><i class="fa fa-check"></i><b>3.4.6</b> CAF and Immune genes</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#expression-methylation-correlation-matrices"><i class="fa fa-check"></i><b>3.5</b> Expression-methylation correlation matrices</a><ul>
<li class="chapter" data-level="3.5.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#export-clustering"><i class="fa fa-check"></i><b>3.5.1</b> Export clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#diagnose-tme-normalization"><i class="fa fa-check"></i><b>3.6</b> Diagnose TME normalization</a></li>
<li class="chapter" data-level="3.7" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#plot-cd3d-and-cav1-vs-immune-and-caf-methylation-modules"><i class="fa fa-check"></i><b>3.7</b> Plot CD3D and CAV1 vs immune and CAF methylation modules</a></li>
<li class="chapter" data-level="3.8" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#annotate-tme-scores"><i class="fa fa-check"></i><b>3.8</b> Annotate TME scores</a></li>
<li class="chapter" data-level="3.9" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#appendix-sensitivity-analysis-of-tme-normalization"><i class="fa fa-check"></i><b>3.9</b> Appendix: Sensitivity analysis of TME normalization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html"><i class="fa fa-check"></i><b>4</b> Definition of Clock, MG and ML scores</a><ul>
<li class="chapter" data-level="4.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#initialize-1"><i class="fa fa-check"></i><b>4.2</b> Initialize</a></li>
<li class="chapter" data-level="4.3" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#load-data-1"><i class="fa fa-check"></i><b>4.3</b> Load data</a></li>
<li class="chapter" data-level="4.4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#clustering-of-normalized-methylation-of-er-tumors"><i class="fa fa-check"></i><b>4.4</b> Clustering of normalized methylation of ER+ tumors</a></li>
<li class="chapter" data-level="4.5" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-clustering-on-er--correlation-matrix"><i class="fa fa-check"></i><b>4.5</b> Project clustering on ER- correlation matrix</a></li>
<li class="chapter" data-level="4.6" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#classify-loci-in-the-genome-to-epigenomic-features"><i class="fa fa-check"></i><b>4.6</b> Classify loci in the genome to epigenomic features</a></li>
<li class="chapter" data-level="4.7" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#average-methylation-of-the-features"><i class="fa fa-check"></i><b>4.7</b> Average methylation of the features</a></li>
<li class="chapter" data-level="4.8" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-distribution-over-samples-of-the-correlated-loci"><i class="fa fa-check"></i><b>4.8</b> Plot distribution over samples of the correlated loci</a></li>
<li class="chapter" data-level="4.9" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-loci-classification"><i class="fa fa-check"></i><b>4.9</b> Plot loci classification</a></li>
<li class="chapter" data-level="4.10" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-all-epigenomic-features-using-umap"><i class="fa fa-check"></i><b>4.10</b> Project all epigenomic features using UMAP</a></li>
<li class="chapter" data-level="4.11" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#compare-epigenomic-scores-to-clinical-annotations"><i class="fa fa-check"></i><b>4.11</b> Compare epigenomic scores to clinical annotations</a></li>
<li class="chapter" data-level="4.12" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#distribution-of-the-scores-in-ic10-integrative-clusters"><i class="fa fa-check"></i><b>4.12</b> Distribution of the scores in iC10 integrative clusters</a></li>
<li class="chapter" data-level="4.13" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#epipolymorphism-of-the-epigenomic-layers"><i class="fa fa-check"></i><b>4.13</b> Epipolymorphism of the epigenomic layers</a></li>
<li class="chapter" data-level="4.14" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#comparison-of-the-epigenomic-scores-with-existing-epigenomic-age-metrics"><i class="fa fa-check"></i><b>4.14</b> Comparison of the epigenomic scores with existing epigenomic age metrics</a><ul>
<li class="chapter" data-level="4.14.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#phenoage"><i class="fa fa-check"></i><b>4.14.1</b> phenoAge</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="loss-clock.html"><a href="loss-clock.html"><i class="fa fa-check"></i><b>5</b> Loss clock</a><ul>
<li class="chapter" data-level="5.1" data-path="loss-clock.html"><a href="loss-clock.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="loss-clock.html"><a href="loss-clock.html#initialize-2"><i class="fa fa-check"></i><b>5.2</b> Initialize</a></li>
<li class="chapter" data-level="5.3" data-path="loss-clock.html"><a href="loss-clock.html#plot-methylation-distribution-of-the-clock"><i class="fa fa-check"></i><b>5.3</b> Plot methylation distribution of the clock</a></li>
<li class="chapter" data-level="5.4" data-path="loss-clock.html"><a href="loss-clock.html#annotate-clock-score"><i class="fa fa-check"></i><b>5.4</b> Annotate &quot;clock&quot; score</a><ul>
<li class="chapter" data-level="5.4.1" data-path="loss-clock.html"><a href="loss-clock.html#loci-annotation"><i class="fa fa-check"></i><b>5.4.1</b> Loci annotation</a></li>
<li class="chapter" data-level="5.4.2" data-path="loss-clock.html"><a href="loss-clock.html#clinical-annotation"><i class="fa fa-check"></i><b>5.4.2</b> Clinical annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="loss-clock.html"><a href="loss-clock.html#plot-chromosomal-traces-of-clock-score"><i class="fa fa-check"></i><b>5.5</b> Plot chromosomal traces of clock score</a></li>
<li class="chapter" data-level="5.6" data-path="loss-clock.html"><a href="loss-clock.html#correlation-of-gene-expression-to-clock"><i class="fa fa-check"></i><b>5.6</b> Correlation of gene expression to clock</a></li>
<li class="chapter" data-level="5.7" data-path="loss-clock.html"><a href="loss-clock.html#plot-score-in-different-tor-regimes"><i class="fa fa-check"></i><b>5.7</b> Plot score in different TOR regimes</a></li>
<li class="chapter" data-level="5.8" data-path="loss-clock.html"><a href="loss-clock.html#plot-high-low-clock-tumors-vs-tor"><i class="fa fa-check"></i><b>5.8</b> Plot High-Low clock tumors vs TOR</a></li>
<li class="chapter" data-level="5.9" data-path="loss-clock.html"><a href="loss-clock.html#plot-correlation-of-loci-to-clock-methylation"><i class="fa fa-check"></i><b>5.9</b> Plot correlation of loci to clock methylation</a></li>
<li class="chapter" data-level="5.10" data-path="loss-clock.html"><a href="loss-clock.html#plot-distribution-of-methylation-of-clock-in-normal-samples"><i class="fa fa-check"></i><b>5.10</b> Plot distribution of methylation of clock in normal samples</a></li>
<li class="chapter" data-level="5.11" data-path="loss-clock.html"><a href="loss-clock.html#compare-clock-layer-to-biological-age"><i class="fa fa-check"></i><b>5.11</b> Compare clock layer to biological age</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html"><i class="fa fa-check"></i><b>6</b> Epigenomic instability</a><ul>
<li class="chapter" data-level="6.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#compare-features-to-each-other"><i class="fa fa-check"></i><b>6.1</b> Compare features to each other</a></li>
<li class="chapter" data-level="6.2" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#annotate-epignomic-instability-scores"><i class="fa fa-check"></i><b>6.2</b> Annotate epignomic instability scores</a><ul>
<li class="chapter" data-level="6.2.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#integrative-clusters"><i class="fa fa-check"></i><b>6.2.1</b> Integrative clusters</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#cross-correlation-methylation-expression-of-epigenomic-instability-loci"><i class="fa fa-check"></i><b>6.3</b> Cross correlation (methylation-expression) of epigenomic instability loci</a></li>
<li class="chapter" data-level="6.4" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#epigenomic-instability-in-different-time-of-replication-tor-regimes"><i class="fa fa-check"></i><b>6.4</b> Epigenomic instability in different Time of replication (TOR) regimes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cna.html"><a href="cna.html"><i class="fa fa-check"></i><b>7</b> CNA</a></li>
<li class="chapter" data-level="8" data-path="mutations.html"><a href="mutations.html"><i class="fa fa-check"></i><b>8</b> Mutations</a></li>
<li class="chapter" data-level="9" data-path="survival.html"><a href="survival.html"><i class="fa fa-check"></i><b>9</b> Survival</a><ul>
<li class="chapter" data-level="9.0.1" data-path="survival.html"><a href="survival.html#raw-methylation"><i class="fa fa-check"></i><b>9.0.1</b> Raw-methylation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html"><i class="fa fa-check"></i><b>10</b> Methylation - Expression <em>in cis</em></a><ul>
<li class="chapter" data-level="10.0.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#remove-tme-genes"><i class="fa fa-check"></i><b>10.0.1</b> Remove TME genes</a></li>
<li class="chapter" data-level="10.0.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#load-normalized-methylation-and-separate-it-to-promoters-and-non-promoters"><i class="fa fa-check"></i><b>10.0.2</b> Load normalized methylation and separate it to promoters and non-promoters</a></li>
<li class="chapter" data-level="10.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-promoters"><i class="fa fa-check"></i><b>10.1</b> Cis candidates: promoters</a><ul>
<li class="chapter" data-level="10.1.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-correlation-of-top-cis-regulated-promoter-candidates"><i class="fa fa-check"></i><b>10.1.1</b> Plot correlation of top cis-regulated promoter candidates</a></li>
<li class="chapter" data-level="10.1.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#annotation-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.2</b> Annotation of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#gene-expression-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.3</b> Gene expression of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#correlation-between-cis-regulated-genes"><i class="fa fa-check"></i><b>10.1.4</b> Correlation between cis-regulated genes</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.2</b> Epipolymorphism of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#promoter-examples"><i class="fa fa-check"></i><b>10.3</b> Promoter examples</a><ul>
<li class="chapter" data-level="10.3.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#brca1"><i class="fa fa-check"></i><b>10.3.1</b> BRCA1</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-genomic"><i class="fa fa-check"></i><b>10.4</b> Cis candidates: genomic</a><ul>
<li class="chapter" data-level="10.4.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-cis"><i class="fa fa-check"></i><b>10.4.1</b> Plot % cis</a></li>
<li class="chapter" data-level="10.4.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-distance-to-most-correlated-gene"><i class="fa fa-check"></i><b>10.4.2</b> Plot distance to most correlated gene</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-genomic"><i class="fa fa-check"></i><b>10.5</b> Epipolymorphism of cis-regulated genomic</a></li>
<li class="chapter" data-level="10.6" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#examples-for-cis-regulation-non-promoters"><i class="fa fa-check"></i><b>10.6</b> Examples for cis regulation (non-promoters)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html"><i class="fa fa-check"></i><b>11</b> X dosage compensation</a><ul>
<li class="chapter" data-level="11.1" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#x-analysis"><i class="fa fa-check"></i><b>11.1</b> X analysis</a></li>
<li class="chapter" data-level="11.2" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#autosome-analysis"><i class="fa fa-check"></i><b>11.2</b> autosome analysis</a></li>
<li class="chapter" data-level="11.3" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#checking-dosage-compensation-on-all-x-loci"><i class="fa fa-check"></i><b>11.3</b> Checking dosage compensation on all X loci</a></li>
<li class="chapter" data-level="11.4" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#of-x-lost"><i class="fa fa-check"></i><b>11.4</b> % of X lost</a></li>
<li class="chapter" data-level="11.5" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#plot-expression-methylation-correlation-for-x"><i class="fa fa-check"></i><b>11.5</b> Plot Expression-Methylation correlation for X</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="finding-specific-dosage-compensation-examples.html"><a href="finding-specific-dosage-compensation-examples.html"><i class="fa fa-check"></i><b>12</b> Finding specific dosage compensation examples</a></li>
<li class="chapter" data-level="13" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html"><i class="fa fa-check"></i><b>13</b> Export supplementary tables</a><ul>
<li class="chapter" data-level="13.1" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-1-metabric-samples-metadata"><i class="fa fa-check"></i><b>13.1</b> Table 1: METABRIC samples metadata</a></li>
<li class="chapter" data-level="13.2" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-2-immune-expression-signature"><i class="fa fa-check"></i><b>13.2</b> Table 2: Immune expression signature</a></li>
<li class="chapter" data-level="13.3" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-3-caf-expression-signature"><i class="fa fa-check"></i><b>13.3</b> Table 3: CAF expression signature</a></li>
<li class="chapter" data-level="13.4" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-4-expression-methylation-correlation"><i class="fa fa-check"></i><b>13.4</b> Table 4: Expression-Methylation correlation</a></li>
<li class="chapter" data-level="13.5" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-5-methylation-scores"><i class="fa fa-check"></i><b>13.5</b> Table 5: Methylation scores</a></li>
<li class="chapter" data-level="13.6" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-6-gene-expression-correlation-to-epigenomic-instability"><i class="fa fa-check"></i><b>13.6</b> Table 6: Gene expression correlation to epigenomic instability</a></li>
<li class="chapter" data-level="13.7" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-7-methylation-layers-loci"><i class="fa fa-check"></i><b>13.7</b> Table 7: Methylation layers loci</a></li>
<li class="chapter" data-level="13.8" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-8-cis-regulation-candidates"><i class="fa fa-check"></i><b>13.8</b> Table 8: Cis regulation candidates</a></li>
<li class="chapter" data-level="13.9" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-9-dosage-compensation-in-autosomes"><i class="fa fa-check"></i><b>13.9</b> Table 9: Dosage compensation in autosomes</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html"><i class="fa fa-check"></i><b>14</b> Files needed for the analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#misha-db"><i class="fa fa-check"></i><b>14.1</b> Misha DB</a></li>
<li class="chapter" data-level="14.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#additional-files"><i class="fa fa-check"></i><b>14.2</b> Additional files</a><ul>
<li class="chapter" data-level="14.2.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#metabric-data"><i class="fa fa-check"></i><b>14.2.1</b> METABRIC data</a></li>
<li class="chapter" data-level="14.2.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#notable-generated-files"><i class="fa fa-check"></i><b>14.2.2</b> Notable generated files</a></li>
<li class="chapter" data-level="14.2.3" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#spatial-methylation-distribution"><i class="fa fa-check"></i><b>14.2.3</b> Spatial methylation distribution</a></li>
<li class="chapter" data-level="14.2.4" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#copy-number-abberations"><i class="fa fa-check"></i><b>14.2.4</b> Copy number abberations</a></li>
<li class="chapter" data-level="14.2.5" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#genomic-annotations"><i class="fa fa-check"></i><b>14.2.5</b> Genomic annotations</a></li>
<li class="chapter" data-level="14.2.6" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#external-resources"><i class="fa fa-check"></i><b>14.2.6</b> External resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figures-key.html"><a href="figures-key.html"><i class="fa fa-check"></i><b>15</b> Figures key</a><ul>
<li class="chapter" data-level="15.1" data-path="figures-key.html"><a href="figures-key.html#figure-1"><i class="fa fa-check"></i><b>15.1</b> Figure 1</a></li>
<li class="chapter" data-level="15.2" data-path="figures-key.html"><a href="figures-key.html#figure-3"><i class="fa fa-check"></i><b>15.2</b> Figure 3</a></li>
<li class="chapter" data-level="15.3" data-path="figures-key.html"><a href="figures-key.html#figure-4"><i class="fa fa-check"></i><b>15.3</b> Figure 4</a></li>
<li class="chapter" data-level="15.4" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-1"><i class="fa fa-check"></i><b>15.4</b> Extended Data Figure 1</a></li>
<li class="chapter" data-level="15.5" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-2"><i class="fa fa-check"></i><b>15.5</b> Extended Data Figure 2</a></li>
<li class="chapter" data-level="15.6" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-3"><i class="fa fa-check"></i><b>15.6</b> Extended Data Figure 3</a></li>
<li class="chapter" data-level="15.7" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-4"><i class="fa fa-check"></i><b>15.7</b> Extended Data Figure 4</a></li>
<li class="chapter" data-level="15.8" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-5"><i class="fa fa-check"></i><b>15.8</b> Extended Data Figure 5</a></li>
<li class="chapter" data-level="15.9" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-6"><i class="fa fa-check"></i><b>15.9</b> Extended Data Figure 6</a></li>
<li class="chapter" data-level="15.10" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-7"><i class="fa fa-check"></i><b>15.10</b> Extended Data Figure 7</a></li>
<li class="chapter" data-level="15.11" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-8"><i class="fa fa-check"></i><b>15.11</b> Extended Data Figure 8</a></li>
<li class="chapter" data-level="15.12" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-9"><i class="fa fa-check"></i><b>15.12</b> Extended Data Figure 9</a></li>
<li class="chapter" data-level="15.13" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-10"><i class="fa fa-check"></i><b>15.13</b> Extended Data Figure 10</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="definition-of-clock-mg-and-ml-scores" class="section level1">
<h1><span class="header-section-number">4</span> Definition of Clock, MG and ML scores</h1>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">4.1</span> Introduction</h2>
<p>After cleaning the methylation data from TME effects we went on to look for global trends in the methylation data, i.e. trends that affect all or almost all of the methylome, in particular due to replication age and copy number aberration effects.</p>
</div>
<div id="initialize-1" class="section level2">
<h2><span class="header-section-number">4.2</span> Initialize</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;scripts/init.R&quot;</span>))</code></pre></div>
</div>
<div id="load-data-1" class="section level2">
<h2><span class="header-section-number">4.3</span> Load data</h2>
<p>We load both TME-normalized and raw methylation values.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_norm_meth &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/all_norm_meth.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>() </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_mat_raw &lt;-<span class="st"> </span><span class="kw">get_all_meth</span>()  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>()

<span class="kw">head</span>(all_mat_raw)</code></pre></div>
<pre><code>##                        MB_0006    MB_0028    MB_0030    MB_0035    MB_0040
## chr1_762469_763020 0.000000000 0.00000000 0.00000000 0.00000000 0.00000000
##                       MB_0046    MB_0050     MB_0053    MB_0054     MB_0062
## chr1_762469_763020 0.00000000 0.00000000          NA 0.00000000 0.000000000
##                        MB_0064     MB_0068    MB_0112     MB_0113    MB_0114
## chr1_762469_763020 0.021276596 0.000000000 0.01058201 0.000000000         NA
##                        MB_0118     MB_0119    MB_0120    MB_0121     MB_0123
## chr1_762469_763020 0.007067138 0.007751938 0.00000000 0.00000000 0.000000000
##                       MB_0124     MB_0125    MB_0126     MB_0127    MB_0128
## chr1_762469_763020 0.01731602 0.000000000 0.02469136 0.000000000 0.00000000
##                       MB_0129     MB_0130     MB_0131     MB_0133     MB_0134
## chr1_762469_763020 0.00000000 0.011764706 0.012500000 0.000000000 0.027522936
##                       MB_0135    MB_0136     MB_0137     MB_0138    MB_0140
## chr1_762469_763020 0.00000000 0.00000000 0.008403361 0.000000000 0.00000000
##                        MB_0142     MB_0143    MB_0144    MB_0145    MB_0146
## chr1_762469_763020 0.000000000 0.004901961 0.00000000 0.00990099 0.00000000
##                        MB_0147    MB_0148     MB_0149     MB_0150    MB_0151
## chr1_762469_763020 0.026086957 0.00000000 0.000000000 0.000000000 0.00000000
##                        MB_0152     MB_0154     MB_0155     MB_0157    MB_0162
## chr1_762469_763020 0.000000000          NA 0.004032258 0.000000000 0.00000000
##                        MB_0163     MB_0164    MB_0165     MB_0166    MB_0167
## chr1_762469_763020          NA 0.023809524 0.00000000 0.021126761 0.00000000
##                        MB_0168    MB_0169    MB_0170     MB_0171    MB_0172
## chr1_762469_763020 0.005494505 0.03000000 0.00000000 0.046511628 0.00000000
##                        MB_0173     MB_0174     MB_0175    MB_0176     MB_0177
## chr1_762469_763020 0.009569378 0.019607843 0.000000000 0.00000000 0.013513514
##                        MB_0178     MB_0179     MB_0180     MB_0181    MB_0184
## chr1_762469_763020 0.029411765 0.012195122          NA 0.000000000 0.00000000
##                        MB_0188    MB_0189    MB_0191     MB_0192     MB_0193
## chr1_762469_763020 0.000000000 0.00000000         NA 0.000000000 0.000000000
##                       MB_0194     MB_0195     MB_0197    MB_0198     MB_0199
## chr1_762469_763020         NA 0.017094017 0.000000000 0.01886792 0.000000000
##                       MB_0201    MB_0202    MB_0203     MB_0204    MB_0205
## chr1_762469_763020 0.00000000 0.00000000 0.03000000 0.000000000 0.01442308
##                        MB_0206     MB_0207     MB_0214    MB_0215    MB_0218
## chr1_762469_763020 0.000000000 0.021739130 0.000000000 0.00000000 0.00000000
##                       MB_0220     MB_0221    MB_0222     MB_0223    MB_0224
## chr1_762469_763020         NA          NA 0.00000000 0.018691589 0.02197802
##                        MB_0225     MB_0226     MB_0227     MB_0228     MB_0229
## chr1_762469_763020 0.018867925 0.014925373 0.006993007          NA 0.000000000
##                        MB_0231    MB_0233     MB_0234     MB_0235    MB_0236
## chr1_762469_763020          NA 0.00000000 0.000000000 0.004608295 0.00000000
##                       MB_0238    MB_0239     MB_0240    MB_0241    MB_0242
## chr1_762469_763020         NA 0.00000000 0.000000000 0.00000000 0.00000000
##                        MB_0243     MB_0244    MB_0245     MB_0246     MB_0248
## chr1_762469_763020 0.000000000 0.012048193 0.00000000 0.004739336 0.000000000
##                         MB_0249    MB_0250    MB_0251     MB_0252     MB_0253
## chr1_762469_763020 0.0189873418 0.01449275 0.03255814 0.031914894 0.005882353
##                        MB_0254     MB_0255     MB_0256     MB_0257     MB_0258
## chr1_762469_763020 0.000000000 0.006666667 0.000000000 0.004273504 0.013157895
##                        MB_0259    MB_0260     MB_0261     MB_0262     MB_0263
## chr1_762469_763020          NA 0.01675978 0.008474576 0.048780488 0.000000000
##                       MB_0264     MB_0265    MB_0266     MB_0268    MB_0270
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.009345794         NA
##                        MB_0271     MB_0272    MB_0273    MB_0278    MB_0279
## chr1_762469_763020 0.000000000 0.025423729 0.00000000 0.00000000 0.01716738
##                       MB_0280     MB_0282    MB_0283    MB_0284    MB_0285
## chr1_762469_763020 0.00000000 0.006250000 0.00000000 0.00000000 0.04109589
##                        MB_0286     MB_0287     MB_0288     MB_0290     MB_0291
## chr1_762469_763020 0.000000000 0.000000000 0.061538462          NA          NA
##                        MB_0292    MB_0293     MB_0294     MB_0295     MB_0301
## chr1_762469_763020          NA 0.04301075 0.000000000 0.000000000 0.020833333
##                        MB_0302     MB_0303    MB_0304    MB_0305     MB_0306
## chr1_762469_763020 0.006896552          NA 0.00000000 0.00000000 0.000000000
##                       MB_0307     MB_0308     MB_0309     MB_0310     MB_0311
## chr1_762469_763020 0.00000000 0.012738854          NA 0.012626263 0.000000000
##                       MB_0312     MB_0313     MB_0314    MB_0315     MB_0316
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.00000000 0.000000000
##                       MB_0317    MB_0318    MB_0319     MB_0320     MB_0321
## chr1_762469_763020 0.00000000 0.00000000 0.01190476 0.000000000 0.010752688
##                       MB_0322     MB_0324     MB_0325     MB_0327    MB_0328
## chr1_762469_763020 0.01507538 0.003086420 0.009615385 0.000000000 0.01869159
##                        MB_0329    MB_0330     MB_0336     MB_0339     MB_0340
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.006451613 0.000000000
##                       MB_0341     MB_0342     MB_0343    MB_0345    MB_0348
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.00000000 0.03030303
##                        MB_0349    MB_0350     MB_0352    MB_0353    MB_0356
## chr1_762469_763020 0.000000000 0.00000000 0.000000000         NA 0.05660377
##                       MB_0358     MB_0359    MB_0360    MB_0361    MB_0362
## chr1_762469_763020 0.00000000 0.013793103 0.00000000         NA 0.00000000
##                       MB_0363    MB_0364     MB_0365    MB_0366     MB_0367
## chr1_762469_763020 0.00000000 0.00000000 0.008333333 0.00000000 0.028985507
##                       MB_0368     MB_0369    MB_0370     MB_0371     MB_0373
## chr1_762469_763020 0.01239669 0.020000000 0.00000000 0.003533569 0.000000000
##                        MB_0374    MB_0375     MB_0378     MB_0379    MB_0380
## chr1_762469_763020 0.000000000 0.02380952 0.000000000 0.000000000 0.00000000
##                       MB_0381     MB_0383     MB_0384     MB_0385     MB_0386
## chr1_762469_763020 0.01379310 0.008064516 0.006329114 0.000000000 0.000000000
##                       MB_0388    MB_0389     MB_0391    MB_0392    MB_0393
## chr1_762469_763020 0.00000000 0.00000000 0.000000000 0.00000000 0.00000000
##                       MB_0394     MB_0395     MB_0396    MB_0398    MB_0399
## chr1_762469_763020 0.03015075 0.000000000 0.000000000 0.00000000 0.00000000
##                       MB_0400    MB_0401    MB_0402     MB_0404    MB_0406
## chr1_762469_763020 0.00000000 0.00000000 0.01550388 0.000000000 0.00660066
##                       MB_0408    MB_0410     MB_0411     MB_0412     MB_0413
## chr1_762469_763020 0.02545455 0.00000000 0.021276596 0.000000000 0.016949153
##                        MB_0414     MB_0417     MB_0418     MB_0419     MB_0420
## chr1_762469_763020 0.000000000 0.000000000 0.006802721 0.008695652 0.011560694
##                        MB_0421     MB_0422     MB_0425     MB_0426     MB_0427
## chr1_762469_763020 0.008474576          NA 0.000000000 0.014925373 0.008196721
##                       MB_0428     MB_0429    MB_0431     MB_0432     MB_0434
## chr1_762469_763020 0.00000000 0.033333333 0.00000000 0.000000000 0.000000000
##                       MB_0436    MB_0437     MB_0439    MB_0440     MB_0442
## chr1_762469_763020 0.00462963 0.00000000 0.006410256 0.01724138 0.008888889
##                        MB_0443     MB_0444    MB_0445    MB_0446     MB_0449
## chr1_762469_763020 0.008474576 0.004098361 0.03465347 0.04098361 0.008403361
##                        MB_0451     MB_0452     MB_0453     MB_0454     MB_0455
## chr1_762469_763020 0.000000000 0.015267176 0.010928962 0.000000000 0.000000000
##                       MB_0458    MB_0460    MB_0462     MB_0464    MB_0465
## chr1_762469_763020 0.00000000 0.00000000 0.00000000 0.016666667 0.00000000
##                        MB_0466    MB_0467     MB_0468     MB_0469    MB_0471
## chr1_762469_763020 0.012500000 0.01250000 0.000000000 0.000000000 0.00000000
##                       MB_0474    MB_0475     MB_0476     MB_0478    MB_0479
## chr1_762469_763020 0.02597403         NA 0.000000000 0.012422360 0.00000000
##                        MB_0480     MB_0481     MB_0482    MB_0483     MB_0484
## chr1_762469_763020 0.000000000 0.000000000          NA         NA 0.000000000
##                        MB_0485    MB_0486     MB_0487     MB_0488     MB_0489
## chr1_762469_763020 0.008333333         NA 0.003937008          NA 0.000000000
##                        MB_0492    MB_0496     MB_0497    MB_0500    MB_0501
## chr1_762469_763020 0.057142857 0.00000000 0.000000000 0.00000000 0.01522843
##                        MB_0502     MB_0503    MB_0504     MB_0505     MB_0506
## chr1_762469_763020 0.003558719 0.000000000 0.00000000 0.012048193 0.000000000
##                        MB_0507    MB_0508    MB_0510    MB_0511     MB_0512
## chr1_762469_763020 0.012269939 0.00000000 0.02884615         NA 0.000000000
##                        MB_0513     MB_0514     MB_0516     MB_0519    MB_0521
## chr1_762469_763020 0.000000000 0.000000000 0.027027027 0.010169492 0.00000000
##                        MB_0525    MB_0527    MB_0528    MB_0529     MB_0530
## chr1_762469_763020 0.008695652 0.00000000 0.00000000 0.00000000 0.006756757
##                        MB_0531     MB_0532     MB_0534     MB_0535     MB_0536
## chr1_762469_763020 0.000000000 0.009009009 0.044444444 0.000000000 0.009345794
##                        MB_0538     MB_0542    MB_0544    MB_0545     MB_0546
## chr1_762469_763020 0.000000000 0.022388060 0.01219512 0.00000000 0.000000000
##                        MB_0549     MB_0550    MB_0551     MB_0552     MB_0553
## chr1_762469_763020 0.000000000 0.008474576 0.03278689 0.000000000 0.009174312
##                        MB_0558    MB_0559     MB_0566     MB_0568     MB_0569
## chr1_762469_763020 0.014925373 0.00000000 0.000000000 0.000000000 0.015151515
##                        MB_0570     MB_0571     MB_0572     MB_0574     MB_0575
## chr1_762469_763020 0.012500000 0.021276596 0.000000000 0.017241379 0.000000000
##                       MB_0576     MB_0580     MB_0581    MB_0582    MB_0583
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.00000000 0.00000000
##                       MB_0585    MB_0589     MB_0591    MB_0593    MB_0594
## chr1_762469_763020 0.02739726 0.00000000 0.000000000 0.00000000 0.00000000
##                        MB_0596     MB_0597     MB_0600     MB_0601     MB_0603
## chr1_762469_763020 0.000000000 0.000000000 0.005681818 0.000000000 0.000000000
##                        MB_0605    MB_0606     MB_0608    MB_0609     MB_0610
## chr1_762469_763020 0.004132231 0.00000000          NA 0.00000000 0.014598540
##                        MB_0615     MB_0616    MB_0617     MB_0618     MB_0620
## chr1_762469_763020 0.000000000 0.007692308 0.00000000 0.000000000 0.003968254
##                        MB_0621     MB_0622     MB_0623    MB_0626     MB_0627
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.00000000 0.000000000
##                       MB_0628    MB_0631     MB_0632     MB_0637    MB_0638
## chr1_762469_763020 0.01851852 0.03260870 0.007692308 0.021978022 0.00000000
##                        MB_0639    MB_0642     MB_0643    MB_0646     MB_0649
## chr1_762469_763020 0.051282051 0.00000000 0.000000000 0.00000000 0.000000000
##                       MB_0650    MB_0653     MB_0655    MB_0656    MB_0657
## chr1_762469_763020 0.00000000 0.00000000 0.000000000 0.00000000 0.00000000
##                       MB_0659     MB_0662     MB_0663    MB_0664     MB_0666
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.00000000 0.000000000
##                       MB_0668     MB_0676     MB_0684     MB_0685     MB_0692
## chr1_762469_763020         NA 0.000000000          NA 0.004166667 0.000000000
##                        MB_0698     MB_0700    MB_0719     MB_0724     MB_0732
## chr1_762469_763020 0.003703704 0.000000000 0.00000000 0.000000000 0.000000000
##                       MB_0735    MB_0739     MB_0748      MB_0756     MB_0767
## chr1_762469_763020 0.01234568 0.00000000 0.000000000 0.0000000000 0.000000000
##                       MB_0768      MB_0770     MB_0772     MB_0778     MB_0779
## chr1_762469_763020 0.01538462 0.0000000000 0.000000000 0.000000000 0.000000000
##                        MB_0785     MB_0786     MB_0788     MB_0794      MB_0795
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.000000000 0.0000000000
##                         MB_0801     MB_0803     MB_0818     MB_0825     MB_0826
## chr1_762469_763020 0.0238095238 0.000000000 0.000000000 0.000000000 0.008695652
##                        MB_0828     MB_0833     MB_0835     MB_0841      MB_0850
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.000000000 0.0000000000
##                        MB_0857     MB_0865     MB_0868     MB_0872     MB_0873
## chr1_762469_763020 0.000000000 0.000000000 0.011173184 0.009259259 0.013245033
##                        MB_0875      MB_0877    MB_0878     MB_0879    MB_0880
## chr1_762469_763020 0.004484305 0.0000000000 0.02932551 0.008635579 0.00000000
##                        MB_0881    MB_0884     MB_0886     MB_0890    MB_0892
## chr1_762469_763020          NA 0.02083333 0.000000000 0.031250000 0.01562500
##                        MB_0894     MB_0895     MB_0897     MB_0898     MB_0899
## chr1_762469_763020 0.029569892          NA 0.000000000 0.024096386 0.009708738
##                        MB_0901     MB_0902    MB_0904     MB_0905     MB_0906
## chr1_762469_763020 0.000000000 0.013698630 0.00000000 0.000000000 0.000000000
##                        MB_0908     MB_0923     MB_0924    MB_0925     MB_0927
## chr1_762469_763020 0.006944444 0.015544041 0.019607843 0.00000000 0.000000000
##                       MB_0928     MB_0929     MB_0930     MB_0931    MB_0932
## chr1_762469_763020 0.03149606 0.006944444 0.040000000 0.027397260 0.00000000
##                       MB_0933     MB_0934    MB_0935     MB_0936     MB_0937
## chr1_762469_763020 0.00000000 0.000000000 0.00990099 0.007462687 0.000000000
##                        MB_0939     MB_0941     MB_0942     MB_0943    MB_0944
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.000000000         NA
##                       MB_0945    MB_0947     MB_0948    MB_0949     MB_0950
## chr1_762469_763020 0.01463415 0.02564103 0.000000000 0.00000000 0.020000000
##                        MB_0951     MB_0952    MB_0953     MB_0955     MB_0956
## chr1_762469_763020 0.003649635 0.000000000 0.02020202 0.000000000 0.000000000
##                       MB_0957     MB_0958     MB_0961    MB_0964     MB_0965
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.00000000          NA
##                        MB_0966     MB_0967     MB_0968     MB_0969    MB_0970
## chr1_762469_763020 0.040462428 0.022471910 0.004807692 0.000000000 0.00000000
##                        MB_0971    MB_0973    MB_0974     MB_0975     MB_0976
## chr1_762469_763020 0.033613445 0.00000000 0.00000000 0.000000000 0.000000000
##                        MB_0978     MB_0979     MB_0980     MB_0982     MB_0983
## chr1_762469_763020 0.000000000 0.000000000 0.013986014          NA 0.000000000
##                        MB_0984     MB_0986     MB_0987    MB_0988     MB_0989
## chr1_762469_763020 0.000000000 0.008196721 0.000000000 0.00000000 0.000000000
##                        MB_0990      MB_0991    MB_0992     MB_0993     MB_0994
## chr1_762469_763020 0.002570694 0.0046728972 0.00000000 0.000000000 0.007604563
##                        MB_0995     MB_0996     MB_0997     MB_0998     MB_0999
## chr1_762469_763020 0.017937220 0.000000000 0.007751938 0.013986014 0.004201681
##                        MB_1001    MB_1002     MB_1003     MB_1004     MB_1005
## chr1_762469_763020 0.000000000 0.01190476 0.000000000 0.011560694 0.000000000
##                        MB_1006     MB_1007     MB_1008     MB_1009     MB_1010
## chr1_762469_763020 0.005494505 0.006250000 0.000000000 0.000000000 0.000000000
##                        MB_1011     MB_1012     MB_1013    MB_1014     MB_1015
## chr1_762469_763020 0.000000000 0.004032258 0.014388489 0.00000000 0.020270270
##                       MB_1017     MB_1018     MB_1020     MB_1022     MB_1023
## chr1_762469_763020 0.01923077 0.005025126 0.011560694 0.005899705 0.006666667
##                        MB_1024     MB_1025     MB_1026     MB_1027     MB_1028
## chr1_762469_763020 0.006493506 0.000000000 0.000000000 0.042253521 0.014925373
##                        MB_1029    MB_1031     MB_1032    MB_1033    MB_1034
## chr1_762469_763020 0.014084507 0.00000000 0.000000000 0.01162791 0.00000000
##                        MB_1035     MB_1036    MB_1037    MB_1038     MB_1039
## chr1_762469_763020 0.025641026 0.000000000 0.01333333 0.00000000 0.028169014
##                       MB_1040     MB_1041    MB_1043    MB_1044     MB_1045
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.04545455 0.000000000
##                        MB_1046    MB_1047     MB_1048     MB_1049     MB_1050
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.000000000 0.000000000
##                       MB_1051     MB_1052     MB_1053     MB_1054    MB_1055
## chr1_762469_763020 0.00000000 0.000000000 0.009900990 0.010638298         NA
##                        MB_1056     MB_1057     MB_1058     MB_1059     MB_1061
## chr1_762469_763020 0.000000000 0.000000000 0.016666667 0.005847953 0.000000000
##                       MB_1062     MB_1063     MB_1064     MB_1065     MB_1066
## chr1_762469_763020 0.03125000 0.008695652 0.006172840 0.008196721 0.008264463
##                        MB_1067    MB_1068     MB_1069    MB_1070     MB_1071
## chr1_762469_763020 0.012500000 0.00000000 0.009174312 0.01923077 0.000000000
##                        MB_1072    MB_1073     MB_1074     MB_1075     MB_1076
## chr1_762469_763020 0.000000000 0.01142857 0.006896552 0.016528926 0.000000000
##                       MB_1077    MB_1078    MB_1079    MB_1080    MB_1081
## chr1_762469_763020 0.00000000 0.01176471 0.06451613 0.00000000 0.01818182
##                        MB_1082    MB_1083     MB_1084     MB_1085     MB_1086
## chr1_762469_763020 0.000000000 0.01123596 0.012345679 0.016949153 0.006329114
##                        MB_1087    MB_1090     MB_1091     MB_1093     MB_1096
## chr1_762469_763020 0.000000000         NA 0.000000000 0.000000000 0.020408163
##                       MB_1097    MB_1098     MB_1100     MB_1101     MB_1102
## chr1_762469_763020 0.02941176 0.00000000 0.006172840 0.000000000 0.000000000
##                        MB_1103     MB_1104     MB_1105    MB_1107     MB_1110
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.00000000 0.000000000
##                        MB_1111     MB_1114     MB_1116     MB_1118    MB_1119
## chr1_762469_763020 0.006410256 0.006493506 0.000000000 0.000000000 0.01086957
##                        MB_1120     MB_1121     MB_1122    MB_1123    MB_1124
## chr1_762469_763020 0.004950495 0.005235602 0.000000000 0.01224490 0.04950495
##                        MB_1125     MB_1126     MB_1127     MB_1129     MB_1130
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.008163265 0.000000000
##                        MB_1131     MB_1132     MB_1134     MB_1135     MB_1136
## chr1_762469_763020 0.000000000 0.000000000 0.010025063 0.010810811 0.000000000
##                       MB_1137     MB_1138     MB_1139     MB_1141    MB_1142
## chr1_762469_763020 0.00000000 0.003174603 0.004694836 0.007712082 0.00000000
##                        MB_1143     MB_1144     MB_1145    MB_1146     MB_1147
## chr1_762469_763020 0.004484305 0.009836066 0.008771930 0.01307190 0.009966777
##                       MB_1148     MB_1149     MB_1151     MB_1152    MB_1155
## chr1_762469_763020 0.02479339 0.007092199 0.004901961 0.011235955 0.01877934
##                        MB_1156     MB_2517     MB_2536    MB_2556     MB_2564
## chr1_762469_763020 0.000000000 0.000000000 0.005263158 0.00000000 0.011976048
##                        MB_2610     MB_2613     MB_2614      MB_2616    MB_2617
## chr1_762469_763020 0.023437500 0.000000000 0.023809524 0.0000000000 0.00000000
##                        MB_2618     MB_2624    MB_2626    MB_2632     MB_2634
## chr1_762469_763020 0.000000000 0.000000000 0.00000000 0.05263158 0.000000000
##                       MB_2640     MB_2643    MB_2645    MB_2669     MB_2686
## chr1_762469_763020 0.00000000 0.016528926         NA 0.00000000 0.000000000
##                       MB_2705     MB_2708     MB_2711     MB_2712     MB_2718
## chr1_762469_763020         NA 0.015873016 0.019607843 0.008620690 0.005181347
##                       MB_2724     MB_2728     MB_2730     MB_2742     MB_2744
## chr1_762469_763020 0.02255639 0.000000000 0.000000000          NA 0.000000000
##                        MB_2745     MB_2747    MB_2749     MB_2750    MB_2752
## chr1_762469_763020 0.000000000 0.016042781 0.02439024 0.009345794 0.02777778
##                        MB_2753     MB_2754     MB_2760     MB_2763    MB_2764
## chr1_762469_763020 0.026595745 0.018656716 0.000000000 0.009090909 0.00000000
##                       MB_2765    MB_2767     MB_2769    MB_2770     MB_2772
## chr1_762469_763020 0.00000000         NA 0.000000000 0.04477612 0.018691589
##                        MB_2774     MB_2775     MB_2778    MB_2779      MB_2781
## chr1_762469_763020 0.011235955 0.008333333 0.000000000 0.03448276 0.0288461538
##                        MB_2790     MB_2791     MB_2792     MB_2793    MB_2796
## chr1_762469_763020 0.000000000 0.028571429 0.000000000 0.007246377 0.00000000
##                        MB_2803    MB_2814    MB_2819     MB_2820     MB_2821
## chr1_762469_763020 0.000000000 0.00000000 0.00000000 0.010101010 0.028571429
##                       MB_2827    MB_2833    MB_2834     MB_2835     MB_2838
## chr1_762469_763020 0.01923077 0.02884615 0.00617284 0.011235955 0.000000000
##                       MB_2840     MB_2842    MB_2843    MB_2844    MB_2845
## chr1_762469_763020 0.00000000 0.000000000 0.01986755 0.00000000 0.00000000
##                        MB_2846     MB_2847     MB_2848     MB_2849     MB_2850
## chr1_762469_763020 0.000000000 0.000000000 0.006993007 0.000000000 0.000000000
##                        MB_2851    MB_2853     MB_2854     MB_2862     MB_2895
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.008264463 0.000000000
##                       MB_2896    MB_2901    MB_2912     MB_2916     MB_2917
## chr1_762469_763020 0.00000000 0.00000000 0.00000000 0.000000000 0.019047619
##                       MB_2919    MB_2923     MB_2927     MB_2929     MB_2931
## chr1_762469_763020 0.02898551 0.00000000          NA 0.008264463 0.000000000
##                        MB_2933    MB_2939    MB_2951    MB_2952     MB_2953
## chr1_762469_763020 0.036231884 0.00000000 0.02027027 0.00000000 0.003846154
##                       MB_2954     MB_2957     MB_2960     MB_2963     MB_2969
## chr1_762469_763020 0.01488095 0.021052632 0.005347594 0.013245033 0.000000000
##                        MB_2977    MB_2983    MB_2984     MB_2990     MB_2993
## chr1_762469_763020 0.010791367 0.00000000 0.01910828 0.000000000 0.000000000
##                       MB_2994    MB_2996     MB_3001     MB_3005     MB_3006
## chr1_762469_763020 0.00000000 0.00000000          NA 0.007692308          NA
##                        MB_3007     MB_3008    MB_3014    MB_3016     MB_3021
## chr1_762469_763020 0.000000000 0.010309278 0.00000000         NA 0.000000000
##                       MB_3025     MB_3028     MB_3031     MB_3033     MB_3035
## chr1_762469_763020 0.00000000          NA 0.010101010 0.014423077 0.015748031
##                       MB_3037     MB_3046    MB_3049    MB_3050     MB_3057
## chr1_762469_763020 0.00000000 0.011764706 0.00000000         NA 0.000000000
##                        MB_3058     MB_3063     MB_3064     MB_3067    MB_3079
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.031250000 0.01219512
##                        MB_3082     MB_3083     MB_3085    MB_3088     MB_3089
## chr1_762469_763020 0.008130081 0.018181818 0.007142857 0.01369863 0.000000000
##                        MB_3102     MB_3104     MB_3105    MB_3110    MB_3121
## chr1_762469_763020 0.000000000 0.013157895 0.000000000 0.00000000 0.01754386
##                       MB_3122     MB_3153     MB_3165     MB_3167     MB_3171
## chr1_762469_763020 0.00000000 0.000000000 0.011764706 0.031847134 0.010638298
##                        MB_3211    MB_3218     MB_3228    MB_3235    MB_3252
## chr1_762469_763020 0.004672897         NA 0.000000000 0.00000000 0.00000000
##                        MB_3253     MB_3254     MB_3266     MB_3271     MB_3275
## chr1_762469_763020 0.027027027 0.000000000 0.006622517 0.000000000 0.012096774
##                        MB_3277     MB_3292     MB_3295     MB_3297     MB_3298
## chr1_762469_763020 0.010638298 0.009259259 0.000000000 0.040983607 0.015748031
##                       MB_3300    MB_3301    MB_3315     MB_3328     MB_3329
## chr1_762469_763020 0.02941176 0.00000000 0.02500000 0.000000000          NA
##                       MB_3341     MB_3350    MB_3351     MB_3357     MB_3360
## chr1_762469_763020 0.00000000 0.009615385 0.00000000          NA 0.021052632
##                        MB_3361    MB_3367    MB_3371     MB_3378     MB_3382
## chr1_762469_763020          NA 0.00000000 0.01818182 0.000000000 0.000000000
##                        MB_3383     MB_3395     MB_3396    MB_3402     MB_3403
## chr1_762469_763020 0.000000000 0.000000000 0.019417476 0.00000000 0.000000000
##                        MB_3412     MB_3417     MB_3429     MB_3430    MB_3435
## chr1_762469_763020 0.038251366 0.044692737 0.012422360 0.005617978 0.02040816
##                       MB_3436     MB_3437    MB_3439    MB_3452     MB_3453
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.02727273 0.000000000
##                       MB_3459     MB_3462    MB_3467     MB_3470     MB_3476
## chr1_762469_763020 0.01863354 0.000000000 0.00000000 0.008264463 0.004950495
##                       MB_3479     MB_3490     MB_3492     MB_3497    MB_3500
## chr1_762469_763020 0.01492537 0.012738854 0.000000000 0.000000000 0.00000000
##                       MB_3502     MB_3506     MB_3510     MB_3526    MB_3545
## chr1_762469_763020 0.00000000 0.000000000 0.000000000 0.000000000         NA
##                       MB_3548     MB_3555    MB_3556     MB_3560     MB_3576
## chr1_762469_763020         NA 0.000000000 0.00000000          NA 0.006622517
##                       MB_3582     MB_3600    MB_3614     MB_3676     MB_3702
## chr1_762469_763020 0.00000000 0.004587156 0.00000000 0.000000000 0.005586592
##                        MB_3706     MB_3707     MB_3711    MB_3747    MB_3748
## chr1_762469_763020 0.007194245 0.000000000          NA 0.00000000 0.03750000
##                       MB_3752   MB_3754     MB_3755     MB_3781     MB_3804
## chr1_762469_763020         NA 0.0000000 0.000000000 0.019704433 0.019138756
##                       MB_3823     MB_3824    MB_3838    MB_3840     MB_3842
## chr1_762469_763020 0.00000000 0.022727273 0.02290076 0.00000000 0.034482759
##                        MB_3850    MB_3852     MB_3854     MB_3865    MB_3866
## chr1_762469_763020 0.000000000         NA 0.000000000 0.000000000 0.00000000
##                        MB_3871    MB_3874    MB_3978    MB_4000    MB_4001
## chr1_762469_763020 0.000000000 0.00000000 0.00000000 0.00000000 0.00000000
##                        MB_4003     MB_4004    MB_4008    MB_4012     MB_4015
## chr1_762469_763020 0.009009009 0.000000000 0.00000000 0.00000000 0.006024096
##                       MB_4017     MB_4018    MB_4024    MB_4033     MB_4036
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.00000000 0.013574661
##                       MB_4038     MB_4039    MB_4040     MB_4042     MB_4046
## chr1_762469_763020 0.02117647 0.000000000 0.00000000 0.004115226 0.000000000
##                       MB_4048     MB_4049    MB_4051     MB_4053     MB_4054
## chr1_762469_763020 0.04242424 0.010309278 0.01030928 0.009132420 0.012738854
##                        MB_4055     MB_4056     MB_4057    MB_4058     MB_4059
## chr1_762469_763020 0.012987013 0.058394161 0.000000000 0.02803738 0.000000000
##                       MB_4060    MB_4061     MB_4062    MB_4063     MB_4064
## chr1_762469_763020 0.03007519 0.00000000 0.000000000 0.01612903 0.008522727
##                         MB_4066     MB_4072     MB_4079    MB_4083     MB_4084
## chr1_762469_763020 0.0546448087 0.006711409 0.000000000 0.01298701 0.012875536
##                        MB_4091     MB_4098    MB_4110     MB_4117    MB_4119
## chr1_762469_763020 0.005555556 0.013333333 0.02702703 0.022988506 0.01923077
##                        MB_4120     MB_4123     MB_4126    MB_4127    MB_4139
## chr1_762469_763020 0.005681818 0.000000000 0.000000000 0.00000000 0.00000000
##                       MB_4141    MB_4145     MB_4146    MB_4148     MB_4154
## chr1_762469_763020 0.00000000 0.01282051 0.000000000 0.02083333 0.000000000
##                        MB_4169     MB_4171     MB_4173     MB_4189     MB_4190
## chr1_762469_763020 0.007662835 0.004081633 0.010362694 0.024390244 0.000000000
##                       MB_4211    MB_4212    MB_4213     MB_4222     MB_4224
## chr1_762469_763020 0.00990099 0.02380952 0.00000000 0.004761905 0.000000000
##                        MB_4230     MB_4233    MB_4234     MB_4235     MB_4236
## chr1_762469_763020 0.004347826 0.016000000 0.01086957 0.009868421 0.000000000
##                        MB_4254    MB_4255     MB_4264     MB_4266     MB_4268
## chr1_762469_763020 0.005128205 0.03448276 0.012500000 0.000000000 0.000000000
##                       MB_4270     MB_4272    MB_4274     MB_4276    MB_4278
## chr1_762469_763020 0.01265823 0.000000000 0.00000000 0.005780347 0.01481481
##                       MB_4281     MB_4282     MB_4283    MB_4289     MB_4293
## chr1_762469_763020 0.00000000 0.026455026 0.005102041 0.03333333 0.011834320
##                       MB_4300     MB_4306    MB_4308     MB_4310    MB_4313
## chr1_762469_763020 0.01142857 0.000000000         NA 0.030000000 0.00000000
##                       MB_4317    MB_4318     MB_4322    MB_4323    MB_4328
## chr1_762469_763020 0.00000000         NA 0.000000000 0.00000000 0.00000000
##                       MB_4331     MB_4333    MB_4339     MB_4341     MB_4342
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.000000000 0.000000000
##                        MB_4343    MB_4348    MB_4350    MB_4351     MB_4353
## chr1_762469_763020          NA 0.00000000 0.01492537 0.03030303 0.009009009
##                        MB_4354     MB_4357    MB_4360    MB_4361     MB_4362
## chr1_762469_763020          NA 0.000000000 0.00000000 0.00000000 0.019354839
##                        MB_4368    MB_4374     MB_4375    MB_4381    MB_4390
## chr1_762469_763020 0.012048193         NA 0.000000000 0.00000000 0.01219512
##                        MB_4395    MB_4407     MB_4408    MB_4409    MB_4410
## chr1_762469_763020 0.016393443 0.00000000 0.000000000 0.00000000 0.01785714
##                        MB_4413    MB_4416     MB_4417    MB_4421    MB_4426
## chr1_762469_763020 0.004587156 0.01578947 0.000000000 0.00000000 0.03680982
##                       MB_4431     MB_4434     MB_4442    MB_4444   MB_4455
## chr1_762469_763020         NA 0.013698630 0.008823529 0.00000000 0.0000000
##                        MB_4458     MB_4464     MB_4465    MB_4474    MB_4477
## chr1_762469_763020 0.000000000 0.009852217 0.026315789 0.00000000 0.00000000
##                        MB_4482     MB_4483     MB_4484     MB_4487     MB_4492
## chr1_762469_763020 0.005390836 0.000000000 0.008130081 0.003344482 0.000000000
##                        MB_4495    MB_4511     MB_4521    MB_4528     MB_4529
## chr1_762469_763020 0.008888889         NA 0.000000000 0.00000000 0.005681818
##                       MB_4531    MB_4541     MB_4548    MB_4556     MB_4557
## chr1_762469_763020 0.00000000 0.00000000 0.011235955 0.01851852 0.000000000
##                        MB_4558     MB_4564     MB_4578     MB_4591    MB_4593
## chr1_762469_763020 0.013636364 0.052631579 0.000000000 0.005154639 0.01428571
##                        MB_4598     MB_4599     MB_4601     MB_4602     MB_4603
## chr1_762469_763020 0.024509804 0.003236246 0.008230453 0.000000000 0.000000000
##                       MB_4607     MB_4613     MB_4616     MB_4618     MB_4621
## chr1_762469_763020 0.01187648 0.000000000 0.002624672 0.000000000 0.000000000
##                        MB_4622    MB_4623    MB_4626     MB_4627     MB_4630
## chr1_762469_763020 0.029411765 0.00000000 0.03937008 0.010526316 0.025641026
##                        MB_4633     MB_4634     MB_4635    MB_4639     MB_4640
## chr1_762469_763020 0.008230453 0.000000000 0.000000000 0.00000000 0.007407407
##                        MB_4641     MB_4642     MB_4643    MB_4644    MB_4648
## chr1_762469_763020 0.004149378 0.000000000 0.015772871 0.00000000 0.00000000
##                       MB_4649    MB_4651     MB_4653    MB_4654    MB_4655
## chr1_762469_763020 0.00000000 0.01562500 0.008298755 0.01777778 0.01600000
##                        MB_4661     MB_4663    MB_4665    MB_4666     MB_4669
## chr1_762469_763020 0.005882353 0.000000000 0.01481481 0.01388889 0.017167382
##                        MB_4670     MB_4671    MB_4672    MB_4673     MB_4674
## chr1_762469_763020 0.000000000 0.007843137 0.00000000 0.01212121 0.003115265
##                       MB_4675    MB_4679     MB_4681    MB_4682    MB_4685
## chr1_762469_763020 0.01886792 0.00877193 0.000000000 0.00000000 0.03597122
##                        MB_4686    MB_4688     MB_4691     MB_4692     MB_4694
## chr1_762469_763020 0.009049774 0.02197802 0.005076142 0.012195122 0.000000000
##                       MB_4696    MB_4697    MB_4698    MB_4701     MB_4702
## chr1_762469_763020 0.00000000 0.00000000 0.00000000 0.01626016 0.021126761
##                        MB_4704     MB_4705    MB_4706    MB_4708     MB_4709
## chr1_762469_763020 0.000000000 0.010638298 0.01416431 0.02976190 0.010452962
##                        MB_4710     MB_4711     MB_4712     MB_4714     MB_4715
## chr1_762469_763020 0.010101010 0.014285714 0.008620690 0.012000000 0.028248588
##                       MB_4716     MB_4717    MB_4719     MB_4722     MB_4723
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.008333333 0.016129032
##                        MB_4724    MB_4725     MB_4729     MB_4730     MB_4731
## chr1_762469_763020 0.005847953 0.02127660 0.013245033 0.008403361 0.000000000
##                        MB_4732     MB_4733     MB_4735     MB_4737    MB_4738
## chr1_762469_763020 0.018181818 0.005376344 0.023323615 0.011111111 0.02631579
##                        MB_4739     MB_4741     MB_4742     MB_4743     MB_4744
## chr1_762469_763020 0.010101010 0.009433962 0.010989011 0.014285714 0.000000000
##                        MB_4745     MB_4746    MB_4749     MB_4752     MB_4757
## chr1_762469_763020 0.004310345 0.000000000 0.00000000 0.000000000 0.005617978
##                        MB_4758     MB_4762     MB_4763     MB_4764    MB_4767
## chr1_762469_763020 0.000000000 0.015625000 0.008583691 0.037234043 0.00000000
##                       MB_4769     MB_4770     MB_4771    MB_4778     MB_4779
## chr1_762469_763020 0.02482270 0.003344482 0.013986014 0.00000000 0.006024096
##                        MB_4782     MB_4784    MB_4785     MB_4787     MB_4790
## chr1_762469_763020 0.007092199 0.008264463 0.00000000 0.020547945 0.004830918
##                       MB_4791     MB_4793    MB_4794    MB_4796     MB_4797
## chr1_762469_763020 0.02000000 0.000000000 0.02739726 0.01030928 0.000000000
##                        MB_4798     MB_4800    MB_4801     MB_4802     MB_4805
## chr1_762469_763020 0.010000000 0.000000000 0.02536232 0.007092199 0.029411765
##                        MB_4806    MB_4809     MB_4814     MB_4816     MB_4818
## chr1_762469_763020 0.000000000 0.02884615 0.006289308 0.020833333 0.011904762
##                       MB_4820    MB_4825     MB_4827     MB_4828    MB_4829
## chr1_762469_763020 0.01287554 0.00000000 0.003546099 0.008849558 0.00000000
##                        MB_4832     MB_4834     MB_4836     MB_4838    MB_4839
## chr1_762469_763020 0.037974684 0.000000000 0.008849558 0.011764706 0.00000000
##                        MB_4843    MB_4849    MB_4851     MB_4853     MB_4855
## chr1_762469_763020 0.011173184 0.01980198 0.00000000 0.009852217 0.016666667
##                        MB_4858     MB_4859     MB_4862     MB_4865     MB_4866
## chr1_762469_763020 0.008695652 0.004329004 0.028571429 0.010989011 0.010869565
##                       MB_4867    MB_4869     MB_4871     MB_4872     MB_4873
## chr1_762469_763020 0.01067616 0.00862069 0.007125891 0.000000000 0.006514658
##                        MB_4876    MB_4878    MB_4879    MB_4880     MB_4881
## chr1_762469_763020 0.021126761 0.01764706 0.05769231 0.02702703 0.000000000
##                        MB_4883    MB_4886     MB_4887     MB_4888    MB_4893
## chr1_762469_763020 0.012307692 0.01149425 0.007662835 0.011494253 0.00800000
##                        MB_4894     MB_4896     MB_4897    MB_4898     MB_4899
## chr1_762469_763020 0.009950249 0.000000000 0.007692308 0.04705882 0.000000000
##                        MB_4900    MB_4904     MB_4907    MB_4908     MB_4911
## chr1_762469_763020 0.013513514 0.01648352 0.006993007 0.00000000 0.000000000
##                       MB_4912    MB_4925    MB_4928     MB_4929    MB_4930
## chr1_762469_763020 0.00000000 0.01197605 0.00000000 0.006024096 0.02439024
##                         MB_4931    MB_4933     MB_4934    MB_4935    MB_4937
## chr1_762469_763020 0.0267857143 0.01081081 0.008928571 0.02097902 0.01840491
##                        MB_4941    MB_4942     MB_4944     MB_4945     MB_4949
## chr1_762469_763020 0.010752688 0.02116402 0.007812500 0.003333333 0.005882353
##                       MB_4950     MB_4952     MB_4956    MB_4957    MB_4959
## chr1_762469_763020 0.00000000 0.013333333 0.000000000 0.01333333 0.00000000
##                        MB_4961     MB_4962     MB_4965     MB_4966     MB_4967
## chr1_762469_763020 0.005405405 0.000000000 0.009852217 0.009756098 0.036144578
##                        MB_4968      MB_4969     MB_4970    MB_4976    MB_4977
## chr1_762469_763020 0.000000000 0.0058823529 0.025000000 0.00000000 0.00000000
##                       MB_4981     MB_4986     MB_4987     MB_4991     MB_4992
## chr1_762469_763020 0.01190476 0.012048193 0.008849558 0.000000000 0.000000000
##                        MB_4993     MB_4994     MB_4996     MB_4998     MB_4999
## chr1_762469_763020 0.011560694 0.012396694 0.009009009 0.005000000 0.006289308
##                        MB_5001    MB_5004     MB_5008     MB_5011    MB_5013
## chr1_762469_763020 0.012987013 0.02272727 0.000000000 0.000000000 0.00000000
##                        MB_5014    MB_5015     MB_5018     MB_5019    MB_5022
## chr1_762469_763020 0.004854369 0.00000000 0.014285714 0.000000000 0.00000000
##                        MB_5027     MB_5028     MB_5033     MB_5039     MB_5040
## chr1_762469_763020 0.002832861 0.000000000 0.016000000 0.000000000 0.011764706
##                        MB_5041    MB_5043     MB_5044    MB_5045    MB_5049
## chr1_762469_763020 0.005376344 0.00000000 0.000000000 0.00000000 0.00000000
##                        MB_5050    MB_5052    MB_5053     MB_5054    MB_5057
## chr1_762469_763020 0.014035088 0.02395210 0.00000000 0.013986014 0.01098901
##                        MB_5059    MB_5060    MB_5061     MB_5062     MB_5063
## chr1_762469_763020 0.000000000 0.01075269 0.00000000 0.013157895 0.020833333
##                        MB_5064    MB_5066    MB_5068     MB_5070    MB_5072
## chr1_762469_763020 0.008333333 0.01449275 0.00000000 0.008333333 0.00000000
##                        MB_5073    MB_5074    MB_5078     MB_5079     MB_5081
## chr1_762469_763020 0.005865103 0.02439024 0.02702703 0.009090909 0.009852217
##                       MB_5084    MB_5086     MB_5088    MB_5092     MB_5098
## chr1_762469_763020 0.00304878 0.00000000 0.000000000 0.01648352 0.004950495
##                        MB_5100     MB_5101     MB_5102     MB_5105    MB_5107
## chr1_762469_763020 0.000000000 0.000000000 0.028409091 0.004651163 0.01507538
##                        MB_5110    MB_5114     MB_5115     MB_5116     MB_5118
## chr1_762469_763020 0.005988024 0.01562500 0.000000000 0.013937282 0.003215434
##                        MB_5120     MB_5121     MB_5122     MB_5123    MB_5124
## chr1_762469_763020 0.008474576 0.000000000 0.021739130 0.013513514 0.00000000
##                        MB_5126     MB_5127     MB_5134     MB_5135     MB_5138
## chr1_762469_763020 0.000000000 0.051020408 0.000000000 0.000000000 0.000000000
##                        MB_5139    MB_5144    MB_5145     MB_5147    MB_5150
## chr1_762469_763020 0.000000000 0.00000000 0.01574803 0.000000000 0.00000000
##                       MB_5152    MB_5157    MB_5161     MB_5164    MB_5167
## chr1_762469_763020 0.00000000 0.00000000 0.00000000 0.000000000 0.04477612
##                        MB_5169    MB_5171    MB_5172     MB_5174    MB_5175
## chr1_762469_763020 0.026548673 0.00000000         NA 0.000000000 0.00000000
##                       MB_5176     MB_5182    MB_5184     MB_5186    MB_5188
## chr1_762469_763020 0.00000000 0.011111111 0.02564103 0.011406844 0.04147465
##                        MB_5190    MB_5191     MB_5193      MB_5197     MB_5200
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.0441176471          NA
##                         MB_5205    MB_5208    MB_5209     MB_5211     MB_5212
## chr1_762469_763020           NA 0.00000000 0.00000000 0.023809524 0.007874016
##                        MB_5214     MB_5215     MB_5218    MB_5221    MB_5222
## chr1_762469_763020 0.005763689 0.022727273 0.051282051 0.00000000 0.04177546
##                        MB_5223     MB_5224   MB_5225    MB_5226    MB_5227
## chr1_762469_763020 0.035335689 0.009900990 0.0000000 0.00000000 0.00000000
##                        MB_5230     MB_5232    MB_5233    MB_5235     MB_5238
## chr1_762469_763020 0.007874016 0.003802281 0.00000000 0.00000000 0.000000000
##                        MB_5239    MB_5243    MB_5244     MB_5251    MB_5256
## chr1_762469_763020 0.000000000 0.04166667 0.00000000 0.000000000 0.02453988
##                       MB_5259    MB_5261    MB_5264     MB_5266    MB_5270
## chr1_762469_763020 0.00000000 0.00000000 0.01388889 0.008771930         NA
##                        MB_5271    MB_5272     MB_5273     MB_5275     MB_5277
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.009049774 0.006250000
##                        MB_5278    MB_5279    MB_5280     MB_5281     MB_5284
## chr1_762469_763020 0.007142857 0.00000000 0.01562500 0.000000000 0.000000000
##                        MB_5286     MB_5287    MB_5288    MB_5292    MB_5293
## chr1_762469_763020 0.000000000 0.000000000 0.00000000 0.00000000 0.00000000
##                       MB_5298     MB_5306     MB_5311     MB_5312    MB_5313
## chr1_762469_763020 0.01242236 0.000000000 0.000000000 0.022304833 0.00000000
##                        MB_5315    MB_5322     MB_5327     MB_5329     MB_5331
## chr1_762469_763020 0.004048583 0.00000000 0.045714286 0.009950249 0.000000000
##                        MB_5334    MB_5335    MB_5338     MB_5339    MB_5341
## chr1_762469_763020 0.028169014 0.00000000 0.00000000 0.000000000 0.00000000
##                        MB_5345   MB_5348    MB_5350     MB_5351     MB_5358
## chr1_762469_763020 0.000000000 0.0000000 0.00000000 0.000000000 0.009433962
##                       MB_5360    MB_5361    MB_5364     MB_5366     MB_5381
## chr1_762469_763020         NA 0.01562500 0.01827676 0.029069767 0.006578947
##                       MB_5396     MB_5399    MB_5402     MB_5410     MB_5425
## chr1_762469_763020 0.01142857 0.000000000 0.00000000 0.000000000 0.006410256
##                        MB_5427     MB_5428    MB_5433    MB_5439     MB_5440
## chr1_762469_763020 0.000000000 0.000000000 0.00000000 0.01003344 0.000000000
##                        MB_5442    MB_5491    MB_5493     MB_5495     MB_5498
## chr1_762469_763020          NA 0.00000000 0.01204819 0.031250000 0.000000000
##                        MB_5513    MB_5519    MB_5530    MB_5533     MB_5543
## chr1_762469_763020 0.002777778 0.02298851 0.02985075 0.00000000 0.000000000
##                        MB_5558     MB_5559     MB_5563    MB_5567    MB_5580
## chr1_762469_763020 0.000000000 0.038461538 0.009090909 0.00000000 0.00000000
##                        MB_5582    MB_5584    MB_5588    MB_5592     MB_5593
## chr1_762469_763020 0.005586592 0.00000000 0.00000000 0.00913242 0.014814815
##                       MB_5596    MB_5603    MB_5605    MB_5614     MB_5620
## chr1_762469_763020 0.01666667 0.01315789 0.04705882 0.01273885 0.011299435
##                        MB_5634    MB_5640     MB_5641    MB_5642     MB_5646
## chr1_762469_763020 0.009900990 0.01398601 0.007092199 0.01775148 0.016393443
##                        MB_5651     MB_5656     MB_6007    MB_6008     MB_6010
## chr1_762469_763020 0.000000000 0.009615385 0.000000000 0.00000000 0.000000000
##                       MB_6011     MB_6012    MB_6018     MB_6019     MB_6023
## chr1_762469_763020 0.02054795 0.000000000 0.03846154 0.008695652 0.010752688
##                        MB_6024    MB_6025     MB_6026     MB_6030     MB_6036
## chr1_762469_763020 0.013274336 0.00000000 0.014705882 0.019607843 0.036585366
##                        MB_6039    MB_6042     MB_6044    MB_6046    MB_6047
## chr1_762469_763020 0.006024096 0.02142857 0.000000000 0.02898551 0.00952381
##                       MB_6048    MB_6049     MB_6050     MB_6052    MB_6053
## chr1_762469_763020 0.00952381         NA 0.000000000 0.000000000 0.00000000
##                        MB_6055    MB_6058    MB_6059    MB_6060    MB_6062
## chr1_762469_763020 0.000000000 0.00000000 0.00000000 0.01818182 0.03007519
##                        MB_6063     MB_6065    MB_6068     MB_6069    MB_6071
## chr1_762469_763020 0.013245033 0.008426966 0.00000000          NA 0.02702703
##                       MB_6075    MB_6077     MB_6079     MB_6080    MB_6082
## chr1_762469_763020 0.00000000 0.00000000 0.009433962 0.005714286 0.01910828
##                        MB_6083    MB_6085    MB_6090     MB_6092     MB_6097
## chr1_762469_763020 0.006578947 0.01117318 0.01257862 0.000000000 0.016194332
##                       MB_6098    MB_6100     MB_6101     MB_6103    MB_6105
## chr1_762469_763020 0.00000000 0.00000000 0.009803922 0.008032129 0.00000000
##                        MB_6107     MB_6108     MB_6113    MB_6114    MB_6116
## chr1_762469_763020 0.000000000 0.028409091 0.000000000 0.00000000 0.01183432
##                        MB_6118    MB_6122     MB_6124     MB_6125     MB_6131
## chr1_762469_763020 0.011173184 0.00000000 0.005291005 0.018867925 0.016000000
##                       MB_6133     MB_6135     MB_6138     MB_6141     MB_6143
## chr1_762469_763020 0.01027397 0.000000000 0.000000000 0.007462687 0.016000000
##                        MB_6144     MB_6145     MB_6146     MB_6147    MB_6149
## chr1_762469_763020 0.009950249 0.004975124 0.000000000 0.006191950 0.00000000
##                       MB_6150    MB_6152    MB_6154    MB_6156    MB_6157
## chr1_762469_763020 0.01212121 0.04022989 0.00000000 0.00000000 0.00000000
##                       MB_6160    MB_6163     MB_6164    MB_6167     MB_6168
## chr1_762469_763020 0.02531646 0.01435407 0.007407407 0.01197605 0.019230769
##                        MB_6169    MB_6171     MB_6178     MB_6179     MB_6181
## chr1_762469_763020 0.009090909 0.01282051 0.013245033 0.005586592 0.007407407
##                       MB_6184    MB_6185    MB_6187     MB_6188    MB_6192
## chr1_762469_763020 0.00000000 0.00000000 0.00000000 0.000000000 0.03389831
##                        MB_6194    MB_6195     MB_6201     MB_6204     MB_6207
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.030769231 0.019417476
##                       MB_6208    MB_6211     MB_6212     MB_6213    MB_6214
## chr1_762469_763020 0.00000000 0.00000000 0.000000000 0.000000000 0.00000000
##                        MB_6218    MB_6223     MB_6224     MB_6225    MB_6226
## chr1_762469_763020 0.000000000 0.03409091 0.000000000 0.024193548 0.01149425
##                       MB_6228     MB_6229    MB_6230     MB_6231    MB_6232
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.000000000 0.04166667
##                       MB_6233     MB_6234    MB_6237    MB_6238    MB_6239
## chr1_762469_763020 0.00000000 0.000000000         NA 0.03378378 0.00000000
##                        MB_6242     MB_6245     MB_6246    MB_6248    MB_6251
## chr1_762469_763020 0.000000000          NA 0.004081633 0.00000000 0.01666667
##                       MB_6253     MB_6254     MB_6256     MB_6257    MB_6259
## chr1_762469_763020 0.02127660 0.010256410 0.000000000 0.000000000 0.00000000
##                        MB_6261     MB_6263    MB_6269    MB_6271     MB_6272
## chr1_762469_763020 0.000000000 0.015000000 0.00000000 0.00000000 0.000000000
##                        MB_6274     MB_6275    MB_6281     MB_6283     MB_6284
## chr1_762469_763020 0.000000000 0.000000000 0.01941748 0.025157233 0.009259259
##                        MB_6286    MB_6287    MB_6288    MB_6289     MB_6291
## chr1_762469_763020          NA 0.00000000 0.00000000         NA 0.020833333
##                        MB_6293     MB_6294     MB_6297     MB_6300     MB_6302
## chr1_762469_763020 0.069892473 0.026785714 0.030674847 0.000000000 0.000000000
##                        MB_6305     MB_6306    MB_6308     MB_6312     MB_6314
## chr1_762469_763020 0.000000000 0.000000000 0.00000000 0.057692308 0.000000000
##                        MB_6317     MB_6318    MB_6319     MB_6322    MB_6328
## chr1_762469_763020 0.000000000          NA 0.00000000 0.005649718 0.01574803
##                        MB_6329    MB_6330    MB_6334    MB_6336     MB_6337
## chr1_762469_763020 0.000000000 0.01481481 0.00000000 0.00000000 0.000000000
##                        MB_6346     MB_6358     MB_6359     MB_6360     MB_6363
## chr1_762469_763020 0.041666667 0.025000000 0.021276596 0.004484305 0.020833333
##                        MB_7000     MB_7002    MB_7003    MB_7004    MB_7005
## chr1_762469_763020 0.018181818 0.000000000 0.00000000 0.00877193 0.02112676
##                        MB_7006     MB_7007    MB_7008     MB_7009     MB_7010
## chr1_762469_763020 0.000000000 0.011764706 0.03030303 0.009900990 0.006172840
##                        MB_7012     MB_7014     MB_7015     MB_7016     MB_7017
## chr1_762469_763020 0.015037594 0.018867925 0.009708738 0.000000000 0.000000000
##                       MB_7018     MB_7019     MB_7020     MB_7022     MB_7023
## chr1_762469_763020 0.00000000 0.004716981 0.000000000          NA 0.000000000
##                       MB_7026     MB_7028     MB_7029     MB_7030     MB_7031
## chr1_762469_763020 0.01703578 0.027777778 0.000000000 0.000000000 0.000000000
##                        MB_7032    MB_7034     MB_7035     MB_7036     MB_7037
## chr1_762469_763020 0.034188034 0.00000000 0.000000000 0.000000000 0.000000000
##                        MB_7038     MB_7039    MB_7040    MB_7041     MB_7043
## chr1_762469_763020 0.000000000 0.010638298 0.02857143 0.00000000 0.000000000
##                        MB_7046     MB_7048     MB_7049     MB_7050    MB_7051
## chr1_762469_763020 0.000000000 0.006872852 0.010638298 0.005291005 0.02298851
##                       MB_7052     MB_7053     MB_7054     MB_7055     MB_7056
## chr1_762469_763020 0.03448276 0.006410256 0.000000000 0.006134969 0.009803922
##                       MB_7057     MB_7058    MB_7060     MB_7061    MB_7062
## chr1_762469_763020 0.01507538 0.036363636 0.00000000 0.000000000 0.02189781
##                        MB_7065    MB_7066    MB_7067    MB_7068     MB_7070
## chr1_762469_763020 0.010309278 0.04320988 0.00000000 0.01351351 0.000000000
##                        MB_7071     MB_7072     MB_7073     MB_7074    MB_7075
## chr1_762469_763020 0.008849558 0.000000000 0.016666667 0.008928571 0.01481481
##                        MB_7076     MB_7077     MB_7078     MB_7079     MB_7080
## chr1_762469_763020 0.006211180 0.006802721 0.000000000 0.000000000 0.011494253
##                        MB_7083     MB_7084     MB_7085    MB_7086     MB_7087
## chr1_762469_763020 0.000000000 0.000000000 0.006134969 0.00000000 0.000000000
##                       MB_7088    MB_7089     MB_7090     MB_7091     MB_7092
## chr1_762469_763020 0.01190476 0.04020101 0.000000000 0.026881720 0.012987013
##                        MB_7093     MB_7095     MB_7096     MB_7097    MB_7099
## chr1_762469_763020 0.007874016 0.025641026 0.000000000 0.000000000 0.00000000
##                        MB_7100     MB_7102     MB_7104    MB_7107     MB_7109
## chr1_762469_763020 0.000000000 0.000000000 0.006134969 0.00000000 0.000000000
##                        MB_7113     MB_7115    MB_7116     MB_7118     MB_7119
## chr1_762469_763020 0.010309278 0.000000000         NA 0.000000000 0.000000000
##                        MB_7120    MB_7122    MB_7129    MB_7131    MB_7132
## chr1_762469_763020 0.000000000 0.00000000 0.00000000         NA 0.00729927
##                       MB_7137    MB_7138     MB_7141     MB_7142    MB_7144
## chr1_762469_763020 0.02000000 0.03738318 0.000000000 0.028037383 0.02985075
##                        MB_7145    MB_7151    MB_7153     MB_7154     MB_7157
## chr1_762469_763020 0.021551724 0.00000000 0.00000000 0.010309278 0.000000000
##                       MB_7158    MB_7159     MB_7160    MB_7161     MB_7162
## chr1_762469_763020 0.02739726 0.00000000 0.012820513 0.01136364 0.000000000
##                        MB_7164     MB_7167     MB_7170    MB_7171    MB_7172
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.01408451         NA
##                       MB_7173     MB_7174     MB_7182    MB_7186     MB_7187
## chr1_762469_763020         NA 0.032085561 0.000000000 0.02898551 0.000000000
##                        MB_7189     MB_7192    MB_7197     MB_7198     MB_7199
## chr1_762469_763020 0.000000000 0.032967033 0.00000000 0.000000000 0.008064516
##                        MB_7200     MB_7205     MB_7208     MB_7214     MB_7217
## chr1_762469_763020 0.000000000 0.000000000 0.000000000 0.000000000 0.006410256
##                        MB_7218    MB_7221     MB_7223    MB_7225     MB_7227
## chr1_762469_763020 0.009523810 0.00000000 0.000000000 0.00000000 0.050000000
##                        MB_7228     MB_7229    MB_7232     MB_7233    MB_7236
## chr1_762469_763020 0.000000000 0.000000000 0.01351351 0.000000000 0.00000000
##                        MB_7237     MB_7241     MB_7243     MB_7244    MB_7245
## chr1_762469_763020 0.017857143 0.000000000 0.000000000 0.000000000 0.00000000
##                        MB_7249    MB_7250     MB_7251     MB_7252     MB_7253
## chr1_762469_763020 0.000000000 0.00000000 0.000000000 0.017937220 0.000000000
##                       MB_7254     MB_7256     MB_7261     MB_7262    MB_7263
## chr1_762469_763020 0.01149425 0.000000000 0.000000000 0.000000000 0.00000000
##                       MB_7265     MB_7268     MB_7271    MB_7273    MB_7275
## chr1_762469_763020         NA 0.000000000 0.000000000 0.00000000 0.00000000
##                       MB_7278     MB_7279    MB_7281    MB_7283    MB_7285
## chr1_762469_763020 0.00000000 0.000000000 0.00000000 0.01829268         NA
##                        MB_7288     MB_7289    MB_7291    MB_7292     MB_7293
## chr1_762469_763020 0.000000000 0.000000000 0.00000000 0.00000000 0.006849315
##                       MB_7296     MB_7298
## chr1_762469_763020         NA 0.000000000
##  [ reached getOption(&quot;max.print&quot;) -- omitted 5 rows ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(all_mat_raw)</code></pre></div>
<pre><code>## [1] 201082   1782</code></pre>
</div>
<div id="clustering-of-normalized-methylation-of-er-tumors" class="section level2">
<h2><span class="header-section-number">4.4</span> Clustering of normalized methylation of ER+ tumors</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ER_positive_mat &lt;-<span class="st"> </span>all_norm_meth <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrom<span class="op">:</span>end, <span class="kw">any_of</span>(ER_positive_samples)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ER_positive_mat_raw &lt;-<span class="st"> </span>all_mat_raw[, ER_positive_samples]</code></pre></div>
<p>Filter loci that have low methylation (average of under 0.1):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">means &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(all_mat_raw, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
means_ER_positive &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(ER_positive_mat_raw, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meth_thresh &lt;-<span class="st"> </span><span class="fl">0.1</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
<span class="kw">tibble</span>(<span class="dt">m =</span> means_ER_positive) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>m))  <span class="op">+</span><span class="st"> </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_vline</span>(<span class="dt">xintercept=</span>meth_thresh) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ER_positive_mat_s &lt;-<span class="st"> </span>ER_positive_mat[means_ER_positive <span class="op">&gt;=</span><span class="st"> </span>meth_thresh, ]
<span class="kw">nrow</span>(ER_positive_mat_s)</code></pre></div>
<pre><code>## [1] 171026</code></pre>
<p>We sample 50k loci and calculate a correlation matrix of their methylation values in ER+ samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">17</span>)
ER_positive_mat_s &lt;-<span class="st"> </span>ER_positive_mat_s[<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ER_positive_mat_s), <span class="fl">5e4</span>), ]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cm &lt;-<span class="st"> </span><span class="kw">tgs_cor</span>(<span class="kw">t</span>(ER_positive_mat_s), <span class="dt">pairwise.complete.obs=</span><span class="ot">TRUE</span>) <span class="op">%cache_rds%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_cm_samp.rds&quot;</span>)</code></pre></div>
<p>We remove rows and columns without at least one correlation value above 0.25:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cm1 &lt;-<span class="st"> </span>cm
<span class="kw">diag</span>(cm1) &lt;-<span class="st"> </span><span class="ot">NA</span>
cor_maxs &lt;-<span class="st"> </span>matrixStats<span class="op">::</span><span class="kw">rowMaxs</span>(<span class="kw">abs</span>(cm1), <span class="dt">na.rm=</span><span class="ot">TRUE</span>)
f &lt;-<span class="st"> </span>cor_maxs <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.25</span>
cm_f &lt;-<span class="st"> </span>cm[f, f] </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(cm_f)</code></pre></div>
<pre><code>## [1] 45299 45299</code></pre>
<p>We cluster the correlation matrix using <code>hclust</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hc_meth &lt;-<span class="st"> </span><span class="kw">as.dist</span>(<span class="dv">1</span><span class="op">-</span>cm_f) <span class="op">%&gt;%</span><span class="st"> </span>fastcluster<span class="op">::</span><span class="kw">hclust</span>(<span class="dt">method =</span> <span class="st">&quot;ward.D2&quot;</span>) <span class="op">%cache_rds%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_cm_hclust.rds&quot;</span>)</code></pre></div>
<p>Reorder the dendrogram according to raw average methylation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hc_meth &lt;-<span class="st"> </span>vegan<span class="op">:::</span><span class="kw">reorder.hclust</span>(hc_meth, <span class="kw">rowMeans</span>(ER_positive_mat_raw[<span class="kw">rownames</span>(cm_f), ], <span class="dt">na.rm=</span><span class="ot">TRUE</span>)) <span class="op">%cache_rds%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_cm_hclust_reordered.rds&quot;</span>)</code></pre></div>
<p>We start by cutting the the tree with a large number of clusters (14) which we will then aim to merge.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">k &lt;-<span class="st"> </span><span class="dv">14</span></code></pre></div>
<div id="extended-data-figure-4a" class="section level4">
<h4><span class="header-section-number">4.4.0.1</span> Extended Data Figure 4a</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">8</span>, <span class="dt">repr.plot.height =</span> <span class="dv">8</span>)
<span class="kw">plot_meth_mat_cm</span>(cm_f, <span class="dt">k=</span>k, <span class="dt">width =</span> <span class="dv">1000</span>, <span class="dt">height =</span> <span class="dv">1000</span>, <span class="dt">hc_meth=</span>hc_meth, <span class="dt">downscale =</span> <span class="ot">TRUE</span>, <span class="dt">zlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.3</span>, <span class="fl">0.3</span>), <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;darkred&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;cyan&quot;</span>))</code></pre></div>
<pre><code>## downscaling matrix</code></pre>
<pre><code>## downscale k: 22</code></pre>
<pre><code>## plotting</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>We can see that there is a large group of correlated loci at the top right (9-14), another group in the middle (5), and another one at the bottom left (1-2). In addition, we have another 2 small clusters (3 and 7) which are anti-correlated to each other.</p>
<p>The other clusters (4,6,8) look weak in their intra correalation.</p>
<p>We term the first large group &quot;clock&quot;, the second &quot;ML&quot; and the last &quot;MG&quot; due to reasons that would be disscused at <code>clock</code> and <code>Epigenomic-instability</code> notebooks.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ct &lt;-<span class="st"> </span><span class="kw">cutree_order</span>(hc_meth, <span class="dt">k=</span>k)
ct_new &lt;-<span class="st"> </span><span class="kw">case_when</span>(
    ct <span class="op">%in%</span><span class="st"> </span><span class="dv">9</span><span class="op">:</span><span class="dv">14</span> <span class="op">~</span><span class="st"> &quot;clock&quot;</span>, 
    ct <span class="op">==</span><span class="st"> </span><span class="dv">5</span> <span class="op">~</span><span class="st"> &quot;ML&quot;</span>, 
    ct <span class="op">%in%</span><span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">2</span> <span class="op">~</span><span class="st"> &quot;MG&quot;</span>,
    ct <span class="op">==</span><span class="st"> </span><span class="dv">3</span> <span class="op">~</span><span class="st"> &quot;other1&quot;</span>,     
    ct <span class="op">==</span><span class="st"> </span><span class="dv">7</span> <span class="op">~</span><span class="st"> &quot;other2&quot;</span>, 
    <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &quot;no_cor&quot;</span>)
<span class="kw">names</span>(ct_new) &lt;-<span class="st"> </span><span class="kw">names</span>(ct)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clust_df &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(ct_new) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mat_to_intervs</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">clust =</span> V1) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_clust.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
clust_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(clust)</code></pre></div>
<pre><code>## # A tibble: 6 x 2
##    clust     n
## 1  clock 21657
## 2     MG  4102
## 3     ML  2448
## 4 no_cor 13120
## 5 other1   821
## 6 other2  3151</code></pre>
<p>We will generate a score for each tumor based on the mean methylation of each group:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_mat &lt;-<span class="st"> </span><span class="kw">tgs_matrix_tapply</span>(all_norm_meth <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="kw">names</span>(ct_new), ] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>(), ct_new, mean, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>()</code></pre></div>
<p>We add the TME features for comparison:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tme_df &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/TME_features.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
feats_mat &lt;-<span class="st"> </span><span class="kw">cbind</span>(feats_mat, tme_df  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, caf, immune, caf.meth, immune.meth) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;samp&quot;</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_df &lt;-<span class="st"> </span>feats_mat <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>other1, <span class="op">-</span>other2, <span class="op">-</span>no_cor) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">ER=</span>ER1), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">select</span>(samp, ER, <span class="kw">everything</span>()) <span class="op">%cache_df%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">here</span>(<span class="st">&quot;data/epigenomic_features.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(feats_df)</code></pre></div>
<pre><code>## # A tibble: 6 x 9
##      samp  ER        clock          MG           ML        caf     immune
## 1 MB_0006 ER+  0.020284548 -0.09092389 -0.006315781 -0.2441946 -0.1114837
## 2 MB_0028 ER+  0.134212088 -0.02794677 -0.011634957 -1.7266095 -0.9358944
## 3 MB_0046 ER+  0.017717551  0.01210391 -0.103759329 -0.6545081 -1.0023582
## 4 MB_0050 ER+  0.003517646 -0.05415309 -0.046344401  1.7038169  0.1281640
## 5 MB_0053 ER+ -0.029721111  0.01603443  0.075428738 -0.8152407  0.2103729
## 6 MB_0054 ER+  0.067999324  0.04493482 -0.001099083 -0.5608747  0.9292829
##    caf.meth immune.meth
## 1 0.2950609   0.4722494
## 2 0.4851022   0.6191494
## 3 0.4210078   0.6308352
## 4 0.2336145   0.4808316
## 5 0.4603347   0.5915159
## 6 0.4790603   0.6093489</code></pre>
<p>We save the the cluster association of each locus:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_clust &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(ct_new) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mat_to_intervs</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">clust =</span> V1) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_clust_df.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<p>For the 3 large meta-clusters (clock, MG, ML) we calcualte the <strong>raw</strong> (before TME normalization) average methylation in each sample:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_f &lt;-<span class="st"> </span>loci_clust <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(clust <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_mat_raw &lt;-<span class="st"> </span><span class="kw">tgs_matrix_tapply</span>(all_mat_raw[<span class="kw">rownames</span>(loci_f), ] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>(), loci_f[, <span class="dv">1</span>], mean, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;samp&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_df_raw &lt;-<span class="st"> </span>tme_df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(feats_mat_raw, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">add_ER</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(samp, ER, <span class="kw">everything</span>()) <span class="op">%fcache_df%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">here</span>(<span class="st">&quot;data/epigenomic_features_raw_meth.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>()
<span class="kw">head</span>(feats_df_raw)</code></pre></div>
<pre><code>## # A tibble: 6 x 9
##      samp  ER        caf     immune  caf.meth immune.meth     clock        MG
## 1 MB_0006 ER+ -0.2441946 -0.1114837 0.2950609   0.4722494 0.7843678 0.2557624
## 2 MB_0028 ER+ -1.7266095 -0.9358944 0.4851022   0.6191494 0.8746526 0.3684660
## 3 MB_0046 ER+ -0.6545081 -1.0023582 0.4210078   0.6308352 0.7931583 0.3512826
## 4 MB_0050 ER+  1.7038169  0.1281640 0.2336145   0.4808316 0.8128158 0.1990629
## 5 MB_0053 ER+ -0.8152407  0.2103729 0.4603347   0.5915159 0.7309033 0.3691415
## 6 MB_0054 ER+ -0.5608747  0.9292829 0.4790603   0.6093489 0.8237875 0.3819997
##          ML
## 1 0.7020038
## 2 0.7049314
## 3 0.5910172
## 4 0.6801434
## 5 0.7798542
## 6 0.6909709</code></pre>
</div>
</div>
<div id="project-clustering-on-er--correlation-matrix" class="section level2">
<h2><span class="header-section-number">4.5</span> Project clustering on ER- correlation matrix</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ER_negative_mat &lt;-<span class="st"> </span>all_norm_meth <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrom<span class="op">:</span>end, <span class="kw">any_of</span>(ER_negative_samples)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ER_negative_mat_f &lt;-<span class="st"> </span>ER_negative_mat[<span class="kw">rownames</span>(cm_f), ]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(ER_negative_mat_f)</code></pre></div>
<pre><code>## [1] 45299   310</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cm_f_neg &lt;-<span class="st"> </span><span class="kw">tgs_cor</span>(<span class="kw">t</span>(ER_negative_mat_f), <span class="dt">pairwise.complete.obs=</span><span class="ot">TRUE</span>) <span class="op">%cache_rds%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/ER_negative_loci_cm_samp.rds&quot;</span>)</code></pre></div>
<div id="extended-data-figure-4b" class="section level4">
<h4><span class="header-section-number">4.5.0.1</span> Extended Data Figure 4b</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">8</span>, <span class="dt">repr.plot.height =</span> <span class="dv">8</span>)
<span class="kw">plot_meth_mat_cm</span>(cm_f_neg, <span class="dt">k=</span>k, <span class="dt">width =</span> <span class="dv">1000</span>, <span class="dt">height =</span> <span class="dv">1000</span>, <span class="dt">hc_meth=</span>hc_meth, <span class="dt">downscale =</span> <span class="ot">TRUE</span>, <span class="dt">zlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.3</span>, <span class="fl">0.3</span>), <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;darkred&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;cyan&quot;</span>))</code></pre></div>
<pre><code>## downscaling matrix</code></pre>
<pre><code>## downscale k: 22</code></pre>
<pre><code>## plotting</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<p>We can see that the clock, MG and ML are preserved also in ER- tumors.</p>
</div>
</div>
<div id="classify-loci-in-the-genome-to-epigenomic-features" class="section level2">
<h2><span class="header-section-number">4.6</span> Classify loci in the genome to epigenomic features</h2>
<p>See <code>Loss-clock</code> and <code>Epignomic-instability</code> notebooks for further characterization of the epigenomic scores. We would now continue to look at the broad genomic picture and try to classify loci in the genome based on their correlation with the scores.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate the correlation between normalized methylation and the epigenomic features</span>
epi_features_loci_cors &lt;-<span class="st"> </span><span class="cf">function</span>(all_norm_meth, feats_df, samples){
    samples &lt;-<span class="st"> </span><span class="kw">intersect</span>(samples, feats_df<span class="op">$</span>samp)
    samples &lt;-<span class="st"> </span><span class="kw">intersect</span>(samples, <span class="kw">colnames</span>(all_norm_meth)[<span class="op">-</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)])
    mat &lt;-<span class="st"> </span>all_norm_meth <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrom<span class="op">:</span>end, <span class="kw">any_of</span>(samples)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>()
    feats_mat &lt;-<span class="st"> </span>feats_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.matrix</span>()
    
    loci_cors &lt;-<span class="st"> </span><span class="kw">tgs_cor</span>(<span class="kw">t</span>(mat[, samples]), feats_mat[samples, ], <span class="dt">pairwise.complete.obs =</span> <span class="ot">TRUE</span>)
    loci_cors &lt;-<span class="st"> </span>loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mat_to_intervs</span>()
    <span class="kw">return</span>(loci_cors)    
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_cors &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(
        <span class="kw">epi_features_loci_cors</span>(all_norm_meth, feats_df, ER_positive_samples) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="st">&quot;ER+&quot;</span>), 
        <span class="kw">epi_features_loci_cors</span>(all_norm_meth, feats_df, ER_negative_samples) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="st">&quot;ER-&quot;</span>), 
        <span class="kw">epi_features_loci_cors</span>(all_norm_meth, feats_df, normal_samples) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="st">&quot;normal&quot;</span>)
    ) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(chrom<span class="op">:</span>end, ER, <span class="kw">everything</span>()) <span class="op">%fcache_df%</span>
<span class="st">    </span><span class="kw">here</span>(<span class="st">&quot;data/features_loci_cors.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>()    </code></pre></div>
<p>Looking at all the correaltions we see that there are loci who are not correlated to any of the featurers:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">7</span>)
loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>MG, <span class="dt">y=</span>ML, <span class="dt">color=</span>clock)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.0001</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>()</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>ML, <span class="dt">y=</span>clock, <span class="dt">color=</span>MG)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.0001</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>()</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-35-2.png" width="672" /></p>
<p>We use PCA (pricinpal component analysis) on the features for visualization of the feature space</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(chrom, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>()</code></pre></div>
<pre><code>## [1] 201082</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pca &lt;-<span class="st"> </span>loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>ER, <span class="op">-</span>immune, <span class="op">-</span>caf) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">prcomp</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>pca<span class="op">$</span>rotation <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mat_to_intervs</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(loci_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>), <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div id="figure-2a" class="section level4">
<h4><span class="header-section-number">4.6.0.1</span> Figure 2a</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">5</span>)

remove_axis &lt;-<span class="st"> </span><span class="cf">function</span>(x)
        <span class="kw">theme</span>( <span class="dt">aspect.ratio =</span> <span class="dv">1</span>,
            <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),
            <span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),
            <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),
            <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(),
            <span class="dt">axis.text =</span> <span class="kw">element_blank</span>()
        )
point_size =<span class="st"> </span><span class="fl">1e-10</span>
viridis_opt &lt;-<span class="st"> &quot;E&quot;</span>
limits &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="op">-</span><span class="fl">0.25</span>, <span class="fl">0.8</span>)
p_clock_proj &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC1, <span class="dt">y=</span>PC2, <span class="dt">color=</span>clock)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span>point_size)   <span class="op">+</span><span class="st"> </span><span class="kw">remove_axis</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>(<span class="dt">option =</span> viridis_opt, <span class="dt">limits =</span> limits)
p_MG_proj &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC1, <span class="dt">y=</span>PC2, <span class="dt">color=</span>MG)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span>point_size)   <span class="op">+</span><span class="st"> </span><span class="kw">remove_axis</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>(<span class="dt">option =</span> viridis_opt, <span class="dt">limits =</span> limits)
p_ML_proj &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC1, <span class="dt">y=</span>PC2, <span class="dt">color=</span>ML)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span>point_size)   <span class="op">+</span><span class="st"> </span><span class="kw">remove_axis</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>(<span class="dt">option =</span> viridis_opt, <span class="dt">limits =</span> limits)
p_immune_proj &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC1, <span class="dt">y=</span>PC2, <span class="dt">color=</span>immune)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span>point_size) <span class="op">+</span><span class="st"> </span><span class="kw">remove_axis</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>(<span class="dt">option =</span> viridis_opt, <span class="dt">limits =</span> limits)
p_caf_proj &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC1, <span class="dt">y=</span>PC2, <span class="dt">color=</span>caf)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span>point_size)  <span class="op">+</span><span class="st"> </span><span class="kw">remove_axis</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_viridis_c</span>(<span class="dt">option =</span> viridis_opt, <span class="dt">limits =</span> limits)

(p_clock_proj <span class="op">+</span><span class="st"> </span>p_MG_proj <span class="op">+</span><span class="st"> </span>p_ML_proj) </code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(p_immune_proj <span class="op">+</span><span class="st"> </span>p_caf_proj <span class="op">+</span><span class="st"> </span><span class="kw">plot_spacer</span>())</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-39-2.png" width="672" /></p>
</div>
</div>
<div id="average-methylation-of-the-features" class="section level2">
<h2><span class="header-section-number">4.7</span> Average methylation of the features</h2>
<p>We will now move to look at the average methylation distribution of the scores. We will do that computing the average methylation (in ER+/ER-/normal) of loci that are highly correlated to them (above 0.6 for clock and above 0.5 for MG and ML).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_annot &lt;-<span class="st"> </span>loci_cors <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(<span class="kw">get_loci_annot</span>(), <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%cache_df%</span>
<span class="st">    </span><span class="kw">here</span>(<span class="st">&quot;data/loci_annot_epigenomic_features.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>()
<span class="kw">head</span>(loci_annot)</code></pre></div>
<pre><code>## # A tibble: 6 x 18
##   chrom  start    end        clock          MG          ML          caf
## 1  chr1 762469 763020  0.001575984 -0.01428874 -0.01551105  0.012110476
## 2  chr1 762676 763227 -0.129625330  0.33612856  0.03607766 -0.006588455
## 3  chr1 860619 861170 -0.050509334  0.29208240  0.14625292 -0.008582681
## 4  chr1 895465 896016  0.068537024  0.34194079  0.04775239  0.007077953
## 5  chr1 901375 901926  0.092230378  0.05256030  0.07306855  0.004076673
## 6  chr1 948345 948896  0.019768875  0.02448860  0.03588616 -0.003594507
##          immune    caf.meth immune.meth    cg_cont      tor tss_d    k27me3
## 1 -0.0038356268 -0.01015590  0.01583110 0.10130379 64.78630   158 0.9678000
## 2 -0.0051811514  0.27247376  0.13967267 0.12012641 64.80790   -48 0.9832893
## 3  0.0122705825  0.21490060  0.09494949 0.12763213 74.13830  -226 0.9983592
## 4  0.0037199954  0.25625416  0.17188417 0.09146498 78.45700  -226 0.9882345
## 5  0.0119557122  0.01421310 -0.01829405 0.05323296 79.12555  -226 0.9923163
## 6  0.0001512059  0.01424724  0.02610006 0.03862698 82.83044  -226 0.8459000
##   k4me1_luminal k4me1_myo1 k4me1_myo2 k4me1_hmec
## 1     0.5759000     0.8639  0.6325000  0.8790000
## 2     0.5759000     0.8639  0.7695000  0.8871000
## 3     0.9742000     0.9781  0.9804800  0.9547000
## 4     0.9742000     0.9603  0.9639000  0.9397000
## 5     0.9995784     0.9781  0.9962854  0.9994769
## 6     0.9940496     0.9704  0.9578000  0.9647000</code></pre>
<p>See exons-TME notebook for the generation of this file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_annot_exons &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/exon_annot_epigenomic_features.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
</div>
<div id="plot-distribution-over-samples-of-the-correlated-loci" class="section level2">
<h2><span class="header-section-number">4.8</span> Plot distribution over samples of the correlated loci</h2>
<p>We calculate the average methylation of each sample in loci that pass the threshold of every score:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_MG &lt;-<span class="st"> </span><span class="kw">colMeans</span>(all_mat_raw[loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(MG <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames</span>(), ], <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;MG&quot;</span>)
df_ML &lt;-<span class="st"> </span><span class="kw">colMeans</span>(all_mat_raw[loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ML <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames</span>(), ], <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;ML&quot;</span>)
df_clock &lt;-<span class="st"> </span><span class="kw">colMeans</span>(all_mat_raw[loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(clock <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames</span>(), ], <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;clock&quot;</span>)
df_sum &lt;-<span class="st"> </span>df_clock <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(df_MG, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(df_ML, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">add_ER</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ER))</code></pre></div>
<p>We can now plot the distribution of scores over samples:</p>
<div id="figure-1i" class="section level4">
<h4><span class="header-section-number">4.8.0.1</span> Figure 1i</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
p_avg_clock &lt;-<span class="st"> </span>df_sum <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>clock, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Clock avg. methylation&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim=</span><span class="kw">c</span>(<span class="fl">0.25</span>, <span class="dv">1</span>))

p_avg_clock <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.9</span>)    </code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
</div>
<div id="figure-2b" class="section level4">
<h4><span class="header-section-number">4.8.0.2</span> Figure 2b</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
p_avg_MG &lt;-<span class="st"> </span>df_sum <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>MG, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;MG avg. methylation&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.6</span>))

p_avg_MG <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.9</span>)    </code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-44-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
p_avg_ML &lt;-<span class="st"> </span>df_sum <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>ML, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;ML avg. methylation&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim=</span><span class="kw">c</span>(<span class="fl">0.4</span>, <span class="dv">1</span>))

p_avg_ML <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.9</span>)    </code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-45-1.png" width="672" /></p>
<p>Calculate p-values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_sum <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;meth&quot;</span>, <span class="op">-</span>samp, <span class="op">-</span>ER) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(feat) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">summarise</span>(
        <span class="dt">p_pos_norm =</span> <span class="kw">ks.test</span>(meth[ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>], meth[ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>])<span class="op">$</span>p.value, 
        <span class="dt">p_neg_norm =</span> <span class="kw">ks.test</span>(meth[ER <span class="op">==</span><span class="st"> &quot;ER-&quot;</span>], meth[ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>])<span class="op">$</span>p.value)</code></pre></div>
<pre><code>## # A tibble: 3 x 3
##    feat p_pos_norm p_neg_norm
## 1 clock          0          0
## 2    MG          0          0
## 3    ML          0          0</code></pre>
</div>
</div>
<div id="plot-loci-classification" class="section level2">
<h2><span class="header-section-number">4.9</span> Plot loci classification</h2>
<p>Next, we classify loci to &quot;Promoters&quot;, &quot;Enhancers&quot; or &quot;Background&quot; (non promoter/enhancer), and stratify by CpG content.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_loci_class &lt;-<span class="st"> </span>loci_annot <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">left_join</span>(promoter_intervs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(chrom, start, end) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="st">&quot;Promoters&quot;</span>), <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) 

k4me1_names &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;k4me1&quot;</span>, <span class="kw">colnames</span>(loci_annot), <span class="dt">value=</span><span class="ot">TRUE</span>)
df_loci_class &lt;-<span class="st"> </span>df_loci_class <span class="op">%&gt;%</span><span class="st">            </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">type =</span> 
        <span class="kw">case_when</span>(            
            type <span class="op">==</span><span class="st"> &quot;Promoters&quot;</span> <span class="op">~</span><span class="st"> &quot;Promoters&quot;</span>,
            matrixStats<span class="op">::</span><span class="kw">rowAnys</span>((df_loci_class[, k4me1_names] <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.97</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>) <span class="op">~</span><span class="st"> &quot;Enhancers&quot;</span>, 
            <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &quot;Background&quot;</span>
        )
    ) <span class="op">%&gt;%</span><span class="st">                           </span>
<span class="st">    </span><span class="kw">bind_rows</span>(loci_annot_exons <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="st">&quot;Exons&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">cg_cont =</span> <span class="kw">cut</span>(cg_cont, <span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">0.04</span>,<span class="fl">0.08</span>,<span class="dv">1</span>), <span class="dt">include.lowest=</span><span class="ot">TRUE</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Low&quot;</span>, <span class="st">&quot;Mid&quot;</span>, <span class="st">&quot;High&quot;</span>))) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_loci_layer &lt;-<span class="st"> </span>df_loci_class <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(<span class="kw">get_all_summary_meth</span>(), <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">normal_type =</span> <span class="kw">cut</span>(normal, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>, <span class="fl">0.9</span>, <span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;0-0.2&quot;</span>, <span class="st">&quot;0.2-0.9&quot;</span>, <span class="st">&quot;0.9-1&quot;</span>), <span class="dt">include.lowest=</span><span class="ot">TRUE</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cor_thresh &lt;-<span class="st"> </span><span class="fl">0.25</span>

df_loci_layer &lt;-<span class="st"> </span>df_loci_layer <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">layer =</span> <span class="kw">case_when</span>(clock <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;clock&quot;</span>, MG <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;MG&quot;</span>, ML <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;ML&quot;</span>, <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &quot;other&quot;</span>), <span class="dt">layer =</span> <span class="kw">factor</span>(layer, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;other&quot;</span>, <span class="st">&quot;clock&quot;</span>, <span class="st">&quot;ML&quot;</span>, <span class="st">&quot;MG&quot;</span>)))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_loci_layer <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(type, layer)  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</code></pre></div>
<pre><code>##          type layer     n
## 1  Background other 54248
## 2  Background clock 74369
## 3  Background    ML 11758
## 4  Background    MG  7044
## 5   Enhancers other 16628
## 6   Enhancers clock  5642
## 7   Enhancers    ML  5299
## 8   Enhancers    MG 10401
## 9       Exons other 17923
## 10      Exons clock  5146
## 11      Exons    ML  1944
## 12      Exons    MG  4499
## 13  Promoters other 11539
## 14  Promoters clock   780
## 15  Promoters    ML   379
## 16  Promoters    MG  2995</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_loci_layer <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;Enhancers&quot;</span>, cg_cont <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Mid&quot;</span>, <span class="st">&quot;High&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(layer)  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">p =</span> n <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</code></pre></div>
<pre><code>##   layer    n          p
## 1 other 7076 0.41197019
## 2 clock 1260 0.07335817
## 3    ML 1117 0.06503260
## 4    MG 7723 0.44963903</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_loci_layer <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;Enhancers&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">count</span>(normal_type, layer) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(normal_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">p =</span> n <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(n)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as.data.frame</span>()</code></pre></div>
<pre><code>##    normal_type layer    n           p
## 1        0-0.2 other 7141 0.471228718
## 2        0-0.2 clock  573 0.037811799
## 3        0-0.2    ML  184 0.012142009
## 4        0-0.2    MG 7256 0.478817474
## 5      0.2-0.9 other 7208 0.393965894
## 6      0.2-0.9 clock 3933 0.214965020
## 7      0.2-0.9    ML 4021 0.219774814
## 8      0.2-0.9    MG 3134 0.171294272
## 9        0.9-1 other 2279 0.504203540
## 10       0.9-1 clock 1136 0.251327434
## 11       0.9-1    ML 1094 0.242035398
## 12       0.9-1    MG   11 0.002433628</code></pre>
<div id="figure-2c" class="section level4">
<h4><span class="header-section-number">4.9.0.1</span> Figure 2c</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">6</span>)
df &lt;-<span class="st"> </span>df_loci_layer <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;Enhancers&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">count</span>(normal_type, layer) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(normal_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">p =</span> n <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(n))

p_enh_layer_perc &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(layer <span class="op">!=</span><span class="st"> &quot;other&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>normal_type, <span class="dt">y=</span>p, <span class="dt">fill=</span>layer)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_col</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_fill_manual</span>(<span class="dt">name =</span> <span class="st">&quot;&quot;</span>, <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;ML&quot;</span> =<span class="st"> &quot;darkblue&quot;</span>, <span class="st">&quot;MG&quot;</span> =<span class="st"> &quot;darkred&quot;</span>, <span class="st">&quot;clock&quot;</span> =<span class="st"> &quot;darkgreen&quot;</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">labels=</span>scales<span class="op">::</span>percent) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">vertical_labs</span>() <span class="op">+</span><span class="st">         </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Methylation in normal&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;% of enhancers&quot;</span>)

p_enh_layer_perc <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">vertical_labs</span>()</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
</div>
<div id="extended-data-figure-6c" class="section level4">
<h4><span class="header-section-number">4.9.0.2</span> Extended Data Figure 6c</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">10</span>, <span class="dt">repr.plot.height =</span> <span class="dv">10</span>)
df &lt;-<span class="st"> </span>df_loci_class <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(cg_cont)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">cg_cont =</span> <span class="kw">case_when</span>(cg_cont <span class="op">==</span><span class="st"> &quot;Low&quot;</span> <span class="op">~</span><span class="st"> &quot;Low&quot;</span>, cg_cont <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Mid&quot;</span>, <span class="st">&quot;High&quot;</span>) <span class="op">~</span><span class="st"> &quot;Mid/High&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(
        <span class="dt">cg_cont =</span> <span class="kw">paste</span>(cg_cont, <span class="st">&quot;CpG cont.&quot;</span>), 
        <span class="dt">cg_cont =</span> <span class="kw">factor</span>(cg_cont, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Low CpG cont.&quot;</span>, <span class="st">&quot;Mid/High CpG cont.&quot;</span>)), 
        <span class="dt">type =</span> <span class="kw">factor</span>(type, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Promoters&quot;</span>, <span class="st">&quot;Enhancers&quot;</span>, <span class="st">&quot;Background&quot;</span>, <span class="st">&quot;Exons&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(chrom, start, end, type, cg_cont, clock, MG, ML) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;class&quot;</span>, <span class="st">&quot;cor&quot;</span>, <span class="op">-</span>(chrom<span class="op">:</span>end), <span class="op">-</span>cg_cont, <span class="op">-</span>type)

p_cor_density &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>cor, <span class="dt">color=</span>class)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_density</span>(<span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">name =</span> <span class="st">&quot;&quot;</span>, <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;ML&quot;</span> =<span class="st"> &quot;darkblue&quot;</span>, <span class="st">&quot;MG&quot;</span> =<span class="st"> &quot;darkred&quot;</span>, <span class="st">&quot;clock&quot;</span> =<span class="st"> &quot;darkgreen&quot;</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">guides</span>(<span class="dt">color=</span><span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Correlation to epigenomic score&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">facet_grid</span>(type<span class="op">~</span>cg_cont, <span class="dt">scales=</span><span class="st">&quot;free_y&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> cor_thresh, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="dt">color=</span><span class="st">&quot;darkgray&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_text</span>(<span class="dt">data =</span> df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(type, cg_cont) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(<span class="st">&quot;n=&quot;</span>, scales<span class="op">::</span><span class="kw">comma</span>(n))), <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">label=</span>label), <span class="dt">x =</span> <span class="fl">0.5</span>, <span class="dt">y =</span> <span class="dv">4</span>)</code></pre></div>
<pre><code>## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## &quot;none&quot;)` instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_cor_density <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
</div>
</div>
<div id="project-all-epigenomic-features-using-umap" class="section level2">
<h2><span class="header-section-number">4.10</span> Project all epigenomic features using UMAP</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">17</span>)
raw_feats &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/epigenomic_features_raw_meth.tsv&quot;</span>))
um &lt;-<span class="st"> </span>raw_feats <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">ML =</span> <span class="op">-</span>ML, <span class="dt">clock =</span> <span class="op">-</span>clock, <span class="dt">immune.meth =</span> <span class="op">-</span>immune.meth, <span class="dt">caf.meth =</span> <span class="op">-</span>caf.meth) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">caf =</span> caf <span class="op">/</span><span class="st"> </span><span class="dv">20</span>, <span class="dt">immune =</span> immune <span class="op">/</span><span class="st"> </span><span class="dv">20</span>) <span class="op">%&gt;%</span><span class="st">  </span>
<span class="st">        </span><span class="kw">select</span>(caf, immune, clock, MG, ML) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span>
<span class="st">        </span>umap<span class="op">::</span><span class="kw">umap</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mut_df &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/mutations.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
p53_df &lt;-<span class="st"> </span>mut_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">ER =</span> ER1), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(gene <span class="op">==</span><span class="st"> &quot;TP53&quot;</span>) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">p53=</span> forcats<span class="op">::</span><span class="kw">fct_recode</span>(mutation, <span class="st">&quot;WT&quot;</span> =<span class="st"> &quot;NO MUT&quot;</span>, <span class="st">&quot;TP53+&quot;</span> =<span class="st"> &quot;MUT+&quot;</span>, <span class="st">&quot;TP53-&quot;</span> =<span class="st"> &quot;MUT-&quot;</span>)) <span class="op">%&gt;%</span><span class="st">    </span>
<span class="st">        </span><span class="kw">select</span>(samp, p53)</code></pre></div>
<pre><code>## Warning: Unknown levels in `f`: MUT+</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pik3ca_df &lt;-<span class="st"> </span>mut_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">ER =</span> ER1), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(gene <span class="op">==</span><span class="st"> &quot;PIK3CA&quot;</span>) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">pik3ca =</span> forcats<span class="op">::</span><span class="kw">fct_recode</span>(mutation, <span class="st">&quot;WT&quot;</span> =<span class="st"> &quot;NO MUT&quot;</span>, <span class="st">&quot;PIK3CA+&quot;</span> =<span class="st"> &quot;MUT+&quot;</span>, <span class="st">&quot;PIK3CA-&quot;</span> =<span class="st"> &quot;MUT-&quot;</span>)) <span class="op">%&gt;%</span><span class="st">               </span><span class="kw">select</span>(samp, pik3ca)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_proj &lt;-<span class="st"> </span>raw_feats <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">ML =</span> <span class="op">-</span>ML, <span class="dt">clock =</span> <span class="op">-</span>clock, <span class="dt">immune.meth =</span> <span class="op">-</span>immune.meth, <span class="dt">caf.meth =</span> <span class="op">-</span>caf.meth) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">x =</span> um<span class="op">$</span>layout[, <span class="dv">1</span>], <span class="dt">y =</span> um<span class="op">$</span>layout[, <span class="dv">2</span>]) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, iC10, stage, grade, PAM50, <span class="dt">gi =</span> giScore_width, MathScore, <span class="dt">epi_burden =</span> log10_global_epm), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">iC10 =</span> <span class="kw">factor</span>(iC10, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>iC10))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">stage =</span> <span class="kw">factor</span>(stage, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>stage))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">grade =</span> <span class="kw">factor</span>(grade, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>grade))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">PAM50 =</span> <span class="kw">factor</span>(PAM50, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>PAM50))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(p53_df, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">p53 =</span> forcats<span class="op">::</span><span class="kw">fct_explicit_na</span>(p53)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(pik3ca_df, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">pik3ca =</span> forcats<span class="op">::</span><span class="kw">fct_explicit_na</span>(pik3ca)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">grade =</span> <span class="kw">ifelse</span>(ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>, <span class="st">&quot;ADJNORMAL&quot;</span>, grade)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">stage =</span> <span class="kw">ifelse</span>(ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>, <span class="st">&quot;ADJNORMAL&quot;</span>, stage))</code></pre></div>
<div id="figure-4a." class="section level4">
<h4><span class="header-section-number">4.10.0.1</span> Figure 4a.</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">20</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
plotlist &lt;-<span class="st"> </span><span class="kw">list</span>()
plotlist<span class="op">$</span>ER &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> ER)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1)

plotlist<span class="op">$</span>iC10 &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> iC10)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>iC10)

plotlist<span class="op">$</span>PAM50 &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> PAM50)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>PAM50)

plotlist<span class="op">$</span>grade &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> grade)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>grade)

plotlist<span class="op">$</span>stage &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> stage)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>stage)

<span class="cf">for</span> (feat <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;immune&quot;</span>, <span class="st">&quot;caf&quot;</span>, <span class="st">&quot;clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>, <span class="st">&quot;gi&quot;</span>, <span class="st">&quot;MathScore&quot;</span>, <span class="st">&quot;epi_burden&quot;</span>)) {
    df &lt;-<span class="st"> </span>feats_proj
    df[[feat]] &lt;-<span class="st"> </span><span class="kw">clip_vals</span>(df[[feat]], <span class="kw">quantile</span>(df[[feat]], <span class="fl">0.05</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="kw">quantile</span>(df[[feat]], <span class="fl">0.95</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>))
    plotlist[[feat]] &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes_string</span>(<span class="dt">x =</span> <span class="st">&quot;x&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;y&quot;</span>, <span class="dt">color =</span> feat)) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_colour_distiller</span>(<span class="dt">palette =</span> <span class="st">&quot;RdYlBu&quot;</span>)
}

plotlist<span class="op">$</span>p53 &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> p53)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;TP53+&quot;</span> =<span class="st"> &quot;darkred&quot;</span>, <span class="st">&quot;TP53-&quot;</span> =<span class="st"> &quot;darkred&quot;</span>, <span class="st">&quot;WT&quot;</span> =<span class="st"> &quot;darkgray&quot;</span>, <span class="st">&quot;(Missing)&quot;</span> =<span class="st"> &quot;darkgray&quot;</span>))
plotlist<span class="op">$</span>pik3ca &lt;-<span class="st"> </span>feats_proj <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> pik3ca)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;PIK3CA-&quot;</span> =<span class="st"> &quot;darkred&quot;</span>, <span class="st">&quot;PIK3CA+&quot;</span> =<span class="st"> &quot;darkblue&quot;</span>, <span class="st">&quot;WT&quot;</span> =<span class="st"> &quot;darkgray&quot;</span>, <span class="st">&quot;(Missing)&quot;</span> =<span class="st"> &quot;darkgray&quot;</span>))

plotlist1 &lt;-<span class="st"> </span><span class="kw">map</span>(plotlist, <span class="op">~</span>
.x <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(
        <span class="dt">aspect.ratio =</span> <span class="dv">1</span>,
        <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),
        <span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),
        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),
        <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(),
        <span class="dt">axis.text =</span> <span class="kw">element_blank</span>()
    ))

p &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> <span class="kw">map</span>(plotlist1, <span class="op">~</span><span class="st"> </span>.x <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.8</span>)))

plots_ppt1 &lt;-<span class="st"> </span><span class="kw">map2</span>(
        plotlist[<span class="kw">c</span>(<span class="st">&quot;immune&quot;</span>, <span class="st">&quot;caf&quot;</span>, <span class="st">&quot;clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>)],
        <span class="kw">c</span>(<span class="st">&quot;Immune&quot;</span>, <span class="st">&quot;CAF&quot;</span>, <span class="st">&quot;Clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>),
        <span class="op">~</span><span class="st"> </span>.x <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.05</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(.y)
    )

plots_ppt2 &lt;-<span class="st"> </span><span class="kw">map2</span>(
        plotlist[<span class="kw">c</span>(<span class="st">&quot;ER&quot;</span>, <span class="st">&quot;grade&quot;</span>, <span class="st">&quot;stage&quot;</span>, <span class="st">&quot;pik3ca&quot;</span>, <span class="st">&quot;p53&quot;</span>)],
        <span class="kw">c</span>(<span class="st">&quot;ER&quot;</span>, <span class="st">&quot;Grade&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;PIK3CA&quot;</span>, <span class="st">&quot;TP53&quot;</span>),
        <span class="op">~</span><span class="st"> </span>.x <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.05</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(.y)
)

<span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> <span class="kw">map</span>(plots_ppt1, <span class="op">~</span><span class="st"> </span>.x <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>, <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),        <span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.8</span>)), <span class="dt">nrow=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-58-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> <span class="kw">map</span>(plots_ppt2, <span class="op">~</span><span class="st"> </span>.x <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>, <span class="dt">strip.background =</span> <span class="kw">element_blank</span>(),        <span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.8</span>)), <span class="dt">nrow=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-58-2.png" width="672" /></p>
</div>
</div>
<div id="compare-epigenomic-scores-to-clinical-annotations" class="section level2">
<h2><span class="header-section-number">4.11</span> Compare epigenomic scores to clinical annotations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_feats &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/epigenomic_features.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ML =</span> <span class="op">-</span>ML, <span class="dt">clock =</span> <span class="op">-</span>clock, <span class="dt">immune.meth =</span> <span class="op">-</span>immune.meth, <span class="dt">caf.meth =</span> <span class="op">-</span>caf.meth) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy &lt;-<span class="st"> </span>all_feats <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">select</span>(<span class="op">-</span>caf, <span class="op">-</span>immune) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">rename</span>(<span class="dt">caf =</span> caf.meth, <span class="dt">immune =</span> immune.meth) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;score&quot;</span>, <span class="op">-</span>ER, <span class="op">-</span>samp) </code></pre></div>
<div id="figure-4c-extended-data-figure-10b" class="section level4">
<h4><span class="header-section-number">4.11.0.1</span> Figure 4c, Extended Data Figure 10b</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">10</span>)

p_mut &lt;-<span class="st"> </span><span class="kw">plot_score_feats_boxp</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">clin_feat =</span> MathScore), <span class="st">&quot;Mutational load (MATH score)</span>
<span class="st">&quot;</span>, feats_tidy, <span class="dt">nbins =</span> <span class="dv">5</span>)
p_gi &lt;-<span class="st"> </span><span class="kw">plot_score_feats_boxp</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">clin_feat =</span> giScore_width), <span class="st">&quot;Chromosomal Instability (CIN)&quot;</span>, feats_tidy, <span class="dt">nbins =</span> <span class="dv">5</span>) 

p_mut <span class="op">+</span><span class="st"> </span>p_gi </code></pre></div>
<pre><code>## Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
## use `guide = &quot;none&quot;` instead.

## Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
## use `guide = &quot;none&quot;` instead.</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<p>p.values</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">clin_feat =</span> MathScore)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(ER, feat) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">do</span>({broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>score, .<span class="op">$</span>clin_feat, <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="dt">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>))}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(p.value) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">signif =</span> <span class="kw">case_when</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.0001</span> <span class="op">~</span><span class="st"> &quot;****&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span> <span class="op">~</span><span class="st"> &quot;***&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span> <span class="op">~</span><span class="st"> &quot;**&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">~</span><span class="st"> &quot;*&quot;</span>))</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<pre><code>## Warning in cor.test.default(.$score, .$clin_feat, method = &quot;spearman&quot;, use =
## &quot;pairwise.complete.obs&quot;): Cannot compute exact p-value with ties

## Warning in cor.test.default(.$score, .$clin_feat, method = &quot;spearman&quot;, use =
## &quot;pairwise.complete.obs&quot;): Cannot compute exact p-value with ties

## Warning in cor.test.default(.$score, .$clin_feat, method = &quot;spearman&quot;, use =
## &quot;pairwise.complete.obs&quot;): Cannot compute exact p-value with ties

## Warning in cor.test.default(.$score, .$clin_feat, method = &quot;spearman&quot;, use =
## &quot;pairwise.complete.obs&quot;): Cannot compute exact p-value with ties

## Warning in cor.test.default(.$score, .$clin_feat, method = &quot;spearman&quot;, use =
## &quot;pairwise.complete.obs&quot;): Cannot compute exact p-value with ties</code></pre>
<pre><code>## # A tibble: 5 x 8
## # groups: ER, feat
##    ER   feat   estimate statistic       p.value                          method
## 1 ER+    caf -0.1344037 203009259 0.00001594158 Spearman&#39;s rank correlation rho
## 2 ER+     MG  0.1246595 156648143 0.00006338479 Spearman&#39;s rank correlation rho
## 3 ER+     ML  0.1224722 157039572 0.00008525637 Spearman&#39;s rank correlation rho
## 4 ER-    caf -0.2064473   3657784 0.00077389932 Spearman&#39;s rank correlation rho
## 5 ER- immune -0.1645390   3530724 0.00755496383 Spearman&#39;s rank correlation rho
##   alternative signif
## 1   two.sided   ****
## 2   two.sided   ****
## 3   two.sided   ****
## 4   two.sided    ***
## 5   two.sided     **</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">clin_feat =</span> giScore_width)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(ER, feat) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">na.omit</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">do</span>({broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>score, .<span class="op">$</span>clin_feat, <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="dt">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>))}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(p.value) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">signif =</span> <span class="kw">case_when</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.0001</span> <span class="op">~</span><span class="st"> &quot;****&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span> <span class="op">~</span><span class="st"> &quot;***&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span> <span class="op">~</span><span class="st"> &quot;**&quot;</span>, p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">~</span><span class="st"> &quot;*&quot;</span>))</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<pre><code>## # A tibble: 7 x 8
## # groups: ER, feat
##    ER feat   estimate statistic      p.value                          method
## 1 ER+  caf -0.2858945 238304890 7.387265e-21 Spearman&#39;s rank correlation rho
## 2 ER+   ML  0.2837975 132728278 1.497974e-20 Spearman&#39;s rank correlation rho
## 3 ER+   MG  0.2689927 135471936 1.636804e-18 Spearman&#39;s rank correlation rho
## 4 ER-  caf -0.2551031   3805302 3.005832e-05 Spearman&#39;s rank correlation rho
## 5 ER-   ML  0.1784401   2490858 3.734964e-03 Spearman&#39;s rank correlation rho
## 6 ER-   MG -0.1256257   3412744 4.183906e-02 Spearman&#39;s rank correlation rho
##   alternative signif
## 1   two.sided   ****
## 2   two.sided   ****
## 3   two.sided   ****
## 4   two.sided   ****
## 5   two.sided     **
## 6   two.sided      *
## # ... with 1 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">clin_feat =</span> giScore_width), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(ER, feat)</code></pre></div>
<pre><code>## # A tibble: 15 x 3
##    ER   feat    n
## 1 ER-    caf  310
## 2 ER-  clock  310
## 3 ER- immune  310
## 4 ER-     MG  310
## 5 ER-     ML  310
## 6 ER+    caf 1108
## # ... with 9 more rows</code></pre>
</div>
</div>
<div id="distribution-of-the-scores-in-ic10-integrative-clusters" class="section level2">
<h2><span class="header-section-number">4.12</span> Distribution of the scores in iC10 integrative clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy &lt;-<span class="st"> </span>all_feats <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;score&quot;</span>, <span class="op">-</span>samp, <span class="op">-</span>ER) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="kw">factor</span>(ER, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>)))
df_iC10 &lt;-<span class="st"> </span>samp_data <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">select</span>(samp, iC10) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">iC10 =</span> <span class="kw">factor</span>(iC10, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>iC10))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(feats_tidy, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>, <span class="op">!</span><span class="kw">is.na</span>(feat))
df_iC10_count &lt;-<span class="st"> </span>df_iC10 <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">group_by</span>(feat) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cut</span>(score, <span class="dt">breaks =</span> <span class="kw">quantile</span>(score, <span class="dv">0</span><span class="op">:</span><span class="dv">5</span> <span class="op">/</span><span class="st"> </span><span class="dv">5</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">include.lowest =</span> <span class="ot">TRUE</span>, <span class="dt">labels =</span> <span class="kw">as.character</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">count</span>(iC10, feat, score) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">group_by</span>(feat, score) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">p =</span> n <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(n))</code></pre></div>
<div id="extended-data-figure-10a" class="section level4">
<h4><span class="header-section-number">4.12.0.1</span> Extended Data Figure 10a</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">5</span>, <span class="dt">repr.plot.height =</span> <span class="dv">10</span>)
p_iC10_count &lt;-<span class="st"> </span>df_iC10_count <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">feat =</span> <span class="kw">factor</span>(feat, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;caf&quot;</span>, <span class="st">&quot;immune&quot;</span>, <span class="st">&quot;clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(feat)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> iC10, <span class="dt">y =</span> p, <span class="dt">x =</span> score, <span class="dt">label =</span> n)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_col</span>(<span class="dt">width =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>iC10) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_text</span>(<span class="dt">family =</span> <span class="st">&quot;Arial&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">position =</span> <span class="kw">position_stack</span>(<span class="dt">vjust =</span> <span class="fl">0.5</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">facet_grid</span>(feat <span class="op">~</span><span class="st"> </span>.) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;% of samples&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Score&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span>percent) <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.6</span>, <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.background =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.border =</span> <span class="kw">element_blank</span>())

p_iC10_count</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_ic10 &lt;-<span class="st"> </span>df_iC10 <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> iC10, <span class="dt">y =</span> score, <span class="dt">fill =</span> iC10)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">color =</span> <span class="st">&quot;darkgray&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_boxplot</span>(<span class="dt">lwd =</span> <span class="fl">0.2</span>, <span class="dt">outlier.size =</span> <span class="fl">0.2</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>iC10, <span class="dt">guide =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">facet_grid</span>(feat <span class="op">~</span><span class="st"> </span>., <span class="dt">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;iC10&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.5</span>)
p_ic10</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-67-1.png" width="672" /></p>
</div>
</div>
<div id="epipolymorphism-of-the-epigenomic-layers" class="section level2">
<h2><span class="header-section-number">4.13</span> Epipolymorphism of the epigenomic layers</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loci_epi_mean &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/loci_epipoly_mean.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>() </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gvtrack.create</span>(<span class="st">&quot;d_exon&quot;</span>, <span class="st">&quot;intervs.global.exon&quot;</span>, <span class="st">&quot;distance&quot;</span>)
<span class="kw">gvtrack.create</span>(<span class="st">&quot;d_intron&quot;</span>, <span class="st">&quot;intervs.global.introns&quot;</span>, <span class="st">&quot;distance&quot;</span>)
<span class="kw">gvtrack.create</span>(<span class="st">&quot;d_promoter&quot;</span>, promoter_intervs, <span class="st">&quot;distance&quot;</span>)
loci_annot &lt;-<span class="st"> </span><span class="kw">gextract.left_join</span>(<span class="kw">c</span>(<span class="st">&quot;d_promoter&quot;</span>, <span class="st">&quot;d_exon&quot;</span>, <span class="st">&quot;d_intron&quot;</span>), <span class="dt">intervals =</span> loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(chrom, start, end), <span class="dt">iterator =</span> loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(chrom, start, end)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">right_join</span>(loci_annot, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>))
loci_annot &lt;-<span class="st"> </span>loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">case_when</span>(
        d_promoter <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;promoter&quot;</span>,
        d_exon <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;exon&quot;</span>,
        d_intron <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">~</span><span class="st"> &quot;intron&quot;</span>, 
        <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &quot;inter-genic&quot;</span>        
    ))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plot_epipoly_strip &lt;-<span class="st"> </span><span class="cf">function</span>(df, df_bg, ER, <span class="dt">k=</span><span class="dv">15</span>){
    p_bg &lt;-<span class="st"> </span>df_bg <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> </span><span class="op">!!</span>ER) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">arrange</span>(pat_meth) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(epipoly)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">group_by</span>(type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">med =</span> zoo<span class="op">::</span><span class="kw">rollmedian</span>(epipoly, <span class="dt">k =</span> k, <span class="dt">na.pad =</span> <span class="ot">TRUE</span>)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> pat_meth, <span class="dt">y =</span> epipoly)) <span class="op">+</span><span class="st">        </span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.001</span>, <span class="dt">color=</span><span class="st">&quot;gray&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_line</span>(<span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="dt">data =</span> <span class="kw">tibble</span>(<span class="dt">m =</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">by =</span> <span class="fl">0.001</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">epipoly =</span> <span class="dv">2</span> <span class="op">*</span><span class="st"> </span>m <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>m)), <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="kw">aes</span>(<span class="dt">x =</span> m, <span class="dt">y =</span> epipoly)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_line</span>(<span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="dt">data =</span> <span class="kw">tibble</span>(<span class="dt">m =</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">by =</span> <span class="fl">0.001</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">epipoly =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>((<span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>m <span class="op">+</span><span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>m <span class="op">*</span><span class="st"> </span>m)<span class="op">^</span><span class="dv">5</span>)), <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="kw">aes</span>(<span class="dt">x =</span> m, <span class="dt">y =</span> epipoly)) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Avg. methylation&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Epi-polymorphism&quot;</span>)
    df &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> </span><span class="op">!!</span>ER) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">arrange</span>(pat_meth) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(epipoly)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">group_by</span>(type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">med =</span> zoo<span class="op">::</span><span class="kw">rollmedian</span>(epipoly, <span class="dt">k =</span> k, <span class="dt">na.pad =</span> <span class="ot">TRUE</span>))
    p &lt;-<span class="st"> </span>p_bg <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">data =</span> df, <span class="dt">size =</span> <span class="fl">0.001</span>, <span class="dt">color=</span><span class="st">&quot;darkred&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_line</span>(<span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">x =</span> pat_meth, <span class="dt">y =</span> med), <span class="dt">lwd =</span> <span class="fl">0.6</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df, <span class="dt">inherit.aes =</span> <span class="ot">FALSE</span>, <span class="kw">aes</span>(<span class="dt">x =</span> pat_meth, <span class="dt">y =</span> med), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">lwd =</span> <span class="fl">0.6</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) 
    p <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(.<span class="op">~</span>type)
}</code></pre></div>
<div id="extended-data-figure-7a" class="section level4">
<h4><span class="header-section-number">4.13.0.1</span> Extended Data Figure 7a</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">5</span>)
df &lt;-<span class="st"> </span>loci_epi_mean <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">inner_join</span>(loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(MG <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>))</code></pre></div>
<pre><code>## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_bg &lt;-<span class="st"> </span>loci_epi_mean <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">anti_join</span>(loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(MG <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(loci_annot)</code></pre></div>
<pre><code>## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)
## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_epipoly_MG &lt;-<span class="st"> </span><span class="kw">plot_epipoly_strip</span>(df, df_bg, <span class="st">&quot;ER+&quot;</span>, <span class="dt">k=</span><span class="dv">101</span>)  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;MG&quot;</span>)
p_epipoly_MG</code></pre></div>
<pre><code>## Warning: Removed 100 row(s) containing missing values (geom_path).

## Warning: Removed 100 row(s) containing missing values (geom_path).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-71-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">5</span>)
df &lt;-<span class="st"> </span>loci_epi_mean <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">inner_join</span>(loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ML <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>))</code></pre></div>
<pre><code>## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_bg &lt;-<span class="st"> </span>loci_epi_mean <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">anti_join</span>(loci_annot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ML <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(loci_annot)</code></pre></div>
<pre><code>## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)
## Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_epipoly_ML &lt;-<span class="st"> </span><span class="kw">plot_epipoly_strip</span>(df, df_bg, <span class="st">&quot;ER+&quot;</span>, <span class="dt">k=</span><span class="dv">101</span>)  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;ML&quot;</span>)
p_epipoly_ML</code></pre></div>
<pre><code>## Warning: Removed 100 row(s) containing missing values (geom_path).</code></pre>
<pre><code>## Warning: Removed 109 row(s) containing missing values (geom_path).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
</div>
</div>
<div id="comparison-of-the-epigenomic-scores-with-existing-epigenomic-age-metrics" class="section level2">
<h2><span class="header-section-number">4.14</span> Comparison of the epigenomic scores with existing epigenomic age metrics</h2>
<div id="phenoage" class="section level3">
<h3><span class="header-section-number">4.14.1</span> phenoAge</h3>
<p>Compare the epigenomic scores with phenoAge score from Steve Horvath's lab (PMID: 29676998)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg_450k &lt;-<span class="st"> </span><span class="kw">gintervals.load</span>(<span class="st">&quot;intervs.450k_27k.cpgs&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
pheno_age &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/phenoAge.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="dt">id =</span> CpG) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(cpg_450k, <span class="dt">by =</span> <span class="st">&quot;id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="op">-</span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrom, start, end, id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pheno_age_mb &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/pheno_age_score.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<p>Unfortunately, the METABRIC data doesn't cover all of <em>phenoAge</em> CpGs, but we have sufficient CpGs for correlating with the epigenomic scores:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">7</span>)
n_cpgs_pheno &lt;-<span class="st"> </span><span class="kw">colSums</span>(<span class="op">!</span><span class="kw">is.na</span>(pheno_age_mb <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>())) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;n_cpgs&quot;</span>)
n_cpgs_pheno <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n_cpgs, <span class="dt">y =</span> <span class="dv">1</span><span class="op">-</span>..y..)) <span class="op">+</span><span class="st"> </span><span class="kw">stat_ecdf</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;# of CpGs&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of samples&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">7</span>)
n_samples_pheno &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="op">!</span><span class="kw">is.na</span>(pheno_age_mb <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>())) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;cpg&quot;</span>, <span class="st">&quot;n_samples&quot;</span>)
n_samples_pheno <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n_samples, <span class="dt">y =</span> <span class="dv">1</span><span class="op">-</span>..y..)) <span class="op">+</span><span class="st"> </span><span class="kw">stat_ecdf</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;# of samples&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;fraction of CpGs&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)  <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_log10</span>()</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>## Warning: Removed 324 rows containing non-finite values (stat_ecdf).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pheno_age_df &lt;-<span class="st"> </span>pheno_age_mb <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">colMeans</span>(<span class="dt">na.rm =</span> <span class="ot">TRUE</span>)  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;pheno_age_score&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(n_cpgs_pheno) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(n_cpgs <span class="op">&gt;=</span><span class="st"> </span><span class="dv">50</span>)</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">8</span>)

df &lt;-<span class="st"> </span>all_feats <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;score&quot;</span>, clock<span class="op">:</span>immune.meth) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(pheno_age_df)</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_pheno_age &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>score, <span class="dt">y=</span>pheno_age_score, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(ER<span class="op">~</span>feat, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)

p_pheno_age <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning: Removed 5649 rows containing missing values (geom_point).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-78-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(feat) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">cor =</span> <span class="kw">cor</span>(pheno_age_score, score, <span class="dt">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">arrange</span>(cor) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as.data.frame</span>()</code></pre></div>
<pre><code>##          feat         cor
## 1    caf.meth -0.45571500
## 2          ML -0.28793733
## 3 immune.meth -0.24872108
## 4         caf -0.18080950
## 5       clock -0.06886096
## 6      immune  0.10135506
## 7          MG  0.56060188</code></pre>
<div id="ext-data-figure-5e" class="section level4">
<h4><span class="header-section-number">4.14.1.1</span> Ext Data Figure 5e</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">5</span>)
p_pheno_age_clock &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(feat <span class="op">==</span><span class="st"> &quot;clock&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="kw">factor</span>(ER, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>))) <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>score, <span class="dt">y=</span>pheno_age_score, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(.<span class="op">~</span>ER) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;PhenoAge&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Clock&quot;</span>)
p_pheno_age_clock <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning: Removed 807 rows containing missing values (geom_point).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-80-1.png" width="672" /></p>
</div>
<div id="ext-data-figure-5f" class="section level4">
<h4><span class="header-section-number">4.14.1.2</span> Ext Data Figure 5f</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">15</span>, <span class="dt">repr.plot.height =</span> <span class="dv">5</span>)
df &lt;-<span class="st"> </span>pheno_age_df <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, age)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">add_ER</span>()</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">cor =</span> <span class="kw">cor</span>(age, pheno_age_score, <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="dt">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>))</code></pre></div>
<pre><code>## # A tibble: 1 x 1
##          cor
## 1 0.06263803</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_age_pheno &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>age, <span class="dt">y=</span>pheno_age_score, <span class="dt">color=</span>ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">facet_grid</span>(.<span class="op">~</span>ER) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Biological age&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;phenoAge&quot;</span>)

p_age_pheno <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning: Removed 33 rows containing missing values (geom_point).</code></pre>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_mat &lt;-<span class="st"> </span>all_feats <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.matrix</span>()
pheno_mat &lt;-<span class="st"> </span><span class="kw">intervs_to_mat</span>(pheno_age_mb)
samples &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">colnames</span>(pheno_mat), <span class="kw">rownames</span>(feats_mat))
cor_pheno &lt;-<span class="st"> </span><span class="kw">tgs_cor</span>(feats_mat[samples, ], <span class="kw">t</span>(pheno_mat[, samples]), <span class="dt">pairwise.complete.obs=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mat_to_intervs</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
</div>
<div id="ext-data-figure-5d" class="section level4">
<h4><span class="header-section-number">4.14.1.3</span> Ext Data Figure 5d</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">4</span>, <span class="dt">repr.plot.height =</span> <span class="dv">4</span>)
cor_thresh &lt;-<span class="st"> </span><span class="fl">0.25</span>
cor_pheno &lt;-<span class="st"> </span>cor_pheno <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">type =</span> <span class="kw">case_when</span>(clock <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;clock&quot;</span>, MG <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;MG&quot;</span>, ML <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">~</span><span class="st"> &quot;ML&quot;</span>, <span class="kw">is.na</span>(MG) <span class="op">|</span><span class="st"> </span><span class="kw">is.na</span>(ML) <span class="op">|</span><span class="st"> </span><span class="kw">is.na</span>(clock) <span class="op">~</span><span class="st"> &quot;missing&quot;</span>, <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> &quot;other&quot;</span>))
p_pheno_cpgs &lt;-<span class="st"> </span>cor_pheno <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">count</span>(type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">reorder</span>(type, n), <span class="dt">y=</span>n)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_col</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;CpG type&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;phenoAge CpGs&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">vertical_labs</span>()
p_pheno_cpgs</code></pre></div>
<p><img src="Epigenomic-scores_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gc</span>()</code></pre></div>
<pre><code>##               used    (Mb)  gc trigger     (Mb)    max used     (Mb)
## Ncells     5190360   277.2    13921326    743.5    13921326    743.5
## Vcells 10668643577 81395.3 15420004920 117645.4 13689732018 104444.4</code></pre>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tumor-microenvironment-tme-deconvolution.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="loss-clock.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
