<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13 Export supplementary tables | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</title>
  <meta name="description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="13 Export supplementary tables | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="tanaylab.github.io/metabric_rrbs" />
  
  <meta property="og:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="github-repo" content="tanaylab/metabric_rrbs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13 Export supplementary tables | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  
  <meta name="twitter:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  

<meta name="author" content="Aviezer Lifshitz" />


<meta name="date" content="2021-08-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="finding-specific-dosage-compensation-examples.html"/>
<link rel="next" href="files-needed-for-the-analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#run-the-notebooks"><i class="fa fa-check"></i><b>1.1</b> Run the notebooks</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notebook-order"><i class="fa fa-check"></i><b>1.3</b> Notebook order</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#find-a-specific-figure"><i class="fa fa-check"></i><b>1.4</b> Find a specific figure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html"><i class="fa fa-check"></i><b>2</b> METABRIC methylation data statistics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#breakdown-of-samples"><i class="fa fa-check"></i><b>2.1</b> Breakdown of samples</a></li>
<li class="chapter" data-level="2.2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#number-of-samples"><i class="fa fa-check"></i><b>2.2</b> Number of samples</a></li>
<li class="chapter" data-level="2.3" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#total-number-of-reads"><i class="fa fa-check"></i><b>2.3</b> Total number of reads</a></li>
<li class="chapter" data-level="2.4" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-cpg-coverage-per-sample"><i class="fa fa-check"></i><b>2.4</b> Distribution of CpG coverage per sample</a></li>
<li class="chapter" data-level="2.5" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-covered-cpgs"><i class="fa fa-check"></i><b>2.5</b> Distribution of covered CpGs</a></li>
<li class="chapter" data-level="2.6" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#promoter-coverage"><i class="fa fa-check"></i><b>2.6</b> Promoter coverage</a></li>
<li class="chapter" data-level="2.7" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#coverage-per-clinical-annotations"><i class="fa fa-check"></i><b>2.7</b> Coverage per clinical annotations</a></li>
<li class="chapter" data-level="2.8" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage"><i class="fa fa-check"></i><b>2.8</b> Marginal CpG coverage</a><ul>
<li class="chapter" data-level="2.8.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage-of-k4me1-hotspots"><i class="fa fa-check"></i><b>2.8.1</b> Marginal CpG coverage of k4me1 hotspots</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#sample-cellularity"><i class="fa fa-check"></i><b>2.9</b> Sample cellularity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html"><i class="fa fa-check"></i><b>3</b> Tumor microenvironment (TME) deconvolution</a><ul>
<li class="chapter" data-level="3.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#initialize"><i class="fa fa-check"></i><b>3.2</b> Initialize</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#load-data"><i class="fa fa-check"></i><b>3.3</b> Load data</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#deconvolve-tme"><i class="fa fa-check"></i><b>3.4</b> Deconvolve TME</a><ul>
<li class="chapter" data-level="3.4.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er-samples"><i class="fa fa-check"></i><b>3.4.1</b> ER+ samples</a></li>
<li class="chapter" data-level="3.4.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er--samples"><i class="fa fa-check"></i><b>3.4.2</b> ER- samples</a></li>
<li class="chapter" data-level="3.4.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#normal-samples"><i class="fa fa-check"></i><b>3.4.3</b> Normal samples</a></li>
<li class="chapter" data-level="3.4.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-normalized-methylation"><i class="fa fa-check"></i><b>3.4.4</b> Merge normalized methylation</a></li>
<li class="chapter" data-level="3.4.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-tme-scores"><i class="fa fa-check"></i><b>3.4.5</b> Merge TME scores</a></li>
<li class="chapter" data-level="3.4.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#caf-and-immune-genes"><i class="fa fa-check"></i><b>3.4.6</b> CAF and Immune genes</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#expression-methylation-correlation-matrices"><i class="fa fa-check"></i><b>3.5</b> Expression-methylation correlation matrices</a><ul>
<li class="chapter" data-level="3.5.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#export-clustering"><i class="fa fa-check"></i><b>3.5.1</b> Export clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#diagnose-tme-normalization"><i class="fa fa-check"></i><b>3.6</b> Diagnose TME normalization</a></li>
<li class="chapter" data-level="3.7" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#plot-cd3d-and-cav1-vs-immune-and-caf-methylation-modules"><i class="fa fa-check"></i><b>3.7</b> Plot CD3D and CAV1 vs immune and CAF methylation modules</a></li>
<li class="chapter" data-level="3.8" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#annotate-tme-scores"><i class="fa fa-check"></i><b>3.8</b> Annotate TME scores</a></li>
<li class="chapter" data-level="3.9" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#appendix-sensitivity-analysis-of-tme-normalization"><i class="fa fa-check"></i><b>3.9</b> Appendix: Sensitivity analysis of TME normalization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html"><i class="fa fa-check"></i><b>4</b> Definition of Clock, MG and ML scores</a><ul>
<li class="chapter" data-level="4.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#initialize-1"><i class="fa fa-check"></i><b>4.2</b> Initialize</a></li>
<li class="chapter" data-level="4.3" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#load-data-1"><i class="fa fa-check"></i><b>4.3</b> Load data</a></li>
<li class="chapter" data-level="4.4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#clustering-of-normalized-methylation-of-er-tumors"><i class="fa fa-check"></i><b>4.4</b> Clustering of normalized methylation of ER+ tumors</a></li>
<li class="chapter" data-level="4.5" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-clustering-on-er--correlation-matrix"><i class="fa fa-check"></i><b>4.5</b> Project clustering on ER- correlation matrix</a></li>
<li class="chapter" data-level="4.6" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#classify-loci-in-the-genome-to-epigenomic-features"><i class="fa fa-check"></i><b>4.6</b> Classify loci in the genome to epigenomic features</a></li>
<li class="chapter" data-level="4.7" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#average-methylation-of-the-features"><i class="fa fa-check"></i><b>4.7</b> Average methylation of the features</a></li>
<li class="chapter" data-level="4.8" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-distribution-over-samples-of-the-correlated-loci"><i class="fa fa-check"></i><b>4.8</b> Plot distribution over samples of the correlated loci</a></li>
<li class="chapter" data-level="4.9" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-loci-classification"><i class="fa fa-check"></i><b>4.9</b> Plot loci classification</a></li>
<li class="chapter" data-level="4.10" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-all-epigenomic-features-using-umap"><i class="fa fa-check"></i><b>4.10</b> Project all epigenomic features using UMAP</a></li>
<li class="chapter" data-level="4.11" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#compare-epigenomic-scores-to-clinical-annotations"><i class="fa fa-check"></i><b>4.11</b> Compare epigenomic scores to clinical annotations</a></li>
<li class="chapter" data-level="4.12" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#distribution-of-the-scores-in-ic10-integrative-clusters"><i class="fa fa-check"></i><b>4.12</b> Distribution of the scores in iC10 integrative clusters</a></li>
<li class="chapter" data-level="4.13" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#epipolymorphism-of-the-epigenomic-layers"><i class="fa fa-check"></i><b>4.13</b> Epipolymorphism of the epigenomic layers</a></li>
<li class="chapter" data-level="4.14" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#comparison-of-the-epigenomic-scores-with-existing-epigenomic-age-metrics"><i class="fa fa-check"></i><b>4.14</b> Comparison of the epigenomic scores with existing epigenomic age metrics</a><ul>
<li class="chapter" data-level="4.14.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#phenoage"><i class="fa fa-check"></i><b>4.14.1</b> phenoAge</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="loss-clock.html"><a href="loss-clock.html"><i class="fa fa-check"></i><b>5</b> Loss clock</a><ul>
<li class="chapter" data-level="5.1" data-path="loss-clock.html"><a href="loss-clock.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="loss-clock.html"><a href="loss-clock.html#initialize-2"><i class="fa fa-check"></i><b>5.2</b> Initialize</a></li>
<li class="chapter" data-level="5.3" data-path="loss-clock.html"><a href="loss-clock.html#plot-methylation-distribution-of-the-clock"><i class="fa fa-check"></i><b>5.3</b> Plot methylation distribution of the clock</a></li>
<li class="chapter" data-level="5.4" data-path="loss-clock.html"><a href="loss-clock.html#annotate-clock-score"><i class="fa fa-check"></i><b>5.4</b> Annotate &quot;clock&quot; score</a><ul>
<li class="chapter" data-level="5.4.1" data-path="loss-clock.html"><a href="loss-clock.html#loci-annotation"><i class="fa fa-check"></i><b>5.4.1</b> Loci annotation</a></li>
<li class="chapter" data-level="5.4.2" data-path="loss-clock.html"><a href="loss-clock.html#clinical-annotation"><i class="fa fa-check"></i><b>5.4.2</b> Clinical annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="loss-clock.html"><a href="loss-clock.html#plot-chromosomal-traces-of-clock-score"><i class="fa fa-check"></i><b>5.5</b> Plot chromosomal traces of clock score</a></li>
<li class="chapter" data-level="5.6" data-path="loss-clock.html"><a href="loss-clock.html#correlation-of-gene-expression-to-clock"><i class="fa fa-check"></i><b>5.6</b> Correlation of gene expression to clock</a></li>
<li class="chapter" data-level="5.7" data-path="loss-clock.html"><a href="loss-clock.html#plot-score-in-different-tor-regimes"><i class="fa fa-check"></i><b>5.7</b> Plot score in different TOR regimes</a></li>
<li class="chapter" data-level="5.8" data-path="loss-clock.html"><a href="loss-clock.html#plot-high-low-clock-tumors-vs-tor"><i class="fa fa-check"></i><b>5.8</b> Plot High-Low clock tumors vs TOR</a></li>
<li class="chapter" data-level="5.9" data-path="loss-clock.html"><a href="loss-clock.html#plot-correlation-of-loci-to-clock-methylation"><i class="fa fa-check"></i><b>5.9</b> Plot correlation of loci to clock methylation</a></li>
<li class="chapter" data-level="5.10" data-path="loss-clock.html"><a href="loss-clock.html#plot-distribution-of-methylation-of-clock-in-normal-samples"><i class="fa fa-check"></i><b>5.10</b> Plot distribution of methylation of clock in normal samples</a></li>
<li class="chapter" data-level="5.11" data-path="loss-clock.html"><a href="loss-clock.html#compare-clock-layer-to-biological-age"><i class="fa fa-check"></i><b>5.11</b> Compare clock layer to biological age</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html"><i class="fa fa-check"></i><b>6</b> Epigenomic instability</a><ul>
<li class="chapter" data-level="6.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#compare-features-to-each-other"><i class="fa fa-check"></i><b>6.1</b> Compare features to each other</a></li>
<li class="chapter" data-level="6.2" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#annotate-epignomic-instability-scores"><i class="fa fa-check"></i><b>6.2</b> Annotate epignomic instability scores</a><ul>
<li class="chapter" data-level="6.2.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#integrative-clusters"><i class="fa fa-check"></i><b>6.2.1</b> Integrative clusters</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#cross-correlation-methylation-expression-of-epigenomic-instability-loci"><i class="fa fa-check"></i><b>6.3</b> Cross correlation (methylation-expression) of epigenomic instability loci</a></li>
<li class="chapter" data-level="6.4" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#epigenomic-instability-in-different-time-of-replication-tor-regimes"><i class="fa fa-check"></i><b>6.4</b> Epigenomic instability in different Time of replication (TOR) regimes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cna.html"><a href="cna.html"><i class="fa fa-check"></i><b>7</b> CNA</a></li>
<li class="chapter" data-level="8" data-path="mutations.html"><a href="mutations.html"><i class="fa fa-check"></i><b>8</b> Mutations</a></li>
<li class="chapter" data-level="9" data-path="survival.html"><a href="survival.html"><i class="fa fa-check"></i><b>9</b> Survival</a><ul>
<li class="chapter" data-level="9.0.1" data-path="survival.html"><a href="survival.html#raw-methylation"><i class="fa fa-check"></i><b>9.0.1</b> Raw-methylation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html"><i class="fa fa-check"></i><b>10</b> Methylation - Expression <em>in cis</em></a><ul>
<li class="chapter" data-level="10.0.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#remove-tme-genes"><i class="fa fa-check"></i><b>10.0.1</b> Remove TME genes</a></li>
<li class="chapter" data-level="10.0.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#load-normalized-methylation-and-separate-it-to-promoters-and-non-promoters"><i class="fa fa-check"></i><b>10.0.2</b> Load normalized methylation and separate it to promoters and non-promoters</a></li>
<li class="chapter" data-level="10.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-promoters"><i class="fa fa-check"></i><b>10.1</b> Cis candidates: promoters</a><ul>
<li class="chapter" data-level="10.1.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-correlation-of-top-cis-regulated-promoter-candidates"><i class="fa fa-check"></i><b>10.1.1</b> Plot correlation of top cis-regulated promoter candidates</a></li>
<li class="chapter" data-level="10.1.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#annotation-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.2</b> Annotation of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#gene-expression-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.3</b> Gene expression of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#correlation-between-cis-regulated-genes"><i class="fa fa-check"></i><b>10.1.4</b> Correlation between cis-regulated genes</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.2</b> Epipolymorphism of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#promoter-examples"><i class="fa fa-check"></i><b>10.3</b> Promoter examples</a><ul>
<li class="chapter" data-level="10.3.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#brca1"><i class="fa fa-check"></i><b>10.3.1</b> BRCA1</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-genomic"><i class="fa fa-check"></i><b>10.4</b> Cis candidates: genomic</a><ul>
<li class="chapter" data-level="10.4.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-cis"><i class="fa fa-check"></i><b>10.4.1</b> Plot % cis</a></li>
<li class="chapter" data-level="10.4.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-distance-to-most-correlated-gene"><i class="fa fa-check"></i><b>10.4.2</b> Plot distance to most correlated gene</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-genomic"><i class="fa fa-check"></i><b>10.5</b> Epipolymorphism of cis-regulated genomic</a></li>
<li class="chapter" data-level="10.6" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#examples-for-cis-regulation-non-promoters"><i class="fa fa-check"></i><b>10.6</b> Examples for cis regulation (non-promoters)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html"><i class="fa fa-check"></i><b>11</b> X dosage compensation</a><ul>
<li class="chapter" data-level="11.1" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#x-analysis"><i class="fa fa-check"></i><b>11.1</b> X analysis</a></li>
<li class="chapter" data-level="11.2" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#autosome-analysis"><i class="fa fa-check"></i><b>11.2</b> autosome analysis</a></li>
<li class="chapter" data-level="11.3" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#checking-dosage-compensation-on-all-x-loci"><i class="fa fa-check"></i><b>11.3</b> Checking dosage compensation on all X loci</a></li>
<li class="chapter" data-level="11.4" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#of-x-lost"><i class="fa fa-check"></i><b>11.4</b> % of X lost</a></li>
<li class="chapter" data-level="11.5" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#plot-expression-methylation-correlation-for-x"><i class="fa fa-check"></i><b>11.5</b> Plot Expression-Methylation correlation for X</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="finding-specific-dosage-compensation-examples.html"><a href="finding-specific-dosage-compensation-examples.html"><i class="fa fa-check"></i><b>12</b> Finding specific dosage compensation examples</a></li>
<li class="chapter" data-level="13" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html"><i class="fa fa-check"></i><b>13</b> Export supplementary tables</a><ul>
<li class="chapter" data-level="13.1" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-1-metabric-samples-metadata"><i class="fa fa-check"></i><b>13.1</b> Table 1: METABRIC samples metadata</a></li>
<li class="chapter" data-level="13.2" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-2-immune-expression-signature"><i class="fa fa-check"></i><b>13.2</b> Table 2: Immune expression signature</a></li>
<li class="chapter" data-level="13.3" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-3-caf-expression-signature"><i class="fa fa-check"></i><b>13.3</b> Table 3: CAF expression signature</a></li>
<li class="chapter" data-level="13.4" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-4-expression-methylation-correlation"><i class="fa fa-check"></i><b>13.4</b> Table 4: Expression-Methylation correlation</a></li>
<li class="chapter" data-level="13.5" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-5-methylation-scores"><i class="fa fa-check"></i><b>13.5</b> Table 5: Methylation scores</a></li>
<li class="chapter" data-level="13.6" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-6-gene-expression-correlation-to-epigenomic-instability"><i class="fa fa-check"></i><b>13.6</b> Table 6: Gene expression correlation to epigenomic instability</a></li>
<li class="chapter" data-level="13.7" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-7-methylation-layers-loci"><i class="fa fa-check"></i><b>13.7</b> Table 7: Methylation layers loci</a></li>
<li class="chapter" data-level="13.8" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-8-cis-regulation-candidates"><i class="fa fa-check"></i><b>13.8</b> Table 8: Cis regulation candidates</a></li>
<li class="chapter" data-level="13.9" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-9-dosage-compensation-in-autosomes"><i class="fa fa-check"></i><b>13.9</b> Table 9: Dosage compensation in autosomes</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html"><i class="fa fa-check"></i><b>14</b> Files needed for the analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#misha-db"><i class="fa fa-check"></i><b>14.1</b> Misha DB</a></li>
<li class="chapter" data-level="14.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#additional-files"><i class="fa fa-check"></i><b>14.2</b> Additional files</a><ul>
<li class="chapter" data-level="14.2.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#metabric-data"><i class="fa fa-check"></i><b>14.2.1</b> METABRIC data</a></li>
<li class="chapter" data-level="14.2.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#notable-generated-files"><i class="fa fa-check"></i><b>14.2.2</b> Notable generated files</a></li>
<li class="chapter" data-level="14.2.3" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#spatial-methylation-distribution"><i class="fa fa-check"></i><b>14.2.3</b> Spatial methylation distribution</a></li>
<li class="chapter" data-level="14.2.4" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#copy-number-abberations"><i class="fa fa-check"></i><b>14.2.4</b> Copy number abberations</a></li>
<li class="chapter" data-level="14.2.5" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#genomic-annotations"><i class="fa fa-check"></i><b>14.2.5</b> Genomic annotations</a></li>
<li class="chapter" data-level="14.2.6" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#external-resources"><i class="fa fa-check"></i><b>14.2.6</b> External resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figures-key.html"><a href="figures-key.html"><i class="fa fa-check"></i><b>15</b> Figures key</a><ul>
<li class="chapter" data-level="15.1" data-path="figures-key.html"><a href="figures-key.html#figure-1"><i class="fa fa-check"></i><b>15.1</b> Figure 1</a></li>
<li class="chapter" data-level="15.2" data-path="figures-key.html"><a href="figures-key.html#figure-3"><i class="fa fa-check"></i><b>15.2</b> Figure 3</a></li>
<li class="chapter" data-level="15.3" data-path="figures-key.html"><a href="figures-key.html#figure-4"><i class="fa fa-check"></i><b>15.3</b> Figure 4</a></li>
<li class="chapter" data-level="15.4" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-1"><i class="fa fa-check"></i><b>15.4</b> Extended Data Figure 1</a></li>
<li class="chapter" data-level="15.5" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-2"><i class="fa fa-check"></i><b>15.5</b> Extended Data Figure 2</a></li>
<li class="chapter" data-level="15.6" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-3"><i class="fa fa-check"></i><b>15.6</b> Extended Data Figure 3</a></li>
<li class="chapter" data-level="15.7" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-4"><i class="fa fa-check"></i><b>15.7</b> Extended Data Figure 4</a></li>
<li class="chapter" data-level="15.8" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-5"><i class="fa fa-check"></i><b>15.8</b> Extended Data Figure 5</a></li>
<li class="chapter" data-level="15.9" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-6"><i class="fa fa-check"></i><b>15.9</b> Extended Data Figure 6</a></li>
<li class="chapter" data-level="15.10" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-7"><i class="fa fa-check"></i><b>15.10</b> Extended Data Figure 7</a></li>
<li class="chapter" data-level="15.11" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-8"><i class="fa fa-check"></i><b>15.11</b> Extended Data Figure 8</a></li>
<li class="chapter" data-level="15.12" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-9"><i class="fa fa-check"></i><b>15.12</b> Extended Data Figure 9</a></li>
<li class="chapter" data-level="15.13" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-10"><i class="fa fa-check"></i><b>15.13</b> Extended Data Figure 10</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="export-supplementary-tables" class="section level1">
<h1><span class="header-section-number">13</span> Export supplementary tables</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;scripts/init.R&quot;</span>))</code></pre></div>
<div id="table-1-metabric-samples-metadata" class="section level2">
<h2><span class="header-section-number">13.1</span> Table 1: METABRIC samples metadata</h2>
<p>METABRIC samples profiled in this study.</p>
<p>Metadata fields are:</p>
<ul>
<li>samp</li>
<li>patient</li>
<li>batch</li>
<li>track</li>
<li>total_reads</li>
<li>mapped_reads</li>
<li>mapped_frac</li>
<li>cg_num</li>
<li>meth_calls</li>
<li>global_avg_meth</li>
<li>type</li>
<li>age</li>
<li>grade</li>
<li>stage</li>
<li>ER</li>
<li>IHC</li>
<li>iC10</li>
<li>PAM50</li>
<li>matched_normal</li>
<li>matched_tumor</li>
<li>digpath_lymph</li>
<li>digpath_stromal</li>
<li>IMC_Fibroblasts</li>
<li>IMC_Lymphocytes</li>
<li>ASCAT_cellularity</li>
<li>ASCAT_ploidy</li>
<li>MathScore</li>
<li>log10_global_epm</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_qc &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/sample_qc.csv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span>samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(samp_qc) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, patient, batch, track, total_reads, mapped_reads, mapped_frac, cg_num, meth_calls, global_avg_meth, type, age, grade, stage, ER, IHC, iC10, PAM50, matched_normal, matched_tumor, digpath_lymph, digpath_stromal, IMC_Fibroblasts, IMC_Lymphocytes, ASCAT_cellularity, MathScore, log10_global_epm)</code></pre></div>
<pre><code>## Joining, by = &quot;track&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(tab, <span class="kw">here</span>(<span class="st">&quot;export/S1 - Sample Information.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-2-immune-expression-signature" class="section level2">
<h2><span class="header-section-number">13.2</span> Table 2: Immune expression signature</h2>
<p>Table of genes that were used to define the Immune expression signature, separated by ER status.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">immune_genes_tab &lt;-<span class="st"> </span><span class="kw">map2_dfr</span>(<span class="kw">c</span>(<span class="st">&quot;ER_positive&quot;</span>, <span class="st">&quot;ER_negative&quot;</span>, <span class="st">&quot;normal&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>), <span class="op">~</span>
<span class="st">        </span><span class="kw">tibble</span>(<span class="dt">gene =</span> <span class="kw">get_TME_genes</span>(readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="kw">glue</span>(<span class="st">&quot;data/{.x}_norm_meth.rds&quot;</span>)))<span class="op">$</span>em_cross_clust, <span class="dt">caf_gene =</span> <span class="ot">NULL</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> .y) ) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/immune_genes_by_er.tsv&quot;</span>)

immune_genes_tab <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(ER)</code></pre></div>
<pre><code>##       ER   n
## 1    ER- 345
## 2    ER+ 195
## 3 normal 864</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(immune_genes_tab)</code></pre></div>
<pre><code>##      gene  ER
## 1   ACAP1 ER+
## 2     ADA ER+
## 3   ADAM7 ER+
## 4    AIM2 ER+
## 5    AOAH ER+
## 6 APBB1IP ER+</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(immune_genes_tab, <span class="kw">here</span>(<span class="st">&quot;export/S2 - Immune expression signature.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-3-caf-expression-signature" class="section level2">
<h2><span class="header-section-number">13.3</span> Table 3: CAF expression signature</h2>
<p>Table of genes that were used to define the CAF expression signature, separated by ER status.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">caf_genes_tab &lt;-<span class="st"> </span><span class="kw">map2_dfr</span>(<span class="kw">c</span>(<span class="st">&quot;ER_positive&quot;</span>, <span class="st">&quot;ER_negative&quot;</span>, <span class="st">&quot;normal&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>), <span class="op">~</span>
<span class="st">        </span><span class="kw">tibble</span>(<span class="dt">gene =</span> <span class="kw">get_TME_genes</span>(readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="kw">glue</span>(<span class="st">&quot;data/{.x}_norm_meth.rds&quot;</span>)))<span class="op">$</span>em_cross_clust, <span class="dt">immune_gene =</span> <span class="ot">NULL</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> .y) ) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/caf_genes_by_er.tsv&quot;</span>)

caf_genes_tab <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(ER)</code></pre></div>
<pre><code>##       ER   n
## 1    ER- 360
## 2    ER+ 207
## 3 normal 592</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(caf_genes_tab)</code></pre></div>
<pre><code>##      gene  ER
## 1   ABCA6 ER+
## 2   ABCA8 ER+
## 3   ABCB1 ER+
## 4  ACVRL1 ER+
## 5 ADAMTS9 ER+
## 6 ALDH1A2 ER+</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(caf_genes_tab, <span class="kw">here</span>(<span class="st">&quot;export/S3 - CAF expression signature.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-4-expression-methylation-correlation" class="section level2">
<h2><span class="header-section-number">13.4</span> Table 4: Expression-Methylation correlation</h2>
<p>Expression-Methylation correlation tables. Shown are pairs of tables of the following datasets: 1. Raw promoter methylation (Extended Data Fig. 2a,b). 2. Immune-CAF normalized methylation for loci that were correlated with MG/ML epigenomic instability (Fig. 2f). 3. X chromosome immune-CAF normalized methylation (Extended Data Fig 9a).</p>
<p>Each dataset is represented by an expression table, showing the mean correlation of each one of 30/32 methylation clusters in each gene, and a methylation table showing the mean correlation of every locus to 30/32 expression clusters.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">em_raw &lt;-<span class="st"> </span><span class="kw">list</span>(
        <span class="st">`</span><span class="dt">ER+</span><span class="st">`</span> =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="st">&quot;data/ER_positive_norm_meth.rds&quot;</span>))<span class="op">$</span>em_cross_clust,
        <span class="st">`</span><span class="dt">ER-</span><span class="st">`</span> =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="st">&quot;data/ER_negative_norm_meth.rds&quot;</span>))<span class="op">$</span>em_cross_clust,
        <span class="st">`</span><span class="dt">normal</span><span class="st">`</span> =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="st">&quot;data/normal_norm_meth.rds&quot;</span>))<span class="op">$</span>em_cross_clust
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">em_raw &lt;-<span class="st"> </span><span class="kw">map</span>(em_raw, parse_em_cors)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">norm_em &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="st">&quot;data/MG_ML_em_cross_cor_clust.rds&quot;</span>))
norm_em &lt;-<span class="st"> </span><span class="kw">parse_em_cors</span>(norm_em)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x_em &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_rds</span>(<span class="kw">here</span>(<span class="st">&quot;data/X_er_positive_em_cross_cor_clust.rds&quot;</span>))
x_em &lt;-<span class="st"> </span><span class="kw">parse_em_cors</span>(x_em)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sheets &lt;-<span class="st"> </span><span class="kw">list</span>(
    <span class="st">&quot;Raw promoters ER+ (Expression)&quot;</span> =<span class="st"> </span>em_raw[[<span class="st">&quot;ER+&quot;</span>]]<span class="op">$</span>expr_tab, 
    <span class="st">&quot;Raw promoters ER+ (Methylation)&quot;</span> =<span class="st"> </span>em_raw[[<span class="st">&quot;ER+&quot;</span>]]<span class="op">$</span>meth_tab,
    <span class="st">&quot;Raw promoters ER- (Expression)&quot;</span> =<span class="st"> </span>em_raw[[<span class="st">&quot;ER-&quot;</span>]]<span class="op">$</span>expr_tab, 
    <span class="st">&quot;Raw promoters ER- (Methylation)&quot;</span> =<span class="st"> </span>em_raw[[<span class="st">&quot;ER-&quot;</span>]]<span class="op">$</span>meth_tab, 
    <span class="st">&quot;MG,ML loci ER+ (Expression)&quot;</span> =<span class="st"> </span>norm_em<span class="op">$</span>expr_tab,
    <span class="st">&quot;MG,ML loci ER+ (Methylation)&quot;</span> =<span class="st"> </span>norm_em<span class="op">$</span>meth_tab,
    <span class="st">&quot;X chromosome ER+ (Expression)&quot;</span> =<span class="st"> </span>x_em<span class="op">$</span>expr_tab, 
    <span class="st">&quot;X chromosome ER+ (Methylation)&quot;</span> =<span class="st"> </span>x_em<span class="op">$</span>meth_tab
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(sheets, <span class="kw">here</span>(<span class="st">&quot;export/S4 - Expression-methylation correlations.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-5-methylation-scores" class="section level2">
<h2><span class="header-section-number">13.5</span> Table 5: Methylation scores</h2>
<p>Table with CAF, Immune, Clock, MG and ML methylation scores per METABRIC sample.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span><span class="kw">get_all_features</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(tab, <span class="kw">here</span>(<span class="st">&quot;export/S5 - Methylation scores.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-6-gene-expression-correlation-to-epigenomic-instability" class="section level2">
<h2><span class="header-section-number">13.6</span> Table 6: Gene expression correlation to epigenomic instability</h2>
<p>Tables of genes that have an absolute expression correlation higher than 0.3 to MG/ML/Clock scores, together with their respective correlations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feat_gene_cors &lt;-<span class="st"> </span><span class="kw">get_expression_features_cors</span>()
cor_thresh &lt;-<span class="st"> </span><span class="fl">0.3</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span>feat_gene_cors <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">abs</span>(MG) <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">|</span><span class="st"> </span><span class="kw">abs</span>(ML) <span class="op">&gt;=</span><span class="st"> </span>cor_thresh <span class="op">|</span><span class="st"> </span><span class="kw">abs</span>(clock) <span class="op">&gt;</span><span class="st"> </span>cor_thresh) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER =</span> <span class="kw">factor</span>(ER, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(name, ER,  <span class="dt">clock.cor =</span> clock, <span class="dt">MG.cor =</span> MG, <span class="dt">ML.cor =</span> ML, <span class="dt">caf.cor =</span> caf, <span class="dt">immune.cor =</span> immune) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(ER, clock.cor, MG.cor, ML.cor)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(tab, <span class="kw">here</span>(<span class="st">&quot;export/S6 - Gene expression correlation to methylation scores.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-7-methylation-layers-loci" class="section level2">
<h2><span class="header-section-number">13.7</span> Table 7: Methylation layers loci</h2>
<p>Coordinates of loci that were used in order to calculate the CAF, Immune, Clock, MG and ML scores.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clust_df &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/ER_positive_loci_clust.tsv&quot;</span>) ) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span>clust_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">layer =</span> clust) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(layer <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ML&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;clock&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(layer, chrom, start, end) 
<span class="kw">head</span>(tab)</code></pre></div>
<pre><code>## # A tibble: 6 x 4
##   chrom   start     end layer
## 1  chr1  134998  135215 clock
## 2  chr1  837884  838076 clock
## 3  chr1  907866  908002 clock
## 4  chr1  996514  996647 clock
## 5  chr1 1086481 1086732 clock
## 6  chr1 1381236 1381347 clock</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(tab, <span class="kw">here</span>(<span class="st">&quot;export/S7 - Methylation layers loci.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-8-cis-regulation-candidates" class="section level2">
<h2><span class="header-section-number">13.8</span> Table 8: Cis regulation candidates</h2>
<p>Pairs of loci and genes that are candidates for cis regulation in different FDR thresholds. Promoters and genomic loci are shown in separate tables.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">min_dist &lt;-<span class="st"> </span><span class="fl">5e5</span>
min_tss_dist &lt;-<span class="st"> </span><span class="dv">200</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genomic_cis_cands &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(
    <span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/genomic_cis_cands_ER_positive.tsv&quot;</span>)),
    <span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/genomic_cis_cands_ER_negative.tsv&quot;</span>)),
    <span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/genomic_cis_cands_normal.tsv&quot;</span>)))
<span class="kw">head</span>(genomic_cis_cands)</code></pre></div>
<pre><code>##   chrom  start    end type rank   gene        cor chrom_expr start_expr
## 1  chr1  10496  10587  obs    1 CT45A3 -0.2446525       chrX  134883487
## 2  chr1  10588  10639  obs    1  DSCR8 -0.1658337      chr21   39493544
## 3  chr1 134998 135215  obs    1 MAGEC2 -0.2381125       chrX  141293076
## 4  chr1 546168 546310  obs    1 MAGEA8 -0.2542376       chrX  149009940
## 5  chr1 565396 565791  obs    1 RAD51C -0.2153869      chr17   56769933
## 6  chr1 567121 567237  obs    1 TIMM23 -0.2271249      chr10   51623386
##    end_expr strand_expr dist n_obs n_shuff        fdr  ER
## 1 134883488           1   NA  2680      92 0.03432836 ER+
## 2  39493545           1   NA  2680      92 0.03432836 ER+
## 3 141293077          -1   NA  2680      92 0.03432836 ER+
## 4 149009941           1   NA  2680      92 0.03432836 ER+
## 5  56769934           1   NA  2680      92 0.03432836 ER+
## 6  51623387          -1   NA  2680      92 0.03432836 ER+</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(genomic_cis_cands)</code></pre></div>
<pre><code>## [1] 55616700       16</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cands_genomic &lt;-<span class="st"> </span>genomic_cis_cands <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;obs&quot;</span>, <span class="op">!</span><span class="kw">is.na</span>(dist), <span class="kw">abs</span>(dist) <span class="op">&lt;=</span><span class="st"> </span>min_dist, <span class="kw">abs</span>(dist) <span class="op">&gt;=</span><span class="st"> </span>min_tss_dist) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">arrange</span>(cor) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;scripts/init.R&quot;</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genomic_list &lt;-<span class="st"> </span><span class="kw">list</span>(
    <span class="st">`</span><span class="dt">Genomic (best correlation)</span><span class="st">`</span> =<span class="st"> </span>cands_genomic <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(rank <span class="op">==</span><span class="st"> </span><span class="dv">1</span>),
    <span class="st">`</span><span class="dt">Genomic (FDR &lt; 0.05)</span><span class="st">`</span> =<span class="st"> </span>cands_genomic <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(fdr <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>), 
    <span class="st">`</span><span class="dt">Genomic (FDR &lt; 0.1)</span><span class="st">`</span> =<span class="st"> </span>cands_genomic <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(fdr <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>)
)

genomic_list &lt;-<span class="st"> </span><span class="kw">map</span>(genomic_list, <span class="op">~</span><span class="st"> </span><span class="kw">annotate_cis_cands</span>(.x, <span class="dt">sigma_meth =</span> <span class="dv">2</span>, <span class="dt">sigma_expr =</span> <span class="dv">2</span>, <span class="dt">meth_diff =</span> <span class="fl">0.2</span>, <span class="dt">expr_diff =</span> <span class="dv">1</span>))</code></pre></div>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genomic_list &lt;-<span class="st"> </span><span class="kw">map</span>(genomic_list, <span class="op">~</span><span class="st"> </span>.x <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(gene, 
           ER, 
           chrom, 
           start, 
           end,            
           <span class="dt">distance =</span> dist, 
           fdr, 
           rank, 
           cor, 
           mean_meth, 
           sd_meth, 
           mean_expr, 
           sd_expr, 
           normal_meth, 
           normal_meth_sd, 
           normal_expr, 
           normal_expr_sd, 
           n_hypometh, 
           n_induced, 
           n_hypermeth, 
           n_repressed, 
           N_considered, 
           n_hypometh_vs_normal,
           n_stable_vs_normal, 
           n_hypermeth_vs_normal, 
           n_repressed_vs_normal, 
           n_stable_expr_vs_normal, 
           n_induced_vs_normal) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                    </span><span class="kw">arrange</span>(cor) 
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cands_prom &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/promoter_cis_cands.tsv&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>() </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prom_list &lt;-<span class="st"> </span><span class="kw">list</span>(
    <span class="st">`</span><span class="dt">Promoters (best correlation)</span><span class="st">`</span> =<span class="st"> </span>cands_prom <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(r <span class="op">==</span><span class="st"> </span><span class="dv">1</span>),
    <span class="st">`</span><span class="dt">Promoters (FDR &lt; 0.005)</span><span class="st">`</span> =<span class="st"> </span>cands_prom <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(fdr <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.005</span>), 
    <span class="st">`</span><span class="dt">Promoters (FDR &lt; 0.01)</span><span class="st">`</span> =<span class="st"> </span>cands_prom <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(fdr <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>),
    <span class="st">`</span><span class="dt">Promoters (FDR &lt; 0.05)</span><span class="st">`</span> =<span class="st"> </span>cands_prom <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(fdr <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prom_list &lt;-<span class="st"> </span><span class="kw">map</span>(prom_list, <span class="op">~</span><span class="st"> </span><span class="kw">annotate_cis_cands</span>(.x <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rename</span>(<span class="dt">gene =</span> name), <span class="dt">sigma_meth =</span> <span class="dv">2</span>, <span class="dt">sigma_expr =</span> <span class="dv">2</span>, <span class="dt">meth_diff =</span> <span class="fl">0.2</span>, <span class="dt">expr_diff =</span> <span class="dv">1</span>))</code></pre></div>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;</code></pre>
<pre><code>## Joining, by = &quot;coord&quot;
## Joining, by = &quot;coord&quot;</code></pre>
<pre><code>## Joining, by = &quot;gene&quot;
## Joining, by = &quot;gene&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prom_list &lt;-<span class="st"> </span><span class="kw">map</span>(prom_list, <span class="op">~</span><span class="st"> </span>
<span class="st">                </span>.x <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">diff =</span> <span class="kw">abs</span>(best <span class="op">-</span><span class="st"> </span>kth)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">select</span>(gene, ER, chrom, start, end, fdr, <span class="dt">rank =</span> r, cor, <span class="dt">diff_from_second_best =</span> diff, mean_meth, sd_meth, mean_expr, sd_expr, normal_meth, normal_meth_sd, normal_expr, normal_expr_sd, n_hypometh, n_induced, n_hypermeth, n_repressed, N_considered, n_hypometh_vs_normal, n_stable_vs_normal, n_hypermeth_vs_normal, n_repressed_vs_normal, n_stable_expr_vs_normal, n_induced_vs_normal) <span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">arrange</span>(cor) )</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cis_cands &lt;-<span class="st"> </span><span class="kw">c</span>(prom_list, genomic_list)    </code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">writexl<span class="op">::</span><span class="kw">write_xlsx</span>(<span class="dt">x =</span> cis_cands, <span class="dt">path =</span> <span class="kw">here</span>(<span class="st">&quot;export/S8 - Cis Regulation Candidates.xlsx&quot;</span>))</code></pre></div>
</div>
<div id="table-9-dosage-compensation-in-autosomes" class="section level2">
<h2><span class="header-section-number">13.9</span> Table 9: Dosage compensation in autosomes</h2>
<p>Table of autosome promoters that show increase of at least 10% in methylation when amplified to at least 3N vs 2N (‘Gain 3N’ table), at least 4N vs 2N (‘Amplification 4N’). The table ‘Loss’ shows loci that showed decreased methylation of at least 10% when losing a copy (1N vs 2N).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gen_dosage_comp_excel</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gc</span>()</code></pre></div>
<pre><code>##              used    (Mb) gc trigger    (Mb)   max used    (Mb)
## Ncells    4173577   222.9    8029078   428.8    8029078   428.8
## Vcells 2635840254 20109.9 5129628008 39136.0 5101547905 38921.8</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="finding-specific-dosage-compensation-examples.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="files-needed-for-the-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
