<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 METABRIC methylation data statistics | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</title>
  <meta name="description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="2 METABRIC methylation data statistics | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="tanaylab.github.io/metabric_rrbs" />
  
  <meta property="og:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="github-repo" content="tanaylab/metabric_rrbs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 METABRIC methylation data statistics | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  
  <meta name="twitter:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  

<meta name="author" content="Aviezer Lifshitz" />


<meta name="date" content="2021-08-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="tumor-microenvironment-tme-deconvolution.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#run-the-notebooks"><i class="fa fa-check"></i><b>1.1</b> Run the notebooks</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notebook-order"><i class="fa fa-check"></i><b>1.3</b> Notebook order</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#find-a-specific-figure"><i class="fa fa-check"></i><b>1.4</b> Find a specific figure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html"><i class="fa fa-check"></i><b>2</b> METABRIC methylation data statistics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#breakdown-of-samples"><i class="fa fa-check"></i><b>2.1</b> Breakdown of samples</a></li>
<li class="chapter" data-level="2.2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#number-of-samples"><i class="fa fa-check"></i><b>2.2</b> Number of samples</a></li>
<li class="chapter" data-level="2.3" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#total-number-of-reads"><i class="fa fa-check"></i><b>2.3</b> Total number of reads</a></li>
<li class="chapter" data-level="2.4" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-cpg-coverage-per-sample"><i class="fa fa-check"></i><b>2.4</b> Distribution of CpG coverage per sample</a></li>
<li class="chapter" data-level="2.5" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-covered-cpgs"><i class="fa fa-check"></i><b>2.5</b> Distribution of covered CpGs</a></li>
<li class="chapter" data-level="2.6" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#promoter-coverage"><i class="fa fa-check"></i><b>2.6</b> Promoter coverage</a></li>
<li class="chapter" data-level="2.7" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#coverage-per-clinical-annotations"><i class="fa fa-check"></i><b>2.7</b> Coverage per clinical annotations</a></li>
<li class="chapter" data-level="2.8" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage"><i class="fa fa-check"></i><b>2.8</b> Marginal CpG coverage</a><ul>
<li class="chapter" data-level="2.8.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage-of-k4me1-hotspots"><i class="fa fa-check"></i><b>2.8.1</b> Marginal CpG coverage of k4me1 hotspots</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#sample-cellularity"><i class="fa fa-check"></i><b>2.9</b> Sample cellularity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html"><i class="fa fa-check"></i><b>3</b> Tumor microenvironment (TME) deconvolution</a><ul>
<li class="chapter" data-level="3.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#initialize"><i class="fa fa-check"></i><b>3.2</b> Initialize</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#load-data"><i class="fa fa-check"></i><b>3.3</b> Load data</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#deconvolve-tme"><i class="fa fa-check"></i><b>3.4</b> Deconvolve TME</a><ul>
<li class="chapter" data-level="3.4.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er-samples"><i class="fa fa-check"></i><b>3.4.1</b> ER+ samples</a></li>
<li class="chapter" data-level="3.4.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er--samples"><i class="fa fa-check"></i><b>3.4.2</b> ER- samples</a></li>
<li class="chapter" data-level="3.4.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#normal-samples"><i class="fa fa-check"></i><b>3.4.3</b> Normal samples</a></li>
<li class="chapter" data-level="3.4.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-normalized-methylation"><i class="fa fa-check"></i><b>3.4.4</b> Merge normalized methylation</a></li>
<li class="chapter" data-level="3.4.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-tme-scores"><i class="fa fa-check"></i><b>3.4.5</b> Merge TME scores</a></li>
<li class="chapter" data-level="3.4.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#caf-and-immune-genes"><i class="fa fa-check"></i><b>3.4.6</b> CAF and Immune genes</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#expression-methylation-correlation-matrices"><i class="fa fa-check"></i><b>3.5</b> Expression-methylation correlation matrices</a><ul>
<li class="chapter" data-level="3.5.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#export-clustering"><i class="fa fa-check"></i><b>3.5.1</b> Export clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#diagnose-tme-normalization"><i class="fa fa-check"></i><b>3.6</b> Diagnose TME normalization</a></li>
<li class="chapter" data-level="3.7" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#plot-cd3d-and-cav1-vs-immune-and-caf-methylation-modules"><i class="fa fa-check"></i><b>3.7</b> Plot CD3D and CAV1 vs immune and CAF methylation modules</a></li>
<li class="chapter" data-level="3.8" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#annotate-tme-scores"><i class="fa fa-check"></i><b>3.8</b> Annotate TME scores</a></li>
<li class="chapter" data-level="3.9" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#appendix-sensitivity-analysis-of-tme-normalization"><i class="fa fa-check"></i><b>3.9</b> Appendix: Sensitivity analysis of TME normalization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html"><i class="fa fa-check"></i><b>4</b> Definition of Clock, MG and ML scores</a><ul>
<li class="chapter" data-level="4.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#initialize-1"><i class="fa fa-check"></i><b>4.2</b> Initialize</a></li>
<li class="chapter" data-level="4.3" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#load-data-1"><i class="fa fa-check"></i><b>4.3</b> Load data</a></li>
<li class="chapter" data-level="4.4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#clustering-of-normalized-methylation-of-er-tumors"><i class="fa fa-check"></i><b>4.4</b> Clustering of normalized methylation of ER+ tumors</a></li>
<li class="chapter" data-level="4.5" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-clustering-on-er--correlation-matrix"><i class="fa fa-check"></i><b>4.5</b> Project clustering on ER- correlation matrix</a></li>
<li class="chapter" data-level="4.6" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#classify-loci-in-the-genome-to-epigenomic-features"><i class="fa fa-check"></i><b>4.6</b> Classify loci in the genome to epigenomic features</a></li>
<li class="chapter" data-level="4.7" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#average-methylation-of-the-features"><i class="fa fa-check"></i><b>4.7</b> Average methylation of the features</a></li>
<li class="chapter" data-level="4.8" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-distribution-over-samples-of-the-correlated-loci"><i class="fa fa-check"></i><b>4.8</b> Plot distribution over samples of the correlated loci</a></li>
<li class="chapter" data-level="4.9" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-loci-classification"><i class="fa fa-check"></i><b>4.9</b> Plot loci classification</a></li>
<li class="chapter" data-level="4.10" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-all-epigenomic-features-using-umap"><i class="fa fa-check"></i><b>4.10</b> Project all epigenomic features using UMAP</a></li>
<li class="chapter" data-level="4.11" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#compare-epigenomic-scores-to-clinical-annotations"><i class="fa fa-check"></i><b>4.11</b> Compare epigenomic scores to clinical annotations</a></li>
<li class="chapter" data-level="4.12" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#distribution-of-the-scores-in-ic10-integrative-clusters"><i class="fa fa-check"></i><b>4.12</b> Distribution of the scores in iC10 integrative clusters</a></li>
<li class="chapter" data-level="4.13" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#epipolymorphism-of-the-epigenomic-layers"><i class="fa fa-check"></i><b>4.13</b> Epipolymorphism of the epigenomic layers</a></li>
<li class="chapter" data-level="4.14" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#comparison-of-the-epigenomic-scores-with-existing-epigenomic-age-metrics"><i class="fa fa-check"></i><b>4.14</b> Comparison of the epigenomic scores with existing epigenomic age metrics</a><ul>
<li class="chapter" data-level="4.14.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#phenoage"><i class="fa fa-check"></i><b>4.14.1</b> phenoAge</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="loss-clock.html"><a href="loss-clock.html"><i class="fa fa-check"></i><b>5</b> Loss clock</a><ul>
<li class="chapter" data-level="5.1" data-path="loss-clock.html"><a href="loss-clock.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="loss-clock.html"><a href="loss-clock.html#initialize-2"><i class="fa fa-check"></i><b>5.2</b> Initialize</a></li>
<li class="chapter" data-level="5.3" data-path="loss-clock.html"><a href="loss-clock.html#plot-methylation-distribution-of-the-clock"><i class="fa fa-check"></i><b>5.3</b> Plot methylation distribution of the clock</a></li>
<li class="chapter" data-level="5.4" data-path="loss-clock.html"><a href="loss-clock.html#annotate-clock-score"><i class="fa fa-check"></i><b>5.4</b> Annotate &quot;clock&quot; score</a><ul>
<li class="chapter" data-level="5.4.1" data-path="loss-clock.html"><a href="loss-clock.html#loci-annotation"><i class="fa fa-check"></i><b>5.4.1</b> Loci annotation</a></li>
<li class="chapter" data-level="5.4.2" data-path="loss-clock.html"><a href="loss-clock.html#clinical-annotation"><i class="fa fa-check"></i><b>5.4.2</b> Clinical annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="loss-clock.html"><a href="loss-clock.html#plot-chromosomal-traces-of-clock-score"><i class="fa fa-check"></i><b>5.5</b> Plot chromosomal traces of clock score</a></li>
<li class="chapter" data-level="5.6" data-path="loss-clock.html"><a href="loss-clock.html#correlation-of-gene-expression-to-clock"><i class="fa fa-check"></i><b>5.6</b> Correlation of gene expression to clock</a></li>
<li class="chapter" data-level="5.7" data-path="loss-clock.html"><a href="loss-clock.html#plot-score-in-different-tor-regimes"><i class="fa fa-check"></i><b>5.7</b> Plot score in different TOR regimes</a></li>
<li class="chapter" data-level="5.8" data-path="loss-clock.html"><a href="loss-clock.html#plot-high-low-clock-tumors-vs-tor"><i class="fa fa-check"></i><b>5.8</b> Plot High-Low clock tumors vs TOR</a></li>
<li class="chapter" data-level="5.9" data-path="loss-clock.html"><a href="loss-clock.html#plot-correlation-of-loci-to-clock-methylation"><i class="fa fa-check"></i><b>5.9</b> Plot correlation of loci to clock methylation</a></li>
<li class="chapter" data-level="5.10" data-path="loss-clock.html"><a href="loss-clock.html#plot-distribution-of-methylation-of-clock-in-normal-samples"><i class="fa fa-check"></i><b>5.10</b> Plot distribution of methylation of clock in normal samples</a></li>
<li class="chapter" data-level="5.11" data-path="loss-clock.html"><a href="loss-clock.html#compare-clock-layer-to-biological-age"><i class="fa fa-check"></i><b>5.11</b> Compare clock layer to biological age</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html"><i class="fa fa-check"></i><b>6</b> Epigenomic instability</a><ul>
<li class="chapter" data-level="6.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#compare-features-to-each-other"><i class="fa fa-check"></i><b>6.1</b> Compare features to each other</a></li>
<li class="chapter" data-level="6.2" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#annotate-epignomic-instability-scores"><i class="fa fa-check"></i><b>6.2</b> Annotate epignomic instability scores</a><ul>
<li class="chapter" data-level="6.2.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#integrative-clusters"><i class="fa fa-check"></i><b>6.2.1</b> Integrative clusters</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#cross-correlation-methylation-expression-of-epigenomic-instability-loci"><i class="fa fa-check"></i><b>6.3</b> Cross correlation (methylation-expression) of epigenomic instability loci</a></li>
<li class="chapter" data-level="6.4" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#epigenomic-instability-in-different-time-of-replication-tor-regimes"><i class="fa fa-check"></i><b>6.4</b> Epigenomic instability in different Time of replication (TOR) regimes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cna.html"><a href="cna.html"><i class="fa fa-check"></i><b>7</b> CNA</a></li>
<li class="chapter" data-level="8" data-path="mutations.html"><a href="mutations.html"><i class="fa fa-check"></i><b>8</b> Mutations</a></li>
<li class="chapter" data-level="9" data-path="survival.html"><a href="survival.html"><i class="fa fa-check"></i><b>9</b> Survival</a><ul>
<li class="chapter" data-level="9.0.1" data-path="survival.html"><a href="survival.html#raw-methylation"><i class="fa fa-check"></i><b>9.0.1</b> Raw-methylation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html"><i class="fa fa-check"></i><b>10</b> Methylation - Expression <em>in cis</em></a><ul>
<li class="chapter" data-level="10.0.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#remove-tme-genes"><i class="fa fa-check"></i><b>10.0.1</b> Remove TME genes</a></li>
<li class="chapter" data-level="10.0.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#load-normalized-methylation-and-separate-it-to-promoters-and-non-promoters"><i class="fa fa-check"></i><b>10.0.2</b> Load normalized methylation and separate it to promoters and non-promoters</a></li>
<li class="chapter" data-level="10.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-promoters"><i class="fa fa-check"></i><b>10.1</b> Cis candidates: promoters</a><ul>
<li class="chapter" data-level="10.1.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-correlation-of-top-cis-regulated-promoter-candidates"><i class="fa fa-check"></i><b>10.1.1</b> Plot correlation of top cis-regulated promoter candidates</a></li>
<li class="chapter" data-level="10.1.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#annotation-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.2</b> Annotation of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#gene-expression-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.3</b> Gene expression of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#correlation-between-cis-regulated-genes"><i class="fa fa-check"></i><b>10.1.4</b> Correlation between cis-regulated genes</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.2</b> Epipolymorphism of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#promoter-examples"><i class="fa fa-check"></i><b>10.3</b> Promoter examples</a><ul>
<li class="chapter" data-level="10.3.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#brca1"><i class="fa fa-check"></i><b>10.3.1</b> BRCA1</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-genomic"><i class="fa fa-check"></i><b>10.4</b> Cis candidates: genomic</a><ul>
<li class="chapter" data-level="10.4.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-cis"><i class="fa fa-check"></i><b>10.4.1</b> Plot % cis</a></li>
<li class="chapter" data-level="10.4.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-distance-to-most-correlated-gene"><i class="fa fa-check"></i><b>10.4.2</b> Plot distance to most correlated gene</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-genomic"><i class="fa fa-check"></i><b>10.5</b> Epipolymorphism of cis-regulated genomic</a></li>
<li class="chapter" data-level="10.6" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#examples-for-cis-regulation-non-promoters"><i class="fa fa-check"></i><b>10.6</b> Examples for cis regulation (non-promoters)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html"><i class="fa fa-check"></i><b>11</b> X dosage compensation</a><ul>
<li class="chapter" data-level="11.1" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#x-analysis"><i class="fa fa-check"></i><b>11.1</b> X analysis</a></li>
<li class="chapter" data-level="11.2" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#autosome-analysis"><i class="fa fa-check"></i><b>11.2</b> autosome analysis</a></li>
<li class="chapter" data-level="11.3" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#checking-dosage-compensation-on-all-x-loci"><i class="fa fa-check"></i><b>11.3</b> Checking dosage compensation on all X loci</a></li>
<li class="chapter" data-level="11.4" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#of-x-lost"><i class="fa fa-check"></i><b>11.4</b> % of X lost</a></li>
<li class="chapter" data-level="11.5" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#plot-expression-methylation-correlation-for-x"><i class="fa fa-check"></i><b>11.5</b> Plot Expression-Methylation correlation for X</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="finding-specific-dosage-compensation-examples.html"><a href="finding-specific-dosage-compensation-examples.html"><i class="fa fa-check"></i><b>12</b> Finding specific dosage compensation examples</a></li>
<li class="chapter" data-level="13" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html"><i class="fa fa-check"></i><b>13</b> Export supplementary tables</a><ul>
<li class="chapter" data-level="13.1" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-1-metabric-samples-metadata"><i class="fa fa-check"></i><b>13.1</b> Table 1: METABRIC samples metadata</a></li>
<li class="chapter" data-level="13.2" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-2-immune-expression-signature"><i class="fa fa-check"></i><b>13.2</b> Table 2: Immune expression signature</a></li>
<li class="chapter" data-level="13.3" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-3-caf-expression-signature"><i class="fa fa-check"></i><b>13.3</b> Table 3: CAF expression signature</a></li>
<li class="chapter" data-level="13.4" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-4-expression-methylation-correlation"><i class="fa fa-check"></i><b>13.4</b> Table 4: Expression-Methylation correlation</a></li>
<li class="chapter" data-level="13.5" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-5-methylation-scores"><i class="fa fa-check"></i><b>13.5</b> Table 5: Methylation scores</a></li>
<li class="chapter" data-level="13.6" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-6-gene-expression-correlation-to-epigenomic-instability"><i class="fa fa-check"></i><b>13.6</b> Table 6: Gene expression correlation to epigenomic instability</a></li>
<li class="chapter" data-level="13.7" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-7-methylation-layers-loci"><i class="fa fa-check"></i><b>13.7</b> Table 7: Methylation layers loci</a></li>
<li class="chapter" data-level="13.8" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-8-cis-regulation-candidates"><i class="fa fa-check"></i><b>13.8</b> Table 8: Cis regulation candidates</a></li>
<li class="chapter" data-level="13.9" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-9-dosage-compensation-in-autosomes"><i class="fa fa-check"></i><b>13.9</b> Table 9: Dosage compensation in autosomes</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html"><i class="fa fa-check"></i><b>14</b> Files needed for the analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#misha-db"><i class="fa fa-check"></i><b>14.1</b> Misha DB</a></li>
<li class="chapter" data-level="14.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#additional-files"><i class="fa fa-check"></i><b>14.2</b> Additional files</a><ul>
<li class="chapter" data-level="14.2.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#metabric-data"><i class="fa fa-check"></i><b>14.2.1</b> METABRIC data</a></li>
<li class="chapter" data-level="14.2.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#notable-generated-files"><i class="fa fa-check"></i><b>14.2.2</b> Notable generated files</a></li>
<li class="chapter" data-level="14.2.3" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#spatial-methylation-distribution"><i class="fa fa-check"></i><b>14.2.3</b> Spatial methylation distribution</a></li>
<li class="chapter" data-level="14.2.4" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#copy-number-abberations"><i class="fa fa-check"></i><b>14.2.4</b> Copy number abberations</a></li>
<li class="chapter" data-level="14.2.5" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#genomic-annotations"><i class="fa fa-check"></i><b>14.2.5</b> Genomic annotations</a></li>
<li class="chapter" data-level="14.2.6" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#external-resources"><i class="fa fa-check"></i><b>14.2.6</b> External resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figures-key.html"><a href="figures-key.html"><i class="fa fa-check"></i><b>15</b> Figures key</a><ul>
<li class="chapter" data-level="15.1" data-path="figures-key.html"><a href="figures-key.html#figure-1"><i class="fa fa-check"></i><b>15.1</b> Figure 1</a></li>
<li class="chapter" data-level="15.2" data-path="figures-key.html"><a href="figures-key.html#figure-3"><i class="fa fa-check"></i><b>15.2</b> Figure 3</a></li>
<li class="chapter" data-level="15.3" data-path="figures-key.html"><a href="figures-key.html#figure-4"><i class="fa fa-check"></i><b>15.3</b> Figure 4</a></li>
<li class="chapter" data-level="15.4" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-1"><i class="fa fa-check"></i><b>15.4</b> Extended Data Figure 1</a></li>
<li class="chapter" data-level="15.5" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-2"><i class="fa fa-check"></i><b>15.5</b> Extended Data Figure 2</a></li>
<li class="chapter" data-level="15.6" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-3"><i class="fa fa-check"></i><b>15.6</b> Extended Data Figure 3</a></li>
<li class="chapter" data-level="15.7" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-4"><i class="fa fa-check"></i><b>15.7</b> Extended Data Figure 4</a></li>
<li class="chapter" data-level="15.8" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-5"><i class="fa fa-check"></i><b>15.8</b> Extended Data Figure 5</a></li>
<li class="chapter" data-level="15.9" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-6"><i class="fa fa-check"></i><b>15.9</b> Extended Data Figure 6</a></li>
<li class="chapter" data-level="15.10" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-7"><i class="fa fa-check"></i><b>15.10</b> Extended Data Figure 7</a></li>
<li class="chapter" data-level="15.11" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-8"><i class="fa fa-check"></i><b>15.11</b> Extended Data Figure 8</a></li>
<li class="chapter" data-level="15.12" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-9"><i class="fa fa-check"></i><b>15.12</b> Extended Data Figure 9</a></li>
<li class="chapter" data-level="15.13" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-10"><i class="fa fa-check"></i><b>15.13</b> Extended Data Figure 10</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="metabric-methylation-data-statistics" class="section level1">
<h1><span class="header-section-number">2</span> METABRIC methylation data statistics</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;scripts/init.R&quot;</span>))</code></pre></div>
<div id="breakdown-of-samples" class="section level2">
<h2><span class="header-section-number">2.1</span> Breakdown of samples</h2>
<div id="figure-1a" class="section level4">
<h4><span class="header-section-number">2.1.0.1</span> Figure 1A</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">7</span>)

p_donut &lt;-<span class="st"> </span>samp_data <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">count</span>(IHC) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(IHC)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(
            <span class="kw">tibble</span>(
                <span class="dt">IHC =</span> <span class="kw">c</span>(<span class="st">&quot;ER+HER2-&quot;</span>, <span class="st">&quot;ER+HER2+&quot;</span>, <span class="st">&quot;ER-HER2+&quot;</span>, <span class="st">&quot;TNBC&quot;</span>, <span class="st">&quot;ADJNORMAL&quot;</span>),
                <span class="dt">group =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER+HER2+&quot;</span>, <span class="st">&quot;ER-HER2+&quot;</span>, <span class="st">&quot;Triple negative&quot;</span>, <span class="st">&quot;Adjacent normal&quot;</span>)
            ), <span class="dt">by =</span> <span class="st">&quot;IHC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(
            <span class="dt">IHC =</span> <span class="kw">factor</span>(IHC, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ER+HER2-&quot;</span>, <span class="st">&quot;ER+HER2+&quot;</span>, <span class="st">&quot;ER-HER2+&quot;</span>, <span class="st">&quot;TNBC&quot;</span>, <span class="st">&quot;ADJNORMAL&quot;</span>)),
            <span class="dt">lab =</span> <span class="kw">glue</span>(<span class="st">&quot;{group}</span><span class="ch">\n</span><span class="st">({n})&quot;</span>)
        ) <span class="op">%&gt;%</span>
<span class="st">        </span>ggpubr<span class="op">::</span><span class="kw">ggdonutchart</span>(<span class="st">&quot;n&quot;</span>, <span class="st">&quot;lab&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;IHC&quot;</span>, <span class="dt">lab.pos =</span> <span class="st">&quot;out&quot;</span>, <span class="dt">palette =</span> annot_colors<span class="op">$</span>IHC, <span class="dt">font.family =</span> <span class="st">&quot;Arial&quot;</span>, <span class="dt">lab.font =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="st">&quot;bold&quot;</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">ggtheme =</span> ggpubr<span class="op">::</span><span class="kw">theme_pubr</span>(<span class="dt">base_size =</span> <span class="dv">6</span>, <span class="dt">base_family =</span> <span class="st">&quot;Arial&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="st">&quot;none&quot;</span>)


p_donut1 &lt;-<span class="st"> </span>samp_data <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">count</span>(ER1) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">rename</span>(<span class="dt">ER =</span> ER1) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(
            <span class="kw">tibble</span>(
                <span class="dt">ER =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>),
                <span class="dt">group =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;Adjacent normal&quot;</span>)
            ), <span class="dt">by =</span> <span class="st">&quot;ER&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(
            <span class="dt">ER =</span> <span class="kw">factor</span>(ER, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;ER+&quot;</span>, <span class="st">&quot;ER-&quot;</span>, <span class="st">&quot;normal&quot;</span>)),
            <span class="dt">lab =</span> <span class="kw">glue</span>(<span class="st">&quot;{group}</span><span class="ch">\n</span><span class="st">({n})&quot;</span>)
        ) <span class="op">%&gt;%</span>
<span class="st">        </span>ggpubr<span class="op">::</span><span class="kw">ggdonutchart</span>(<span class="st">&quot;n&quot;</span>, <span class="st">&quot;lab&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;ER&quot;</span>, <span class="dt">lab.pos =</span> <span class="st">&quot;out&quot;</span>, <span class="dt">palette =</span> annot_colors<span class="op">$</span>ER1, <span class="dt">font.family =</span> <span class="st">&quot;Arial&quot;</span>, <span class="dt">lab.font =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="st">&quot;bold&quot;</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">ggtheme =</span> ggpubr<span class="op">::</span><span class="kw">theme_pubr</span>(<span class="dt">base_size =</span> <span class="dv">6</span>, <span class="dt">base_family =</span> <span class="st">&quot;Arial&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="st">&quot;none&quot;</span>)

p_donut</code></pre></div>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_donut1</code></pre></div>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
</div>
</div>
<div id="number-of-samples" class="section level2">
<h2><span class="header-section-number">2.2</span> Number of samples</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">ER1 =</span> forcats<span class="op">::</span><span class="kw">fct_explicit_na</span>(ER1)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(ER1)</code></pre></div>
<pre><code>## # A tibble: 4 x 2
##         ER1    n
## 1       ER-  350
## 2       ER+ 1179
## 3    normal  244
## 4 (Missing)    9</code></pre>
</div>
<div id="total-number-of-reads" class="section level2">
<h2><span class="header-section-number">2.3</span> Total number of reads</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tot_reads &lt;-<span class="st"> </span><span class="kw">fread</span>(<span class="kw">here</span>(<span class="st">&quot;data/sample_qc.csv&quot;</span>)) <span class="op">%&gt;%</span><span class="st">     </span>
<span class="st">    </span><span class="kw">inner_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">select</span>(track), <span class="dt">by =</span> <span class="st">&quot;track&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">pull</span>(total_reads) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">sum</span>() 

<span class="kw">glue</span>(<span class="st">&quot;Overall, we used {round(tot_reads / 1e9, digits = 1)}B reads to cover {scales::comma(nrow(samp_data))} breast tumor and normal samples&quot;</span>)</code></pre></div>
<pre><code>## Overall, we used 30.4B reads to cover 1,782 breast tumor and normal samples</code></pre>
</div>
<div id="distribution-of-cpg-coverage-per-sample" class="section level2">
<h2><span class="header-section-number">2.4</span> Distribution of CpG coverage per sample</h2>
<p>We calculate the distribution of every sample's CpG coverage:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_covs &lt;-<span class="st"> </span><span class="kw">get_samples_coverage_dist</span>(
    <span class="dt">tracks =</span> samp_data<span class="op">$</span>track, 
    <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">100</span>, <span class="fl">1e5</span>), 
    <span class="dt">intervals =</span> <span class="kw">gintervals.all</span>()) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/sample_coverage_dist.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_covs &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, <span class="dt">ER =</span> ER1), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>)</code></pre></div>
<div id="figure-1b" class="section level4">
<h4><span class="header-section-number">2.4.0.1</span> Figure 1B</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">7</span>)
p_comulative &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">group_by</span>(samp) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">summarise</span>(
            <span class="st">`</span><span class="dt">&gt;=10</span><span class="st">`</span> =<span class="st"> </span><span class="kw">sum</span>(cpgs[cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">10</span>]),
            <span class="st">`</span><span class="dt">&gt;=20</span><span class="st">`</span> =<span class="st"> </span><span class="kw">sum</span>(cpgs[cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">20</span>]),
            <span class="st">`</span><span class="dt">&gt;=30</span><span class="st">`</span> =<span class="st"> </span><span class="kw">sum</span>(cpgs[cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">30</span>]),
            <span class="st">`</span><span class="dt">&gt;=50</span><span class="st">`</span> =<span class="st"> </span><span class="kw">sum</span>(cpgs[cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">50</span>]),
            <span class="dt">n_tot =</span> <span class="kw">sum</span>(cpgs)
        ) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">gather</span>(<span class="st">&quot;type&quot;</span>, <span class="st">&quot;n&quot;</span>, <span class="op">-</span>samp, <span class="op">-</span>n_tot) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n, <span class="dt">color =</span> type, <span class="dt">y =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>..y..)) <span class="op">+</span>
<span class="st">        </span><span class="kw">stat_ecdf</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_color_viridis_d</span>(<span class="st">&quot;CpG coverage&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> <span class="cf">function</span>(x) <span class="kw">round</span>(x <span class="op">*</span><span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(samp_covs<span class="op">$</span>samp)))) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Samples with #CpGs &gt;= x&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;# of CpGs (log10)&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.8</span>)

p_comulative <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="fl">0.8</span>)</code></pre></div>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov_above_<span class="dv">10</span> &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">10</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(samp) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">cpgs =</span> <span class="kw">sum</span>(cpgs))

df &lt;-<span class="st"> </span>cov_above_<span class="dv">10</span> <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_threshold =</span> <span class="kw">sum</span>(cpgs <span class="op">&gt;=</span><span class="st"> </span><span class="fl">1e6</span>), <span class="dt">n_tot =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">p =</span> scales<span class="op">::</span><span class="kw">percent</span>(n_threshold <span class="op">/</span><span class="st"> </span>n_tot))

<span class="kw">glue</span>(<span class="st">&quot;Number of samples (Y axis) with a given number of CpGs (X axis) covered with at least 10, 20, 30 or 50 reads. For example, in all samples 449,710 CpGs are covered with over 10 reads.&quot;</span>, <span class="dt">n_cpgs =</span> scales<span class="op">::</span><span class="kw">comma</span>(<span class="kw">min</span>(cov_above_<span class="dv">10</span><span class="op">$</span>cpgs)))</code></pre></div>
<pre><code>## Number of samples (Y axis) with a given number of CpGs (X axis) covered with at least 10, 20, 30 or 50 reads. For example, in all samples 449,710 CpGs are covered with over 10 reads.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">glue</span>(<span class="st">&quot;Using our version of the RRBS protocol {perc} of the samples were covered by more than 10 reads for more than 1M CpGs&quot;</span>, <span class="dt">perc =</span> df<span class="op">$</span>p)</code></pre></div>
<pre><code>## Using our version of the RRBS protocol 93% of the samples were covered by more than 10 reads for more than 1M CpGs</code></pre>
</div>
</div>
<div id="distribution-of-covered-cpgs" class="section level2">
<h2><span class="header-section-number">2.5</span> Distribution of covered CpGs</h2>
<p>We screen for CpGs that are covered by at least 5 reads in half or more of the samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">min_samples &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="fl">0.5</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(samp_data))
min_samples</code></pre></div>
<pre><code>## [1] 891</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov_cpgs &lt;-<span class="st"> </span><span class="kw">gscreen_coverage</span>(samp_data<span class="op">$</span>track, <span class="dv">5</span>, min_samples) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">annotate_loci</span>() <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/cov_cpgs.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div id="figure-1c" class="section level4">
<h4><span class="header-section-number">2.5.0.1</span> Figure 1C</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">7</span>, <span class="dt">repr.plot.height =</span> <span class="dv">3</span>)
p_tss &lt;-<span class="st"> </span>cov_cpgs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">tss_d =</span> <span class="kw">abs</span>(tss_d) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> tss_d)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_density</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Distance from tss (bp)&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>)

p_cg_cont &lt;-<span class="st"> </span>cov_cpgs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> cg_cont)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_density</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;CpG content (500 bp)&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>)

(p_tss <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()) <span class="op">+</span><span class="st"> </span>(p_cg_cont <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>())</code></pre></div>
<pre><code>## Warning: Removed 147 rows containing non-finite values (stat_density).</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
</div>
<div id="promoter-coverage" class="section level2">
<h2><span class="header-section-number">2.6</span> Promoter coverage</h2>
<p>Distribution of mean promoter coverage over all METABRIC samples, considering 13,282 active promoters. Active promoters were defined as promoters that had log expression &gt;7 in at least one of the METABRIC samples used in this paper.</p>
<div id="figure-1d" class="section level4">
<h4><span class="header-section-number">2.6.0.1</span> Figure 1D</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="fl">4.6</span>, <span class="dt">repr.plot.height =</span> <span class="fl">2.7</span>)
prom_expr_cov &lt;-<span class="st"> </span><span class="kw">get_promoter_expr_covs</span>()

min_n_expr &lt;-<span class="st"> </span><span class="dv">0</span>
max_expr &lt;-<span class="st"> </span><span class="dv">7</span>

prom_expr_cov_active &lt;-<span class="st"> </span>prom_expr_cov <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(n_expr <span class="op">&gt;</span><span class="st"> </span>min_n_expr, max_expr <span class="op">&gt;</span><span class="st"> </span><span class="op">!!</span><span class="st"> </span>max_expr)

p_prom_cov &lt;-<span class="st"> </span>prom_expr_cov_active  <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> cov)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_histogram</span>(<span class="dt">binwidth =</span> <span class="fl">0.1</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">50</span>, <span class="dv">1000</span>), <span class="dt">labels =</span> <span class="cf">function</span>(x) <span class="kw">round</span>(x)) <span class="op">+</span>
<span class="st">        </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw">c</span>(<span class="fl">0.01</span>,<span class="dv">1500</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of promoters&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Mean promoter coverage&quot;</span>)

p_prom_cov <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>## Warning: Removed 119 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(prom_expr_cov_active)</code></pre></div>
<pre><code>## [1] 13198</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">n_cov &lt;-<span class="st"> </span>prom_expr_cov_active <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">20</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>()

<span class="kw">glue</span>(<span class="st">&quot;{perc} of the promoters were nevertheless covered with over 20 reads on average (mean coverage {cov}), facilitating quantitative analysis downstream.&quot;</span>, <span class="dt">perc =</span> scales<span class="op">::</span><span class="kw">percent</span>(n_cov <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(prom_expr_cov_active)), <span class="dt">cov =</span> <span class="kw">round</span>(<span class="kw">mean</span>(prom_expr_cov_active<span class="op">$</span>cov) ))</code></pre></div>
<pre><code>## 75% of the promoters were nevertheless covered with over 20 reads on average (mean coverage 246), facilitating quantitative analysis downstream.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_tot_calls &lt;-<span class="st"> </span><span class="kw">get_sample_tot_meth_calls</span>() <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/sample_tot_meth_calls.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
samp_tot_calls_promoters &lt;-<span class="st"> </span><span class="kw">get_sample_tot_meth_calls</span>(promoter_intervs) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/sample_tot_meth_calls_promoters.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
tot_reads &lt;-<span class="st"> </span><span class="kw">sum</span>(samp_tot_calls<span class="op">$</span>Sum)
tot_promoter_reads &lt;-<span class="st"> </span><span class="kw">sum</span>(samp_tot_calls_promoters<span class="op">$</span>Sum)
<span class="kw">glue</span>(<span class="st">&quot;{perc} of the reads mapped to bona-fide promoters&quot;</span>, <span class="dt">perc =</span> scales<span class="op">::</span><span class="kw">percent</span>(tot_promoter_reads <span class="op">/</span><span class="st"> </span>tot_reads))</code></pre></div>
<pre><code>## 9% of the reads mapped to bona-fide promoters</code></pre>
<p>We can also calculate the % of MSP1 fragments that is covering a promoter:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">well_covered_loci &lt;-<span class="st"> </span>{
    msp1_meth &lt;-<span class="st"> </span><span class="kw">get_msp1_meth</span>()
    msp1_meth_f &lt;-<span class="st"> </span><span class="kw">filter_avg_meth</span>(msp1_meth<span class="op">$</span>avg, <span class="dt">normal_fraction =</span> <span class="fl">0.7</span>, <span class="dt">tumor_fraction =</span> <span class="fl">0.7</span>)
    msp1_meth_f <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chrom, start, end)
} <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/well_covered_msp1_frags.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()

n_loci &lt;-<span class="st"> </span><span class="kw">nrow</span>(well_covered_loci)
n_promoters &lt;-<span class="st"> </span>well_covered_loci <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">gintervals.neighbors1</span>(promoter_intervs) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(dist <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>()

scales<span class="op">::</span><span class="kw">percent</span>(n_promoters <span class="op">/</span><span class="st"> </span>n_loci)</code></pre></div>
<pre><code>## [1] &quot;10%&quot;</code></pre>
</div>
</div>
<div id="coverage-per-clinical-annotations" class="section level2">
<h2><span class="header-section-number">2.7</span> Coverage per clinical annotations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_covs &lt;-<span class="st"> </span><span class="kw">get_sample_tot_meth_calls</span>() <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/sample_tot_meth_calls.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()
samp_covs &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">select</span>(samp, <span class="dt">tot_cov =</span> Sum) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, stage, grade, iC10, <span class="dt">ER =</span> ER1), <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">stage =</span> <span class="kw">ifelse</span>(stage <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="st">&quot;DCIS&quot;</span>, <span class="dv">1</span>), <span class="st">&quot;0-1&quot;</span>, stage)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">stage =</span> <span class="kw">ifelse</span>(ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>, <span class="st">&quot;N&quot;</span>, stage)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">grade =</span> <span class="kw">ifelse</span>(ER <span class="op">==</span><span class="st"> &quot;normal&quot;</span>, <span class="st">&quot;N&quot;</span>, grade)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">stage =</span> <span class="kw">factor</span>(stage, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;N&quot;</span>, <span class="st">&quot;0-1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>, <span class="st">&quot;4&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">grade =</span> <span class="kw">factor</span>(grade, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;N&quot;</span>, <span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>)))        </code></pre></div>
<div id="extended-data-figure-1a" class="section level4">
<h4><span class="header-section-number">2.7.0.1</span> Extended Data Figure 1a</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">10</span>, <span class="dt">repr.plot.height =</span> <span class="dv">3</span>)
p_stage &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(stage), <span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">arrange</span>(stage, tot_cov) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">samp =</span> forcats<span class="op">::</span><span class="kw">fct_inorder</span>(samp)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> samp, <span class="dt">y =</span> tot_cov, <span class="dt">fill =</span> stage)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_col</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span><span class="kw">unit_format</span>(<span class="dt">unit =</span> <span class="st">&quot;M&quot;</span>, <span class="dt">scale =</span> <span class="fl">1e-6</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_fill_manual</span>(<span class="dt">name =</span> <span class="st">&quot;Stage&quot;</span>, <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;N&quot;</span> =<span class="st"> &quot;gray&quot;</span>, <span class="st">&quot;0-1&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;2&quot;</span> =<span class="st"> &quot;blue&quot;</span>, <span class="st">&quot;3&quot;</span> =<span class="st"> &quot;red&quot;</span>, <span class="st">&quot;4&quot;</span> =<span class="st"> &quot;orange&quot;</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of methylation calls&quot;</span>)

p_stage <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>())</code></pre></div>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="extended-data-figure-1b" class="section level4">
<h4><span class="header-section-number">2.7.0.2</span> Extended Data Figure 1b</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">10</span>, <span class="dt">repr.plot.height =</span> <span class="dv">3</span>)

p_iC10 &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(iC10), <span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">iC10 =</span> <span class="kw">factor</span>(iC10, <span class="dt">levels =</span> <span class="kw">names</span>(annot_colors<span class="op">$</span>iC10))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">arrange</span>(iC10, tot_cov) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">samp =</span> forcats<span class="op">::</span><span class="kw">fct_inorder</span>(samp)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> samp, <span class="dt">y =</span> tot_cov, <span class="dt">fill =</span> iC10)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_col</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels =</span> scales<span class="op">::</span><span class="kw">unit_format</span>(<span class="dt">unit =</span> <span class="st">&quot;M&quot;</span>, <span class="dt">scale =</span> <span class="fl">1e-6</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>iC10) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of methylation calls&quot;</span>)

p_iC10 <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>())</code></pre></div>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
</div>
<div id="marginal-cpg-coverage" class="section level2">
<h2><span class="header-section-number">2.8</span> Marginal CpG coverage</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg_mars &lt;-<span class="st"> </span><span class="kw">covs_marginal</span>(samp_data<span class="op">$</span>track) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/cpg_cov_marginal.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg_mars &lt;-<span class="st"> </span>cpg_mars <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">mean_cov =</span> cov <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(samp_data))
cov_cdf &lt;-<span class="st"> </span><span class="kw">map_df</span>(<span class="kw">c</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">100</span>, <span class="kw">seq</span>(<span class="dv">200</span>, <span class="dv">1000</span>, <span class="dv">20</span>)), <span class="op">~</span><span class="st"> </span><span class="kw">tibble</span>(<span class="dt">cov =</span> .x, <span class="dt">n =</span> <span class="kw">sum</span>(cpg_mars<span class="op">$</span>mean_cov <span class="op">&gt;=</span><span class="st"> </span>.x)))  <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/cpg_cov_cdf.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</code></pre></div>
<div id="extended-data-figure-1b-1" class="section level4">
<h4><span class="header-section-number">2.8.0.1</span> Extended Data Figure 1b</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="fl">4.4</span>, <span class="dt">repr.plot.height =</span> <span class="fl">3.5</span>)

p_cov_cdf &lt;-<span class="st"> </span>cov_cdf <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(cov <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> cov, <span class="dt">y =</span> n)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_line</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Average CpG coverage&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of CpGs&quot;</span>)

p_cov_cdf <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() </code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
<div id="marginal-cpg-coverage-of-k4me1-hotspots" class="section level3">
<h3><span class="header-section-number">2.8.1</span> Marginal CpG coverage of k4me1 hotspots</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">conf &lt;-<span class="st"> </span>main_config<span class="op">$</span>genomic_regions<span class="op">$</span>enhancers

conf<span class="op">$</span>H3K4me1_tracks</code></pre></div>
<pre><code>## [1] &quot;Roadmap.Breast_Luminal_Epithelial_Cells.H3K4me1&quot;      
## [2] &quot;Roadmap.Breast_Myoepithelial_Cells.H3K4me1_1&quot;         
## [3] &quot;Roadmap.Breast_Myoepithelial_Cells.H3K4me1_2&quot;         
## [4] &quot;encode.Tfbs.HMECwgEncodeBroadHistoneHmecH3k4me1StdSig&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">conf<span class="op">$</span>H3K4me1_thresh</code></pre></div>
<pre><code>## [1] 0.97</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">conf<span class="op">$</span>H3K4me1_size</code></pre></div>
<pre><code>## [1] 200</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">intervs &lt;-<span class="st"> </span><span class="kw">gpatterns.putative_enhancers</span>(conf<span class="op">$</span>H3K4me1_tracks, <span class="dt">quant_thresh =</span> conf<span class="op">$</span>H3K4me1_thresh, <span class="dt">normalize =</span> conf<span class="op">$</span>H3K4me1_size, <span class="dt">min_tss_dist =</span> conf<span class="op">$</span>H3K4me1_tss_dist)
k4me1_mars &lt;-<span class="st"> </span>cpg_mars <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">gintervals.neighbors1</span>(intervs) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(dist <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(chrom1, start1, end1) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">cov =</span> <span class="kw">sum</span>(cov), <span class="dt">mean_cov =</span> cov <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(samp_data))
k4me1_mars &lt;-<span class="st"> </span>k4me1_mars <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">select</span>(<span class="dt">chrom =</span> chrom1, <span class="dt">start =</span> start1, <span class="dt">end =</span> end1, cov, mean_cov)</code></pre></div>
<div id="extended-data-figure-1c" class="section level4">
<h4><span class="header-section-number">2.8.1.1</span> Extended Data Figure 1c</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="fl">4.4</span>, <span class="dt">repr.plot.height =</span> <span class="fl">3.5</span>)

p_k4me1 &lt;-<span class="st"> </span>k4me1_mars <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> mean_cov, <span class="dt">y =</span> (<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>..y..) <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(k4me1_mars))) <span class="op">+</span>
<span class="st">        </span><span class="kw">stat_ecdf</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="fl">1e3</span>, <span class="fl">1e4</span>, <span class="fl">1e5</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">30</span>, <span class="dv">100</span>, <span class="dv">1000</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Mean coverage&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of k4me1 hotsplots&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">vertical_labs</span>()

p_k4me1 <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">k4me1_mars &lt;-<span class="st"> </span>cpg_mars <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">gintervals.neighbors1</span>(enh_intervs_tumors) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(dist <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(chrom1, start1, end1) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">cov =</span> <span class="kw">sum</span>(cov), <span class="dt">mean_cov =</span> cov <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(samp_data))
k4me1_mars &lt;-<span class="st"> </span>k4me1_mars <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">select</span>(<span class="dt">chrom =</span> chrom1, <span class="dt">start =</span> start1, <span class="dt">end =</span> end1, cov, mean_cov)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="fl">4.4</span>, <span class="dt">repr.plot.height =</span> <span class="fl">3.5</span>)

p_k4me1_tumors &lt;-<span class="st"> </span>k4me1_mars <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> mean_cov, <span class="dt">y =</span> (<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>..y..) <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(k4me1_mars))) <span class="op">+</span>
<span class="st">        </span><span class="kw">stat_ecdf</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_y_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="fl">1e3</span>, <span class="fl">1e4</span>, <span class="fl">1e5</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">30</span>, <span class="dv">100</span>, <span class="dv">1000</span>)) <span class="op">+</span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Mean coverage&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;# of k4me1 hotsplots&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">vertical_labs</span>()

p_k4me1_tumors <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="sample-cellularity" class="section level2">
<h2><span class="header-section-number">2.9</span> Sample cellularity</h2>
<div id="extended-data-figure-1d" class="section level4">
<h4><span class="header-section-number">2.9.0.1</span> Extended Data Figure 1d</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_cellularity_cov &lt;-<span class="st"> </span>samp_covs <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, cna_cellularity)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">slice</span>(<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> tot_cov, <span class="dt">y =</span> cna_cellularity, <span class="dt">color =</span> ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">scale_x_log10</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">xlab</span>(<span class="st">&quot;Sample Coverage&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">ylab</span>(<span class="st">&quot;Cellularity&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">vertical_labs</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">facet_grid</span>(.<span class="op">~</span>ER)</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<pre><code>## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## &quot;none&quot;)` instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_cellularity_cov</code></pre></div>
<pre><code>## Warning: Removed 279 rows containing missing values (geom_point).</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samp_genomic_meth &lt;-<span class="st"> </span><span class="kw">get_genomic_avg_meth</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">intervs_to_mat</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">colMeans</span>(<span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">enframe</span>(<span class="st">&quot;samp&quot;</span>, <span class="st">&quot;meth&quot;</span>) <span class="op">%cache_df%</span><span class="st"> </span><span class="kw">here</span>(<span class="st">&quot;data/samp_genomic_meth.tsv&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()

p_cellularity_meth &lt;-<span class="st"> </span>samp_genomic_meth <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(samp_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(samp, cna_cellularity, <span class="dt">ER =</span> ER1)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ER)) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">slice</span>(<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> meth, <span class="dt">y =</span> cna_cellularity, <span class="dt">color =</span> ER)) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> annot_colors<span class="op">$</span>ER1) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">xlab</span>(<span class="st">&quot;Sample genomic methylation&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Cellularity&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">facet_grid</span>(.<span class="op">~</span>ER)</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<pre><code>## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## &quot;none&quot;)` instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_cellularity_meth</code></pre></div>
<pre><code>## Warning: Removed 279 rows containing missing values (geom_point).</code></pre>
<p><img src="coverage-stats_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gc</span>()</code></pre></div>
<pre><code>##             used   (Mb) gc trigger   (Mb)  max used   (Mb)
## Ncells   4497725  240.3    7606011  406.3   7606011  406.3
## Vcells 211991261 1617.4  562383716 4290.7 490804511 3744.6</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tumor-microenvironment-tme-deconvolution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
