<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Survival | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</title>
  <meta name="description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Survival | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="tanaylab.github.io/metabric_rrbs" />
  
  <meta property="og:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  <meta name="github-repo" content="tanaylab/metabric_rrbs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Survival | DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  
  <meta name="twitter:description" content="Analysis and code for the paper: DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation" />
  

<meta name="author" content="Aviezer Lifshitz" />


<meta name="date" content="2021-08-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations.html"/>
<link rel="next" href="methylation-expression-in-cis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#run-the-notebooks"><i class="fa fa-check"></i><b>1.1</b> Run the notebooks</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notebook-order"><i class="fa fa-check"></i><b>1.3</b> Notebook order</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#find-a-specific-figure"><i class="fa fa-check"></i><b>1.4</b> Find a specific figure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html"><i class="fa fa-check"></i><b>2</b> METABRIC methylation data statistics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#breakdown-of-samples"><i class="fa fa-check"></i><b>2.1</b> Breakdown of samples</a></li>
<li class="chapter" data-level="2.2" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#number-of-samples"><i class="fa fa-check"></i><b>2.2</b> Number of samples</a></li>
<li class="chapter" data-level="2.3" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#total-number-of-reads"><i class="fa fa-check"></i><b>2.3</b> Total number of reads</a></li>
<li class="chapter" data-level="2.4" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-cpg-coverage-per-sample"><i class="fa fa-check"></i><b>2.4</b> Distribution of CpG coverage per sample</a></li>
<li class="chapter" data-level="2.5" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#distribution-of-covered-cpgs"><i class="fa fa-check"></i><b>2.5</b> Distribution of covered CpGs</a></li>
<li class="chapter" data-level="2.6" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#promoter-coverage"><i class="fa fa-check"></i><b>2.6</b> Promoter coverage</a></li>
<li class="chapter" data-level="2.7" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#coverage-per-clinical-annotations"><i class="fa fa-check"></i><b>2.7</b> Coverage per clinical annotations</a></li>
<li class="chapter" data-level="2.8" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage"><i class="fa fa-check"></i><b>2.8</b> Marginal CpG coverage</a><ul>
<li class="chapter" data-level="2.8.1" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#marginal-cpg-coverage-of-k4me1-hotspots"><i class="fa fa-check"></i><b>2.8.1</b> Marginal CpG coverage of k4me1 hotspots</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="metabric-methylation-data-statistics.html"><a href="metabric-methylation-data-statistics.html#sample-cellularity"><i class="fa fa-check"></i><b>2.9</b> Sample cellularity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html"><i class="fa fa-check"></i><b>3</b> Tumor microenvironment (TME) deconvolution</a><ul>
<li class="chapter" data-level="3.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#initialize"><i class="fa fa-check"></i><b>3.2</b> Initialize</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#load-data"><i class="fa fa-check"></i><b>3.3</b> Load data</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#deconvolve-tme"><i class="fa fa-check"></i><b>3.4</b> Deconvolve TME</a><ul>
<li class="chapter" data-level="3.4.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er-samples"><i class="fa fa-check"></i><b>3.4.1</b> ER+ samples</a></li>
<li class="chapter" data-level="3.4.2" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#er--samples"><i class="fa fa-check"></i><b>3.4.2</b> ER- samples</a></li>
<li class="chapter" data-level="3.4.3" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#normal-samples"><i class="fa fa-check"></i><b>3.4.3</b> Normal samples</a></li>
<li class="chapter" data-level="3.4.4" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-normalized-methylation"><i class="fa fa-check"></i><b>3.4.4</b> Merge normalized methylation</a></li>
<li class="chapter" data-level="3.4.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#merge-tme-scores"><i class="fa fa-check"></i><b>3.4.5</b> Merge TME scores</a></li>
<li class="chapter" data-level="3.4.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#caf-and-immune-genes"><i class="fa fa-check"></i><b>3.4.6</b> CAF and Immune genes</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#expression-methylation-correlation-matrices"><i class="fa fa-check"></i><b>3.5</b> Expression-methylation correlation matrices</a><ul>
<li class="chapter" data-level="3.5.1" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#export-clustering"><i class="fa fa-check"></i><b>3.5.1</b> Export clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#diagnose-tme-normalization"><i class="fa fa-check"></i><b>3.6</b> Diagnose TME normalization</a></li>
<li class="chapter" data-level="3.7" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#plot-cd3d-and-cav1-vs-immune-and-caf-methylation-modules"><i class="fa fa-check"></i><b>3.7</b> Plot CD3D and CAV1 vs immune and CAF methylation modules</a></li>
<li class="chapter" data-level="3.8" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#annotate-tme-scores"><i class="fa fa-check"></i><b>3.8</b> Annotate TME scores</a></li>
<li class="chapter" data-level="3.9" data-path="tumor-microenvironment-tme-deconvolution.html"><a href="tumor-microenvironment-tme-deconvolution.html#appendix-sensitivity-analysis-of-tme-normalization"><i class="fa fa-check"></i><b>3.9</b> Appendix: Sensitivity analysis of TME normalization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html"><i class="fa fa-check"></i><b>4</b> Definition of Clock, MG and ML scores</a><ul>
<li class="chapter" data-level="4.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#initialize-1"><i class="fa fa-check"></i><b>4.2</b> Initialize</a></li>
<li class="chapter" data-level="4.3" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#load-data-1"><i class="fa fa-check"></i><b>4.3</b> Load data</a></li>
<li class="chapter" data-level="4.4" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#clustering-of-normalized-methylation-of-er-tumors"><i class="fa fa-check"></i><b>4.4</b> Clustering of normalized methylation of ER+ tumors</a></li>
<li class="chapter" data-level="4.5" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-clustering-on-er--correlation-matrix"><i class="fa fa-check"></i><b>4.5</b> Project clustering on ER- correlation matrix</a></li>
<li class="chapter" data-level="4.6" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#classify-loci-in-the-genome-to-epigenomic-features"><i class="fa fa-check"></i><b>4.6</b> Classify loci in the genome to epigenomic features</a></li>
<li class="chapter" data-level="4.7" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#average-methylation-of-the-features"><i class="fa fa-check"></i><b>4.7</b> Average methylation of the features</a></li>
<li class="chapter" data-level="4.8" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-distribution-over-samples-of-the-correlated-loci"><i class="fa fa-check"></i><b>4.8</b> Plot distribution over samples of the correlated loci</a></li>
<li class="chapter" data-level="4.9" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#plot-loci-classification"><i class="fa fa-check"></i><b>4.9</b> Plot loci classification</a></li>
<li class="chapter" data-level="4.10" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#project-all-epigenomic-features-using-umap"><i class="fa fa-check"></i><b>4.10</b> Project all epigenomic features using UMAP</a></li>
<li class="chapter" data-level="4.11" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#compare-epigenomic-scores-to-clinical-annotations"><i class="fa fa-check"></i><b>4.11</b> Compare epigenomic scores to clinical annotations</a></li>
<li class="chapter" data-level="4.12" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#distribution-of-the-scores-in-ic10-integrative-clusters"><i class="fa fa-check"></i><b>4.12</b> Distribution of the scores in iC10 integrative clusters</a></li>
<li class="chapter" data-level="4.13" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#epipolymorphism-of-the-epigenomic-layers"><i class="fa fa-check"></i><b>4.13</b> Epipolymorphism of the epigenomic layers</a></li>
<li class="chapter" data-level="4.14" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#comparison-of-the-epigenomic-scores-with-existing-epigenomic-age-metrics"><i class="fa fa-check"></i><b>4.14</b> Comparison of the epigenomic scores with existing epigenomic age metrics</a><ul>
<li class="chapter" data-level="4.14.1" data-path="definition-of-clock-mg-and-ml-scores.html"><a href="definition-of-clock-mg-and-ml-scores.html#phenoage"><i class="fa fa-check"></i><b>4.14.1</b> phenoAge</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="loss-clock.html"><a href="loss-clock.html"><i class="fa fa-check"></i><b>5</b> Loss clock</a><ul>
<li class="chapter" data-level="5.1" data-path="loss-clock.html"><a href="loss-clock.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="loss-clock.html"><a href="loss-clock.html#initialize-2"><i class="fa fa-check"></i><b>5.2</b> Initialize</a></li>
<li class="chapter" data-level="5.3" data-path="loss-clock.html"><a href="loss-clock.html#plot-methylation-distribution-of-the-clock"><i class="fa fa-check"></i><b>5.3</b> Plot methylation distribution of the clock</a></li>
<li class="chapter" data-level="5.4" data-path="loss-clock.html"><a href="loss-clock.html#annotate-clock-score"><i class="fa fa-check"></i><b>5.4</b> Annotate &quot;clock&quot; score</a><ul>
<li class="chapter" data-level="5.4.1" data-path="loss-clock.html"><a href="loss-clock.html#loci-annotation"><i class="fa fa-check"></i><b>5.4.1</b> Loci annotation</a></li>
<li class="chapter" data-level="5.4.2" data-path="loss-clock.html"><a href="loss-clock.html#clinical-annotation"><i class="fa fa-check"></i><b>5.4.2</b> Clinical annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="loss-clock.html"><a href="loss-clock.html#plot-chromosomal-traces-of-clock-score"><i class="fa fa-check"></i><b>5.5</b> Plot chromosomal traces of clock score</a></li>
<li class="chapter" data-level="5.6" data-path="loss-clock.html"><a href="loss-clock.html#correlation-of-gene-expression-to-clock"><i class="fa fa-check"></i><b>5.6</b> Correlation of gene expression to clock</a></li>
<li class="chapter" data-level="5.7" data-path="loss-clock.html"><a href="loss-clock.html#plot-score-in-different-tor-regimes"><i class="fa fa-check"></i><b>5.7</b> Plot score in different TOR regimes</a></li>
<li class="chapter" data-level="5.8" data-path="loss-clock.html"><a href="loss-clock.html#plot-high-low-clock-tumors-vs-tor"><i class="fa fa-check"></i><b>5.8</b> Plot High-Low clock tumors vs TOR</a></li>
<li class="chapter" data-level="5.9" data-path="loss-clock.html"><a href="loss-clock.html#plot-correlation-of-loci-to-clock-methylation"><i class="fa fa-check"></i><b>5.9</b> Plot correlation of loci to clock methylation</a></li>
<li class="chapter" data-level="5.10" data-path="loss-clock.html"><a href="loss-clock.html#plot-distribution-of-methylation-of-clock-in-normal-samples"><i class="fa fa-check"></i><b>5.10</b> Plot distribution of methylation of clock in normal samples</a></li>
<li class="chapter" data-level="5.11" data-path="loss-clock.html"><a href="loss-clock.html#compare-clock-layer-to-biological-age"><i class="fa fa-check"></i><b>5.11</b> Compare clock layer to biological age</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html"><i class="fa fa-check"></i><b>6</b> Epigenomic instability</a><ul>
<li class="chapter" data-level="6.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#compare-features-to-each-other"><i class="fa fa-check"></i><b>6.1</b> Compare features to each other</a></li>
<li class="chapter" data-level="6.2" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#annotate-epignomic-instability-scores"><i class="fa fa-check"></i><b>6.2</b> Annotate epignomic instability scores</a><ul>
<li class="chapter" data-level="6.2.1" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#integrative-clusters"><i class="fa fa-check"></i><b>6.2.1</b> Integrative clusters</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#cross-correlation-methylation-expression-of-epigenomic-instability-loci"><i class="fa fa-check"></i><b>6.3</b> Cross correlation (methylation-expression) of epigenomic instability loci</a></li>
<li class="chapter" data-level="6.4" data-path="epigenomic-instability.html"><a href="epigenomic-instability.html#epigenomic-instability-in-different-time-of-replication-tor-regimes"><i class="fa fa-check"></i><b>6.4</b> Epigenomic instability in different Time of replication (TOR) regimes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cna.html"><a href="cna.html"><i class="fa fa-check"></i><b>7</b> CNA</a></li>
<li class="chapter" data-level="8" data-path="mutations.html"><a href="mutations.html"><i class="fa fa-check"></i><b>8</b> Mutations</a></li>
<li class="chapter" data-level="9" data-path="survival.html"><a href="survival.html"><i class="fa fa-check"></i><b>9</b> Survival</a><ul>
<li class="chapter" data-level="9.0.1" data-path="survival.html"><a href="survival.html#raw-methylation"><i class="fa fa-check"></i><b>9.0.1</b> Raw-methylation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html"><i class="fa fa-check"></i><b>10</b> Methylation - Expression <em>in cis</em></a><ul>
<li class="chapter" data-level="10.0.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#remove-tme-genes"><i class="fa fa-check"></i><b>10.0.1</b> Remove TME genes</a></li>
<li class="chapter" data-level="10.0.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#load-normalized-methylation-and-separate-it-to-promoters-and-non-promoters"><i class="fa fa-check"></i><b>10.0.2</b> Load normalized methylation and separate it to promoters and non-promoters</a></li>
<li class="chapter" data-level="10.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-promoters"><i class="fa fa-check"></i><b>10.1</b> Cis candidates: promoters</a><ul>
<li class="chapter" data-level="10.1.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-correlation-of-top-cis-regulated-promoter-candidates"><i class="fa fa-check"></i><b>10.1.1</b> Plot correlation of top cis-regulated promoter candidates</a></li>
<li class="chapter" data-level="10.1.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#annotation-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.2</b> Annotation of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#gene-expression-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.1.3</b> Gene expression of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.1.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#correlation-between-cis-regulated-genes"><i class="fa fa-check"></i><b>10.1.4</b> Correlation between cis-regulated genes</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-promoters"><i class="fa fa-check"></i><b>10.2</b> Epipolymorphism of cis-regulated promoters</a></li>
<li class="chapter" data-level="10.3" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#promoter-examples"><i class="fa fa-check"></i><b>10.3</b> Promoter examples</a><ul>
<li class="chapter" data-level="10.3.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#brca1"><i class="fa fa-check"></i><b>10.3.1</b> BRCA1</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#cis-candidates-genomic"><i class="fa fa-check"></i><b>10.4</b> Cis candidates: genomic</a><ul>
<li class="chapter" data-level="10.4.1" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-cis"><i class="fa fa-check"></i><b>10.4.1</b> Plot % cis</a></li>
<li class="chapter" data-level="10.4.2" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#plot-distance-to-most-correlated-gene"><i class="fa fa-check"></i><b>10.4.2</b> Plot distance to most correlated gene</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#epipolymorphism-of-cis-regulated-genomic"><i class="fa fa-check"></i><b>10.5</b> Epipolymorphism of cis-regulated genomic</a></li>
<li class="chapter" data-level="10.6" data-path="methylation-expression-in-cis.html"><a href="methylation-expression-in-cis.html#examples-for-cis-regulation-non-promoters"><i class="fa fa-check"></i><b>10.6</b> Examples for cis regulation (non-promoters)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html"><i class="fa fa-check"></i><b>11</b> X dosage compensation</a><ul>
<li class="chapter" data-level="11.1" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#x-analysis"><i class="fa fa-check"></i><b>11.1</b> X analysis</a></li>
<li class="chapter" data-level="11.2" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#autosome-analysis"><i class="fa fa-check"></i><b>11.2</b> autosome analysis</a></li>
<li class="chapter" data-level="11.3" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#checking-dosage-compensation-on-all-x-loci"><i class="fa fa-check"></i><b>11.3</b> Checking dosage compensation on all X loci</a></li>
<li class="chapter" data-level="11.4" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#of-x-lost"><i class="fa fa-check"></i><b>11.4</b> % of X lost</a></li>
<li class="chapter" data-level="11.5" data-path="x-dosage-compensation.html"><a href="x-dosage-compensation.html#plot-expression-methylation-correlation-for-x"><i class="fa fa-check"></i><b>11.5</b> Plot Expression-Methylation correlation for X</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="finding-specific-dosage-compensation-examples.html"><a href="finding-specific-dosage-compensation-examples.html"><i class="fa fa-check"></i><b>12</b> Finding specific dosage compensation examples</a></li>
<li class="chapter" data-level="13" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html"><i class="fa fa-check"></i><b>13</b> Export supplementary tables</a><ul>
<li class="chapter" data-level="13.1" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-1-metabric-samples-metadata"><i class="fa fa-check"></i><b>13.1</b> Table 1: METABRIC samples metadata</a></li>
<li class="chapter" data-level="13.2" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-2-immune-expression-signature"><i class="fa fa-check"></i><b>13.2</b> Table 2: Immune expression signature</a></li>
<li class="chapter" data-level="13.3" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-3-caf-expression-signature"><i class="fa fa-check"></i><b>13.3</b> Table 3: CAF expression signature</a></li>
<li class="chapter" data-level="13.4" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-4-expression-methylation-correlation"><i class="fa fa-check"></i><b>13.4</b> Table 4: Expression-Methylation correlation</a></li>
<li class="chapter" data-level="13.5" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-5-methylation-scores"><i class="fa fa-check"></i><b>13.5</b> Table 5: Methylation scores</a></li>
<li class="chapter" data-level="13.6" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-6-gene-expression-correlation-to-epigenomic-instability"><i class="fa fa-check"></i><b>13.6</b> Table 6: Gene expression correlation to epigenomic instability</a></li>
<li class="chapter" data-level="13.7" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-7-methylation-layers-loci"><i class="fa fa-check"></i><b>13.7</b> Table 7: Methylation layers loci</a></li>
<li class="chapter" data-level="13.8" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-8-cis-regulation-candidates"><i class="fa fa-check"></i><b>13.8</b> Table 8: Cis regulation candidates</a></li>
<li class="chapter" data-level="13.9" data-path="export-supplementary-tables.html"><a href="export-supplementary-tables.html#table-9-dosage-compensation-in-autosomes"><i class="fa fa-check"></i><b>13.9</b> Table 9: Dosage compensation in autosomes</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html"><i class="fa fa-check"></i><b>14</b> Files needed for the analysis</a><ul>
<li class="chapter" data-level="14.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#misha-db"><i class="fa fa-check"></i><b>14.1</b> Misha DB</a></li>
<li class="chapter" data-level="14.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#additional-files"><i class="fa fa-check"></i><b>14.2</b> Additional files</a><ul>
<li class="chapter" data-level="14.2.1" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#metabric-data"><i class="fa fa-check"></i><b>14.2.1</b> METABRIC data</a></li>
<li class="chapter" data-level="14.2.2" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#notable-generated-files"><i class="fa fa-check"></i><b>14.2.2</b> Notable generated files</a></li>
<li class="chapter" data-level="14.2.3" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#spatial-methylation-distribution"><i class="fa fa-check"></i><b>14.2.3</b> Spatial methylation distribution</a></li>
<li class="chapter" data-level="14.2.4" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#copy-number-abberations"><i class="fa fa-check"></i><b>14.2.4</b> Copy number abberations</a></li>
<li class="chapter" data-level="14.2.5" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#genomic-annotations"><i class="fa fa-check"></i><b>14.2.5</b> Genomic annotations</a></li>
<li class="chapter" data-level="14.2.6" data-path="files-needed-for-the-analysis.html"><a href="files-needed-for-the-analysis.html#external-resources"><i class="fa fa-check"></i><b>14.2.6</b> External resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figures-key.html"><a href="figures-key.html"><i class="fa fa-check"></i><b>15</b> Figures key</a><ul>
<li class="chapter" data-level="15.1" data-path="figures-key.html"><a href="figures-key.html#figure-1"><i class="fa fa-check"></i><b>15.1</b> Figure 1</a></li>
<li class="chapter" data-level="15.2" data-path="figures-key.html"><a href="figures-key.html#figure-3"><i class="fa fa-check"></i><b>15.2</b> Figure 3</a></li>
<li class="chapter" data-level="15.3" data-path="figures-key.html"><a href="figures-key.html#figure-4"><i class="fa fa-check"></i><b>15.3</b> Figure 4</a></li>
<li class="chapter" data-level="15.4" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-1"><i class="fa fa-check"></i><b>15.4</b> Extended Data Figure 1</a></li>
<li class="chapter" data-level="15.5" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-2"><i class="fa fa-check"></i><b>15.5</b> Extended Data Figure 2</a></li>
<li class="chapter" data-level="15.6" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-3"><i class="fa fa-check"></i><b>15.6</b> Extended Data Figure 3</a></li>
<li class="chapter" data-level="15.7" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-4"><i class="fa fa-check"></i><b>15.7</b> Extended Data Figure 4</a></li>
<li class="chapter" data-level="15.8" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-5"><i class="fa fa-check"></i><b>15.8</b> Extended Data Figure 5</a></li>
<li class="chapter" data-level="15.9" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-6"><i class="fa fa-check"></i><b>15.9</b> Extended Data Figure 6</a></li>
<li class="chapter" data-level="15.10" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-7"><i class="fa fa-check"></i><b>15.10</b> Extended Data Figure 7</a></li>
<li class="chapter" data-level="15.11" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-8"><i class="fa fa-check"></i><b>15.11</b> Extended Data Figure 8</a></li>
<li class="chapter" data-level="15.12" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-9"><i class="fa fa-check"></i><b>15.12</b> Extended Data Figure 9</a></li>
<li class="chapter" data-level="15.13" data-path="figures-key.html"><a href="figures-key.html#extended-data-figure-10"><i class="fa fa-check"></i><b>15.13</b> Extended Data Figure 10</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DNA methylation landscapes of 1538 breast cancers reveal a replication-linked clock, epigenomic instability and cis-regulation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="survival" class="section level1">
<h1><span class="header-section-number">9</span> Survival</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;scripts/init.R&quot;</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy &lt;-<span class="st"> </span><span class="kw">get_all_features</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>immune, <span class="op">-</span>caf) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">rename</span>(<span class="dt">Immune =</span> immune.meth, <span class="dt">CAF =</span> caf.meth, <span class="dt">Clock =</span> clock) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;score&quot;</span>, <span class="op">-</span>ER, <span class="op">-</span>samp) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">feat =</span> <span class="kw">factor</span>(feat, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;CAF&quot;</span>, <span class="st">&quot;Immune&quot;</span>, <span class="st">&quot;Clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>)))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">surv_df &lt;-<span class="st"> </span>survival <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(feats_tidy, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">group_by</span>(ER, feat) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(score)) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cut</span>(score, <span class="dt">breaks =</span> <span class="kw">quantile</span>(score, <span class="dv">0</span><span class="op">:</span><span class="dv">3</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;mid&quot;</span>, <span class="st">&quot;high&quot;</span>), <span class="dt">include.lowest =</span> <span class="ot">TRUE</span>)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">filter</span>(score <span class="op">!=</span><span class="st"> &quot;mid&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">factor</span>(score, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>)))

surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(feat, ER, score)</code></pre></div>
<pre><code>## # A tibble: 20 x 4
##     feat  ER score   n
## 1    CAF ER-   low 104
## 2    CAF ER-  high 103
## 3    CAF ER+   low 370
## 4    CAF ER+  high 369
## 5 Immune ER-   low 104
## 6 Immune ER-  high 103
## # ... with 14 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(feat, ER)</code></pre></div>
<pre><code>## # A tibble: 10 x 3
##     feat  ER   n
## 1    CAF ER- 207
## 2    CAF ER+ 739
## 3 Immune ER- 207
## 4 Immune ER+ 739
## 5  Clock ER- 207
## 6  Clock ER+ 739
## # ... with 4 more rows</code></pre>
<div id="figure-4d" class="section level4">
<h4><span class="header-section-number">9.0.0.1</span> Figure 4D</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">12</span>, <span class="dt">repr.plot.height =</span> <span class="dv">3</span>)

p_surv_pos &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="st">&quot;feat&quot;</span>, <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">censor.size =</span> <span class="dv">3</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>, <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 10</code></pre>
<pre><code>## Warning: `as.tibble()` was deprecated in tibble 2.0.0.
## Please use `as_tibble()` instead.
## The signature and semantics have changed, see `?as_tibble`.</code></pre>
<pre><code>## Warning: `select_()` was deprecated in dplyr 0.7.0.
## Please use `select()` instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_neg &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER-&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="st">&quot;feat&quot;</span>, <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>, <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 10</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_pos <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_neg <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(feat, ER, score) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">n_death =</span> <span class="kw">sum</span>(y <span class="op">&lt;</span><span class="st"> </span><span class="dv">5</span> <span class="op">&amp;</span><span class="st"> </span>death <span class="op">==</span><span class="st"> </span><span class="dv">1</span>), <span class="dt">n_surv =</span> <span class="kw">sum</span>(y <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>), <span class="dt">n =</span> n_surv <span class="op">+</span><span class="st"> </span>n_death, <span class="dt">surv =</span> n_surv <span class="op">/</span><span class="st"> </span>n, <span class="dt">surv_p =</span> scales<span class="op">::</span><span class="kw">percent</span>(surv))  <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</code></pre></div>
<pre><code>##      feat  ER score n_death n_surv   n      surv surv_p
## 1     CAF ER-   low      45     51  96 0.5312500    53%
## 2     CAF ER-  high      35     63  98 0.6428571    64%
## 3     CAF ER+   low      61    277 338 0.8195266    82%
## 4     CAF ER+  high      39    302 341 0.8856305    89%
## 5  Immune ER-   low      47     47  94 0.5000000    50%
## 6  Immune ER-  high      25     71  96 0.7395833    74%
## 7  Immune ER+   low      49    289 338 0.8550296    86%
## 8  Immune ER+  high      47    293 340 0.8617647    86%
## 9   Clock ER-   low      38     60  98 0.6122449    61%
## 10  Clock ER-  high      34     61  95 0.6421053    64%
## 11  Clock ER+   low      51    290 341 0.8504399    85%
## 12  Clock ER+  high      35    298 333 0.8948949    89%
## 13     MG ER-   low      36     64 100 0.6400000    64%
## 14     MG ER-  high      41     51  92 0.5543478    55%
## 15     MG ER+   low      31    308 339 0.9085546    91%
## 16     MG ER+  high      57    281 338 0.8313609    83%
## 17     ML ER-   low      41     56  97 0.5773196    58%
## 18     ML ER-  high      32     61  93 0.6559140    66%
## 19     ML ER+   low      35    309 344 0.8982558    90%
## 20     ML ER+  high      64    267 331 0.8066465    81%</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>, feat <span class="op">==</span><span class="st"> &quot;MG&quot;</span>)
survival<span class="op">::</span><span class="kw">survdiff</span>(survival<span class="op">::</span><span class="kw">Surv</span>(df<span class="op">$</span>y, df<span class="op">$</span>death) <span class="op">~</span><span class="st"> </span>score, <span class="dt">data =</span> <span class="kw">as.data.frame</span>(df))</code></pre></div>
<pre><code>## Call:
## survival::survdiff(formula = survival::Surv(df$y, df$death) ~ 
##     score, data = as.data.frame(df))
## 
##              N Observed Expected (O-E)^2/E (O-E)^2/V
## score=low  370       95      123      6.55      14.1
## score=high 369      136      108      7.51      14.1
## 
##  Chisq= 14.1  on 1 degrees of freedom, p= 0.0002</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER-&quot;</span>, feat <span class="op">==</span><span class="st"> &quot;MG&quot;</span>)
survival<span class="op">::</span><span class="kw">survdiff</span>(survival<span class="op">::</span><span class="kw">Surv</span>(df<span class="op">$</span>y, df<span class="op">$</span>death) <span class="op">~</span><span class="st"> </span>score, <span class="dt">data =</span> <span class="kw">as.data.frame</span>(df))</code></pre></div>
<pre><code>## Call:
## survival::survdiff(formula = survival::Surv(df$y, df$death) ~ 
##     score, data = as.data.frame(df))
## 
##              N Observed Expected (O-E)^2/E (O-E)^2/V
## score=low  104       44     51.1     0.977      2.12
## score=high 103       51     43.9     1.136      2.12
## 
##  Chisq= 2.1  on 1 degrees of freedom, p= 0.1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>, feat <span class="op">==</span><span class="st"> &quot;MG&quot;</span>)
survival<span class="op">::</span><span class="kw">coxph</span>(survival<span class="op">::</span><span class="kw">Surv</span>(df<span class="op">$</span>y, df<span class="op">$</span>death) <span class="op">~</span><span class="st"> </span>score, <span class="dt">data =</span> <span class="kw">as.data.frame</span>(df))</code></pre></div>
<pre><code>## Call:
## survival::coxph(formula = survival::Surv(df$y, df$death) ~ score, 
##     data = as.data.frame(df))
## 
##             coef exp(coef) se(coef)     z        p
## scorehigh 0.4998    1.6484   0.1343 3.722 0.000198
## 
## Likelihood ratio test=14.14  on 1 df, p=0.0001699
## n= 739, number of events= 231</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER-&quot;</span>, feat <span class="op">==</span><span class="st"> &quot;MG&quot;</span>)
survival<span class="op">::</span><span class="kw">coxph</span>(survival<span class="op">::</span><span class="kw">Surv</span>(df<span class="op">$</span>y, df<span class="op">$</span>death) <span class="op">~</span><span class="st"> </span>score, <span class="dt">data =</span> <span class="kw">as.data.frame</span>(df))</code></pre></div>
<pre><code>## Call:
## survival::coxph(formula = survival::Surv(df$y, df$death) ~ score, 
##     data = as.data.frame(df))
## 
##             coef exp(coef) se(coef)     z     p
## scorehigh 0.2988    1.3483   0.2060 1.451 0.147
## 
## Likelihood ratio test=2.11  on 1 df, p=0.1461
## n= 207, number of events= 95</code></pre>
</div>
<div id="extended-data-figure-10d" class="section level4">
<h4><span class="header-section-number">9.0.0.2</span> Extended Data Figure 10D</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span>survival <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(feats_tidy, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">left_join</span>(<span class="kw">get_mut_df</span>() <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(gene <span class="op">==</span><span class="st"> &quot;TP53&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(mutation)) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">select</span>(samp, gene, mutation)) <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>, <span class="op">!</span><span class="kw">is.na</span>(mutation)) </code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_p53 &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">group_by</span>(ER, feat) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(score)) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cut</span>(score, <span class="dt">breaks =</span> <span class="kw">quantile</span>(score, <span class="dv">0</span><span class="op">:</span><span class="dv">3</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;mid&quot;</span>, <span class="st">&quot;high&quot;</span>), <span class="dt">include.lowest =</span> <span class="ot">TRUE</span>)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">filter</span>(score <span class="op">!=</span><span class="st"> &quot;mid&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">factor</span>(score, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>))) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="kw">c</span>(<span class="st">&quot;mutation&quot;</span>, <span class="st">&quot;feat&quot;</span>), <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">censor.size =</span> <span class="dv">3</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 20</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_p53 <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(samp, ER, mutation) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(mutation)</code></pre></div>
<pre><code>## # A tibble: 2 x 2
##   mutation   n
## 1     MUT- 202
## 2   NO MUT 824</code></pre>
</div>
<div id="raw-methylation" class="section level3">
<h3><span class="header-section-number">9.0.1</span> Raw-methylation</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feats_tidy &lt;-<span class="st"> </span><span class="kw">get_all_features_raw</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>immune, <span class="op">-</span>caf) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">rename</span>(<span class="dt">Immune =</span> immune.meth, <span class="dt">CAF =</span> caf.meth, <span class="dt">Clock =</span> clock) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">gather</span>(<span class="st">&quot;feat&quot;</span>, <span class="st">&quot;score&quot;</span>, <span class="op">-</span>ER, <span class="op">-</span>samp) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">feat =</span> <span class="kw">factor</span>(feat, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;CAF&quot;</span>, <span class="st">&quot;Immune&quot;</span>, <span class="st">&quot;Clock&quot;</span>, <span class="st">&quot;MG&quot;</span>, <span class="st">&quot;ML&quot;</span>)))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">surv_df &lt;-<span class="st"> </span>survival <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">select</span>(<span class="op">-</span>ER) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(feats_tidy, <span class="dt">by =</span> <span class="st">&quot;samp&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">!=</span><span class="st"> &quot;normal&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">        </span><span class="kw">group_by</span>(ER, feat) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(<span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(score)) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cut</span>(score, <span class="dt">breaks =</span> <span class="kw">quantile</span>(score, <span class="dv">0</span><span class="op">:</span><span class="dv">3</span> <span class="op">/</span><span class="st"> </span><span class="dv">3</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;mid&quot;</span>, <span class="st">&quot;high&quot;</span>), <span class="dt">include.lowest =</span> <span class="ot">TRUE</span>)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(score <span class="op">!=</span><span class="st"> &quot;mid&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">factor</span>(score, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>)))

surv_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(feat, ER, score)</code></pre></div>
<pre><code>## # A tibble: 20 x 4
##     feat  ER score   n
## 1    CAF ER-   low 104
## 2    CAF ER-  high 103
## 3    CAF ER+   low 370
## 4    CAF ER+  high 369
## 5 Immune ER-   low 104
## 6 Immune ER-  high 103
## # ... with 14 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">12</span>, <span class="dt">repr.plot.height =</span> <span class="dv">3</span>)

p_surv_pos &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="st">&quot;feat&quot;</span>, <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">censor.size =</span> <span class="dv">3</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>, <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 10</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_neg &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER-&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="st">&quot;feat&quot;</span>, <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>, <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 10</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_pos <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_neg <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-13-2.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">repr.plot.width =</span> <span class="dv">12</span>, <span class="dt">repr.plot.height =</span> <span class="dv">6</span>)
surv_df_p53 &lt;-<span class="st"> </span>surv_df <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">left_join</span>(<span class="kw">get_mut_df</span>() <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(gene <span class="op">==</span><span class="st"> &quot;TP53&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(mutation)) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">select</span>(samp, gene, mutation)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">            </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(mutation))</code></pre></div>
<pre><code>## Joining, by = &quot;samp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_p53 &lt;-<span class="st"> </span>surv_df_p53 <span class="op">%&gt;%</span><span class="st">        </span>
<span class="st">        </span><span class="kw">filter</span>(ER <span class="op">==</span><span class="st"> &quot;ER+&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">plot_km</span>(<span class="st">&quot;score&quot;</span>, <span class="dt">facet.by =</span> <span class="kw">c</span>(<span class="st">&quot;mutation&quot;</span>, <span class="st">&quot;feat&quot;</span>), <span class="dt">ggtheme =</span> <span class="kw">theme_arial</span>(<span class="dv">6</span>), <span class="dt">time_breaks =</span> <span class="dv">5</span>, <span class="dt">legend =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">censor.size =</span> <span class="dv">3</span>, <span class="dt">colors =</span> <span class="st">&quot;aaas&quot;</span>, <span class="dt">time_range =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">censor =</span> <span class="ot">FALSE</span>, <span class="dt">pval =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in (function (survsummary, times, survtable = c(&quot;cumevents&quot;,
## &quot;risk.table&quot;, : The length of legend.labs should be 20</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_surv_p53 <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="survival_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="methylation-expression-in-cis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
